<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Robin Wellmann" />

<meta name="date" content="2017-05-23" />

<title></title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore"><ol start="3" style="list-style-type: decimal">
<li>Optimum Contribution Selection</li>
</ol></h1>
<h4 class="author"><em>Robin Wellmann</em></h4>
<h4 class="date"><em>2017-05-23</em></h4>



<p>The aim of optimum contribution selection is to find the optimum number of offspring for each breeding animal and to determine if a young animal (a selection candidate) should be selected for breeding or not. This is done in an optimal way, i.e.Â in a way that ensures that genetic gain is achieved, and that genetic diversity and genetic originality of the population are maintained or recovered. It can be based either on pedigree data or on marker data, whereby the latter approach is recommended. It requires that this data is available for all selection candidates, or, at least from a large sample of selection candidates.</p>
<p>Even if the frequency of use of breeding animals is not regulated by the breeding organization, running the optimization still provides valuable information for a breeder, as the animals with highest optimum contributions are most valuable for a breeding program.</p>
<p>This vignette is organized as follows:</p>
<ul>
<li><a href="#example-data-set">Example Data Set</a></li>
<li><a href="#introductory-example-traditional-ocs">Introductory Example: Traditional OCS</a></li>
<li><a href="#defining-the-objective-of-a-breeding-program">Defining the Objective of a Breeding Program</a></li>
<li><a href="#marker-based-ocs">Marker-based OCS</a>
<ul>
<li><a href="#maximize-genetic-gain-marker-based">Maximize Genetic Gain</a></li>
<li><a href="#minimize-inbreeding-marker-based">Minimize Inbreeding</a></li>
<li><a href="#recover-the-native-genetic-background-marker-based">Recover the Native Genetic Background</a></li>
<li><a href="#increase-diversity-between-breeds">Increase Diversity Between Breeds</a></li>
</ul></li>
<li><a href="#pedigree-based-ocs">Pedigree-based OCS</a>
<ul>
<li><a href="#maximize-genetic-gain-pedigree-based">Maximize Genetic Gain</a></li>
<li><a href="#minimize-inbreeding-pedigree-based">Minimize Inbreeding</a></li>
<li><a href="#recover-the-native-genetic-background-pedigree-based">Recover the Native Genetic Background</a></li>
</ul></li>
</ul>
<div id="example-data-set" class="section level2">
<h2>Example Data Set</h2>
<p>All evaluations using marker data are demonstrated at the example of cattle data included in the package. This multi-breed data has already been described in the <a href="seg-vignette.html">companion vignette for basic marker-based evaluations</a>.</p>
<p>Data frame <code>Cattle</code> includes the phenotypic information and has columns <code>Indiv</code> (individual IDs), <code>Born</code> (year of birth), <code>Breed</code> (breed name), <code>BV</code> (breeding values), <code>Sex</code> (sexes), and <code>herd</code> (herd).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;optiSel&quot;</span>)
<span class="kw">data</span>(Cattle)
<span class="kw">head</span>(Cattle)</code></pre></div>
<pre><code>##                           Indiv Born  Breed         BV    Sex herd
## 276000101676415 276000101676415 1991 Angler -1.0706066   male &lt;NA&gt;
## 276000108612636 276000108612636 1994 Angler -0.3362574 female    2
## 276000102372349 276000102372349 1986 Angler -2.0735649 female    1
## 276000102379430 276000102379430 1987 Angler  1.5968307   male &lt;NA&gt;
## 276000108826036 276000108826036 1994 Angler  1.0023969   male &lt;NA&gt;
## 276000111902076 276000111902076 1998 Angler -0.2426676   male &lt;NA&gt;</code></pre>
<p>The data frame contains information on the 4 breeds Angler, Fleckvieh, Holstein, Rotbunt. The âAnglerâ is an endangered German cattle breed, which had been upgraded with Red Holstein (also called âRotbuntâ). The Rotbunt cattle are a subpopulation of the âHolsteinâ breed. The âFleckviehâ or Simmental breed is unrelated to the Angler. The Angler cattle are the selection candidates.</p>
<p>This small example data set contains only genotypes from the first parts of the first two chromosomes. Vector <code>GTfiles</code> defined below contains the names of the genotype files. There is one file for each chromosome. Data frame <code>map</code> contains the marker map and has columns <code>Name</code> (marker name), <code>Chr</code> (chromosome number), <code>Position</code>, <code>Mb</code> (position in Mega base pairs), and <code>cM</code> (position in centiMorgan):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(map)
dir     &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package=</span><span class="st">&quot;optiSel&quot;</span>)
GTfiles &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir, <span class="kw">paste</span>(<span class="st">&quot;Chr&quot;</span>, <span class="kw">unique</span>(map$Chr), <span class="st">&quot;.phased&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))
<span class="kw">head</span>(map)</code></pre></div>
<pre><code>##                                    Name Chr Position cM       Mb
## ARS-BFGL-NGS-16466   ARS-BFGL-NGS-16466   1   267940  0 0.267940
## ARS-BFGL-NGS-98142   ARS-BFGL-NGS-98142   1   471078  0 0.471078
## ARS-BFGL-NGS-114208 ARS-BFGL-NGS-114208   1   533815  0 0.533815
## ARS-BFGL-NGS-65067   ARS-BFGL-NGS-65067   1   883895  0 0.883895
## ARS-BFGL-BAC-32722   ARS-BFGL-BAC-32722   1   929617  0 0.929617
## ARS-BFGL-BAC-34682   ARS-BFGL-BAC-34682   1   950841  0 0.950841</code></pre>
</div>
<div id="introductory-example-traditional-ocs" class="section level2">
<h2>Introductory Example: Traditional OCS</h2>
<p>As an introductory example you may run a traditional OCS with marker based kinship matrices. All alternative approaches involve the same steps, so it is recommended to read this section even if you want to minimize inbreeding instead of maximizing genetic gain. The following steps are involved:</p>
<p>Compute the kinships that are to be managed and put them into a list with function <a href="https://rdrr.io/cran/optiSel/man/kinlist.html">kinlist</a>. Below, the kinship is named <code>sKin</code>, which is a shorthand for <em>segment based kinship</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Kin  &lt;-<span class="st"> </span><span class="kw">kinlist</span>(<span class="dt">sKin=</span><span class="kw">segIBD</span>(GTfiles, map))</code></pre></div>
<p>Define a data frame containing only the phenotypes of the selection candidates. Make sure that there is one column for each trait that should be improved.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phen &lt;-<span class="st"> </span>Cattle[Cattle$Breed==<span class="st">&quot;Angler&quot;</span>,]</code></pre></div>
<p>Display the objective functions and constraints that are available for your data with function <a href="https://rdrr.io/cran/optiSel/man/help.opticont.html">help.opticont</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help.opticont</span>(Kin, phen)</code></pre></div>
<pre><code>## Available objective functions: 
##        min.sKin  
##        min.BV  
##        max.BV  
## 
## Available constraints: 
##        ub.sKin  
##        lb.BV  
##        ub.BV  
##        eq.BV  
##        ub  lb
## 
## Attention: Probably not all these objective functions and
##            constraints make sense in animal breeding</code></pre>
<p>For numeric columns in data frame <code>phen</code> the possibility is provided to define an upper bound (prefix <code>ub</code>), a lower bound (prefix <code>lb</code>), an equality constraint (prefix <code>eq</code>), or to minimize (prefix <code>min</code>) or to maximize (prefix <code>max</code>) the weighted sum of the values, whereby the weights are the contributions of the selection candidates. If the column contains breeding values, then this is the expected mean breeding value in the offspring.</p>
<p>For each kinship named by function <a href="https://rdrr.io/cran/optiSel/man/kinlist.html">kinlist</a>, the possibility is provided to define an upper bound for the expected mean value of the kinship in the offspring (prefix <code>ub</code>), or to minimize the value (prefix <code>min</code>).</p>
<p>Constraints <code>ub</code> and <code>lb</code> allow to define upper bounds and lower bounds for the contributions of the selection candidates.</p>
<p>Now choose the parameters you want to restrict and the parameters you want to optimize. For traditional OCs the objective is to maximize genetic gain with method <code>max.BV</code>, and to restrict the mean kinship in the offspring by defining constraint <code>ub.sKin</code>.</p>
<p>Create an empty list for the constraints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con &lt;-<span class="st"> </span><span class="kw">list</span>()</code></pre></div>
<p>and put the constraints into the list. To define upper bounds for the contributions of the selection candidates you may define component <code>ub</code> as</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$ub &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>)</code></pre></div>
<p>In this case, <code>ub</code> is a named vector with two components, which define the upper bounds for the contributions of males and females. The value <code>F=-1</code> means that females have equal contributions, so optimization will be done only for the males, and <code>M=NA</code> means that there is no upper bound for the contributions of males. Alternatively, different lower bounds and upper bounds could be defined for each selection candidate (see the help page of function <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a>).</p>
<p>To define an upper bound for the mean kinship <code>sKin</code> in the offspring, put component <code>ub.sKin</code> into the list. In general, if an upper bound for a kinship <span class="math inline">\(K\)</span> should be defined, it is recommended to derive the threshold value from the desired effective size <span class="math inline">\(N_e\)</span> of the population by the formula <span class="math display">\[ub.K=\overline{K}+(1-\overline{K})\Delta F,\]</span> where <span class="math inline">\(\overline{K}\)</span> is the mean kinship in the actual population, and <span class="math inline">\(\Delta F=\frac{1}{2 N_e}\)</span>. We will see below that the mean kinship in the actual population can be estimated as 0.0569. The critical effective size, i. e. the size below which the fitness of the population steadily decreases, depends on the population and is usually between 50 and 100. But there seems to be a consensus that 50-100 is a long-term viable effective size. To be on the safe side, an effective size of <span class="math inline">\(N_e=100\)</span> should be envisaged <span class="citation">(T H E Meuwissen 2009)</span>. Thus, the constraint is defined as</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Ne &lt;-<span class="st"> </span><span class="dv">100</span>
meanKin     &lt;-<span class="st"> </span><span class="fl">0.0569</span>
con$ub.sKin &lt;-<span class="st"> </span>meanKin +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))</code></pre></div>
<p>Now the optimum contributions of the selection candidates can be calculated:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;max.BV&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## Objective: maximizing mean BV of the offspring.
## Constraints:
##   Mean kinship sKin in the offspring is not exceeding ub.sKin.
##   Minimum contribution of males to offspring is defined.
##   Minimum contribution of females to offspring is defined
##   Number of offspring of males is not limited.
##     (Thus, the maximum contribution per male to the offspring is 0.5).
##   All females have equal contributions to the offspring.
##     (Thus, no optimization is done for the females).
##   The total genetic contribution of   males to offspring is 0.5.
##   The total genetic contribution of females to offspring is 0.5.
## 
## Using solver cccp2 with parameters: trace=0, abstol=1e-05, feastol=1e-05, stepadj=0.9, maxiters=100, reltol=1e-06, beta=0.5</code></pre>
<p>Check if the results fulfill the constraints and look at the value of the objective function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV.s &lt;-<span class="st"> </span><span class="kw">summary</span>(maxBV)</code></pre></div>
<pre><code>## Checking constraints:
##   min(oc) &gt;= 0           : TRUE
##   total male cont   = 0.5: TRUE
##   total female cont = 0.5: TRUE
##   females have equal cont: TRUE
##   all male cont &lt;= ub    : TRUE
##   mean sKin &lt;= ub.sKin       : TRUE
## </code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV.s$obj.fun</code></pre></div>
<pre><code>## [1] 1.039909</code></pre>
<p>The results look OK. If they are not OK, then try to use another solver. The solver can be specified in parameter <code>solver</code> of function <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a>. Available solvers are <code>&quot;alabama&quot;</code>, <code>&quot;cccp&quot;</code>, <code>&quot;cccp2&quot;</code>, <code>&quot;csdp&quot;</code>, and <code>&quot;slsqp&quot;</code>. The default is <code>&quot;cccp&quot;</code>. The solvers are described in the help page of function <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a>. Alternatively you may use the same solver but with different tuning parameters. The available paramters are displayed if the function <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a> is called (as shown above).</p>
<p>The optimum contributions of the selection candidates are in component <code>parent</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Candidate &lt;-<span class="st"> </span>maxBV$parent[,  <span class="kw">c</span>(<span class="st">&quot;Indiv&quot;</span>, <span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;oc&quot;</span>)]
<span class="kw">head</span>(Candidate[<span class="kw">rev</span>(<span class="kw">order</span>(Candidate$oc)),])</code></pre></div>
<pre><code>##                           Indiv  Sex         oc
## 276000121507437 276000121507437 male 0.06642355
## 276000121243787 276000121243787 male 0.06541582
## 276000120949468 276000120949468 male 0.06252814
## 276000120061822 276000120061822 male 0.06052558
## 276000110948577 276000110948577 male 0.05587209
## 276000113913566 276000113913566 male 0.04060682</code></pre>
<p>The optimum numbers of offspring can be obtained from the optimum contributions and the size <code>N</code> of the offspring population with function <a href="https://rdrr.io/cran/optiSel/man/noffspring.html">noffspring</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Candidate$nOff &lt;-<span class="st"> </span><span class="kw">noffspring</span>(Candidate, <span class="dt">N=</span><span class="dv">250</span>)$nOff
<span class="kw">head</span>(Candidate[<span class="kw">rev</span>(<span class="kw">order</span>(Candidate$oc)),])</code></pre></div>
<pre><code>##                           Indiv  Sex         oc nOff
## 276000121507437 276000121507437 male 0.06642355   33
## 276000121243787 276000121243787 male 0.06541582   33
## 276000120949468 276000120949468 male 0.06252814   31
## 276000120061822 276000120061822 male 0.06052558   30
## 276000110948577 276000110948577 male 0.05587209   28
## 276000113913566 276000113913566 male 0.04060682   20</code></pre>
<p>Males and females can be allocated for mating with function matings such that all breeding animals have the desired number of offspring. In the example below the mean marker-based inbreeding coefficient in the offspring is minimized. Since the default value for parameter <code>N</code> is used, it is assumed that females have 2 offspring on average available as selection candidates in the next generation. Parameter <code>alpha=0.3</code> means that in each herd at most 30% of the cows are mated with the same bull. This increases genetic connectedness between herds and enables to estimate more accurate breeding values.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Mating &lt;-<span class="st"> </span><span class="kw">matings</span>(maxBV$parent, <span class="dt">Kin=</span><span class="kw">segIBD</span>(GTfiles, map),  <span class="dt">alpha=</span><span class="fl">0.3</span>)</code></pre></div>
<pre><code>## Using  skip = 2 
## Using cskip = 2 
## Reading chromosome 1 ...M=400
## Reading chromosome 2 ...M=400</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(Mating)</code></pre></div>
<pre><code>##              Sire             Dam nOff herd ub.nOff
## 1 276000113913566 276000108612636    2    2      19
## 2 276000121243787 276000102372349    2    1      18
## 3 276000110948577 276000110204007    2    1      18
## 4 276000120949468 276000111909505    2    2      19
## 5 276000120949468 276000111901935    2    1      18
## 6 276000102379430 276000110948587    2    2      19</code></pre>
<p>The mean inbreeding in the offspring (which is equal to the mean kinship of the parents) is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">attributes</span>(Mating)$objval</code></pre></div>
<pre><code>## [1] 0.004837334</code></pre>
</div>
<div id="defining-the-objective-of-a-breeding-program" class="section level2">
<h2>Defining the Objective of a Breeding Program</h2>
<p>The objective of a breeding program depends on several factors. These are the intended use of the breed, the presence of historic bottlenecks, and the importance being placed on the maintenance of genetic originality. In most livestock breeds the focus is on increasing the economic merit, so the objective of the breeding program is to <em>maximize genetic gain</em>. In contrast, companion animals often suffer from historic bottlenecks due to an overuse of popular sires. Hence, in these breeds the objective is to <em>minimize inbreeding</em>. In endangered breeds, which get subsidies for conservation, the focus may be on increasing their conservation values by <em>recovering the native genetic background</em> or by <em>increasing the diversity between breeds</em>.</p>
<p>However, these are conflicting objectives: To maximize genetic gain, the animals with highest breeding values would be used for breeding, which may create a new bottleneck and contribute to inbreeding depression. Maximizing genetic gain would also favor the use of animals with high genetic contributions from commercial breeds because these animals often have the highest breeding values. But this would reduce the genetic originality of the breed. Minimizing inbreeding in the offspring favors the use of animals with high contributions from other breeds because they have low kinship with the population and it may require the use of outcross animals with breeding values below average.</p>
<p>Thus, focussing on only one aspect automatically worsens the other ones. This can be avoided by imposing constraints on the aspects that are not optimized.</p>
<p>In general, best practice is genotying all selection candidates to enable marker based evaluations. A breeding program based on marker information is more efficient than a breeding program based only on pedigree information, provided that the animals are genotyped for a sufficient number of markers. For several species, however, genotyping is still too expensive, so the breeding programs rely only on pedigree information.</p>
<p>Depending on what the objective of your breeding program is, you may continue reading at the appropriate section:</p>
<ul>
<li><a href="#marker-based-ocs">Marker-based OCS</a>
<ul>
<li>Maximize Genetic Gain</li>
<li>Minimize Inbreeding</li>
<li>Recover the Native Genetic Background</li>
<li>Increase Diversity Between Breeds</li>
</ul></li>
<li><a href="#pedigree-based-ocs">Pedigree-based OCS</a>
<ul>
<li>Maximize Genetic Gain</li>
<li>Minimize Inbreeding]</li>
<li>Recover the Native Genetic Background</li>
</ul></li>
</ul>
</div>
<div id="marker-based-ocs" class="section level2">
<h2>Marker-based OCS</h2>
<p>The required genotype file format, the marker map, the parameters <code>minSNP</code>, <code>minL</code>, <code>unitL</code>, <code>unitP</code>, and <code>ubFreq</code>, which are used for estimating the <em>segment based kinship</em>, the <em>kinships at native haplotype segments</em>, and the <em>breed composition</em>, have been described in the <a href="seg-vignette.html">companion vignette for basic marker-based evaluations</a>.</p>
<p>A matrix containing the <em>segment based kinship</em> between all pairs of individuals can be computed with function <a href="https://rdrr.io/cran/optiSel/man/segIBD.html">segIBD</a>, whereas the <em>kinships at native haplotype segments</em> can be calculated from the results of function <a href="https://rdrr.io/cran/optiSel/man/segIBDatN.html">segIBDatN</a>. Both kinships are computed below and combined into a single R-object with function <a href="https://rdrr.io/cran/optiSel/man/kinlist.html">kinlist</a>. Below, the <em>kinship at native haplotype segments</em> is named <code>sKinatN</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Kin  &lt;-<span class="st"> </span><span class="kw">kinlist</span>(
            <span class="dt">sKin    =</span> <span class="kw">segIBD</span>(GTfiles, map, <span class="dt">minSNP=</span><span class="dv">20</span>, <span class="dt">minL=</span><span class="fl">1.0</span>), 
            <span class="dt">sKinatN =</span> <span class="kw">segIBDatN</span>(GTfiles, Cattle, map, <span class="dt">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="dt">ubFreq=</span><span class="fl">0.01</span>, <span class="dt">minL=</span><span class="fl">1.0</span>)
            )</code></pre></div>
<p>The <em>breed composition</em> of individuals can be estimated with function <a href="https://rdrr.io/cran/optiSel/man/segBreedComp.html">segBreedComp</a>. The migrant contributions <code>MC</code> of the Angler cattle are added as an additional column to data frame <code>Cattle</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">wdir  &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">tempdir</span>(), <span class="st">&quot;HaplotypeEval&quot;</span>)
wfile &lt;-<span class="st"> </span><span class="kw">haplofreq</span>(GTfiles, Cattle, map, <span class="dt">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="dt">minSNP=</span><span class="dv">20</span>, <span class="dt">minL=</span><span class="fl">1.0</span>, <span class="dt">w.dir=</span>wdir)
Comp  &lt;-<span class="st"> </span><span class="kw">segBreedComp</span>(wfile$match, map)
Cattle[<span class="kw">rownames</span>(Comp), <span class="st">&quot;MC&quot;</span>] &lt;-<span class="st"> </span><span class="dv">1</span> -<span class="st"> </span>Comp$native</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(Cattle[,-<span class="dv">1</span>])</code></pre></div>
<pre><code>##                 Born  Breed         BV    Sex herd        MC
## 276000101676415 1991 Angler -1.0706066   male &lt;NA&gt; 0.4194310
## 276000108612636 1994 Angler -0.3362574 female    2 0.4312843
## 276000102372349 1986 Angler -2.0735649 female    1 0.2581902
## 276000102379430 1987 Angler  1.5968307   male &lt;NA&gt; 0.5954559
## 276000108826036 1994 Angler  1.0023969   male &lt;NA&gt; 0.7760017
## 276000111902076 1998 Angler -0.2426676   male &lt;NA&gt; 0.7541481</code></pre>
<p>There are two functions that can now be used to perform the optimization, which are <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a> and <a href="https://rdrr.io/cran/optiSel/man/opticont4mb.html">opticont4mb</a>. The latter is used especially if the aim is to decrease the average kinship in a multi-breed population. The parameters that can be constrained or optimized can be viewed with functions <a href="https://rdrr.io/cran/optiSel/man/help.opticont.html">help.opticont</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help.opticont</span>(Kin, <span class="dt">phen=</span>Cattle)</code></pre></div>
<pre><code>## Available objective functions: 
##        min.sKin  min.segIBDandN  min.sKinatN  
##        min.BV  min.MC  
##        max.BV  max.MC  
## 
## Available constraints: 
##        ub.sKin  ub.segIBDandN  ub.sKinatN  
##        lb.BV  lb.MC  
##        ub.BV  ub.MC  
##        eq.BV  eq.MC  
##        ub  lb
## 
## Attention: Probably not all these objective functions and
##            constraints make sense in animal breeding</code></pre>
<p>Compared to the introductory example the possibility to restrict or to minimize migrant contributions becomes available because column <code>MC</code> was added to data frame <code>Cattle</code>. Additionally, the possibility to minimize or to restrict the kinship at native segments <code>sKinatN</code> becomes available since this kinship was also defined in function <a href="https://rdrr.io/cran/optiSel/man/kinlist.html">kinlist</a>.</p>
<p>The options available for function <a href="https://rdrr.io/cran/optiSel/man/opticont4mb.html">opticont4mb</a> are displayed with <a href="https://rdrr.io/cran/optiSel/man/help.opticont4mb.html">help.opticont4mb</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help.opticont4mb</span>(Kin, <span class="dt">phen=</span>Cattle)</code></pre></div>
<pre><code>## Available objective functions: 
##        min.sKin.acrossBreeds  min.sKin  min.segIBDandN  min.sKinatN  
##        min.BV  min.MC  
##        max.BV  max.MC  
## 
## Available constraints: 
##        ub.sKin.acrossBreeds  ub.sKin  ub.segIBDandN  ub.sKinatN  
##        lb.BV  lb.MC  
##        ub.BV  ub.MC  
##        eq.BV  eq.MC  
##        ub  lb
## 
## Attention: Probably not all these objective functions and
##            constraints make sense in animal breeding</code></pre>
<p>Here, the additional possibility becomes available to minimize or to restrict the mean kinship <code>sKin</code> in a core set consisting of individuals from several breeds. The offspring of the selection candidates is included in the core set. This can be done by using method <code>min.sKin.acrossBreeds</code> in function <a href="https://rdrr.io/cran/optiSel/man/opticont4mb.html">opticont4m</a> or by defining the constraint <code>ub.sKin.acrossBreeds</code>.</p>
<p>For defining appropriate threshold values for the constraints, the mean kinships <code>meanKin</code> and <code>meanKinatN</code>, the mean migrant contribution <code>meanMC</code>, and the mean breeding value in the actual population need to be known. They are equal to the respective values in a hypothetical offspring population that is created with no selection at all, meaning that all selection candidates are used for breeding, and all candidates of a particular sex have equal contributions. First we define a data frame containing only the phenotypes of the selection candidates:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phen &lt;-<span class="st"> </span>Cattle[Cattle$Breed==<span class="st">&quot;Angler&quot;</span>,]</code></pre></div>
<p>The values for the actual population can be obtained as</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eqCont     &lt;-<span class="st"> </span><span class="kw">summary</span>(<span class="kw">opticont</span>(<span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span><span class="st">&quot;equal.cont&quot;</span>))
meanKin    &lt;-<span class="st"> </span>eqCont$sKin
meanKinatN &lt;-<span class="st"> </span>eqCont$sKinatN
meanMC     &lt;-<span class="st"> </span>eqCont$meanMC
phen$BV    &lt;-<span class="st"> </span>phen$BV -<span class="st"> </span>eqCont$meanBV
meanBV     &lt;-<span class="st"> </span><span class="dv">0</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eqCont[,<span class="kw">c</span>(<span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>, <span class="st">&quot;meanMC&quot;</span>)]</code></pre></div>
<pre><code>##        sKin    sKinatN    meanMC
##  0.05689674 0.06684541 0.6149141</code></pre>
<p>The breeding values are standardized to have mean 0 in the actual population.</p>
<p>Depending on what the objective of your breeding program is, you may continue reading at the appropriate section:</p>
<ul>
<li><a href="#maximize-genetic-gain-marker-based">Maximize Genetic Gain</a></li>
<li><a href="#minimize-inbreeding-marker-based">Minimize Inbreeding</a></li>
<li><a href="#recover-the-native-genetic-background-marker-based">Recover the Native Genetic Background</a></li>
<li><a href="#increase-diversity-between-breeds">Increase Diversity Between Breeds</a></li>
</ul>
<div id="maximize-genetic-gain" class="section level3">
<h3><a name="maximize-genetic-gain-marker-based"></a>Maximize Genetic Gain</a></h3>
<p>First we create a list of constraints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con         &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.sKin &lt;-<span class="st"> </span>meanKin +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))</code></pre></div>
<p>Again, equal contributions are assumed for the females and only the contributions of males are to be optimized. The upper bound for the mean segment based kinship was derived from the effective population size as explained above. Now the optimum contributions of the selection candidates can be calculated:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;max.BV&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
maxBV.s &lt;-<span class="st"> </span><span class="kw">summary</span>(maxBV)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##       valid   meanBV    meanMC       sKin    sKinatN
## maxBV  TRUE 1.000488 0.6399194 0.06161228 0.08165922</code></pre>
<p>The results are the same as in the introductory example (as expected). This approach may be apppropriate for a population without introgression, but for populations with historic introgression, the kinship at native alleles should be restricted as well in accordance with the desired effective size, and the migrant contributions should be restricted in order not to increase. Otherwise the genetic originality of the breed may get lost in the long term.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$ub.sKinatN &lt;-<span class="st"> </span>meanKinatN +(<span class="dv">1</span>-meanKinatN)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
con$ub.MC      &lt;-<span class="st"> </span>meanMC
maxBV2         &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;max.BV&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;slsqp&quot;</span>)
maxBV2.s       &lt;-<span class="st"> </span><span class="kw">summary</span>(maxBV2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(maxBV.s, maxBV2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid    meanBV    meanMC       sKin    sKinatN
## maxBV   TRUE 1.0004880 0.6399194 0.06161228 0.08165922
## maxBV2  TRUE 0.8813497 0.6149141 0.05912416 0.07151119</code></pre>
<p>Since migrant contributions and breeding values are positively correlated, the genetic gain decreases slightly when migrant contributions are constrained not to increase.</p>
</div>
<div id="minimize-inbreeding" class="section level3">
<h3><a name="minimize-inbreeding-marker-based"></a>Minimize Inbreeding</a></h3>
<p>Minimizing inbreeding means to minimize the average kinship of the population in order to enable breeders to avoid inbreeding. This is the appropriate approach e.g.Â for companion animals suffering from a historic bottleneck. It can be done with or without accounting for breeding values. In the example below no breeding values are considered since accurate breeding values are not available for most of these breeds.</p>
<p>First we create a list of constraints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con  &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))</code></pre></div>
<p>Again, equal contributions are assumed for the females and only the contributions of males are to be optimized. The segment based kinship is not constrained in this example because it should be minimized.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.sKin&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minKin.s &lt;-<span class="st"> </span><span class="kw">summary</span>(minKin)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid    meanBV    meanMC       sKin    sKinatN
## minKin  TRUE -0.371969 0.5445139 0.04216327 0.05524461</code></pre>
<p>Minimizing kinship without constraining the mean breeding value decreases the mean breeding value in the offspring slightly because the individuals with high breeding values are related. For this breed, it also decreases the migrant contribution because individuals from other breeds were related.</p>
<p>While in livestock breeds the migrant contributions should be restricted in order to maintain the genetic originality of the breeds, in several companion breeds the opposite is true. Several companion breeds have high inbreeding coefficients and descend from only very few (e.g.Â 3) founders <span class="citation">(Wellmann and Pfeiffer 2009)</span>, and purging seems to be not feasible. Hence, a sufficient genetic diversity of the population cannot be achieved in the population even if marker data is used to minimize inbreeding. For these breeds it may be appropriate to use unrelated individuals from a variety of other breeds in order to increase the genetic diversity. However, only a small contribution from other breeds is needed, so the migrant contributions should be restricted also for these breeds in order to preserve their genetic originality. Hence, the difference between a breed with high diversity and a breed with low diversity suffering from inbreeding depression is, that the optimum value for the migrant contribution is larger than 0 for the latter.</p>
<p>For such a breed it is advisable to allow the use of individuals from other breeds but to restrict the admissible mean contribution from other breeds in the population. The mean kinship at native alleles should be restricted as well to require only a small amount of introgression:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$ub.MC      &lt;-<span class="st"> </span><span class="fl">0.95</span>*meanMC
con$ub.sKinatN &lt;-<span class="st"> </span>meanKinatN +(<span class="dv">1</span>-meanKinatN)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
minKin2        &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.sKin&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;slsqp&quot;</span>, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minKin2.s      &lt;-<span class="st"> </span><span class="kw">summary</span>(minKin2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(minKin.s, minKin2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##         valid     meanBV    meanMC       sKin    sKinatN
## minKin   TRUE -0.3719690 0.5445139 0.04216327 0.05524461
## minKin2  TRUE -0.3719214 0.5444554 0.04216174 0.05523923</code></pre>
</div>
<div id="recover-the-native-genetic-background" class="section level3">
<h3><a name="recover-the-native-genetic-background-marker-based"></a>Recover the Native Genetic Background</a></h3>
<p>For endangered breeds the priority of a breeding program could be to recover the original genetic background by minimizing migrant contributions. However, since the individuals with smallest migrant contributions are related, this may considerably increase the inbreeding coefficients if the diversity at native alleles is not preserved. Hence, constraints are defined below not only for the segment based kinship but also for the kinship at native segments in accordance with the desired effective size:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phen           &lt;-<span class="st"> </span>Cattle[Cattle$Breed==<span class="st">&quot;Angler&quot;</span>,]
con            &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.sKin    &lt;-<span class="st"> </span>meanKin    +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
con$ub.sKinatN &lt;-<span class="st"> </span>meanKinatN +<span class="st"> </span>(<span class="dv">1</span>-meanKinatN)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
minMC   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.MC&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;slsqp&quot;</span>)
minMC.s &lt;-<span class="st"> </span><span class="kw">summary</span>(minMC)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minMC.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##       valid     meanBV   meanMC       sKin    sKinatN
## minMC  TRUE -0.5564253 0.480897 0.04937383 0.07151119</code></pre>
<p>For this breed, minimizing migrant contributions results in negative genetic gain because migrant contributions and breeding values are positively correlated. This can be avoided by adding an additional constraint for the breeding values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$lb.BV &lt;-<span class="st"> </span><span class="dv">0</span>
minMC2    &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.MC&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;cccp&quot;</span>, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minMC2.s  &lt;-<span class="st"> </span><span class="kw">summary</span>(minMC2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(minMC.s, minMC2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid        meanBV    meanMC       sKin    sKinatN
## minMC   TRUE -5.564253e-01 0.4808970 0.04937383 0.07151119
## minMC2  TRUE  6.115291e-07 0.4983187 0.04934570 0.07132758</code></pre>
</div>
<div id="increase-diversity-between-breeds" class="section level3">
<h3>Increase Diversity Between Breeds</h3>
<p>While removing introgressed genetic material from the population is one possibility to increase the conservation value of an endangered breed, an alternative approach is to increase the genetic distance between the endangered breed and commercial breeds. In this case we do not care about whether alleles are native or not. We just want to accumulate haplotype segments which are rare in commercial breeds. This can be done with a core set approach.</p>
<p>In the core set approach, a hypothetical subdivided population is considered, consisting of individuals from various breeds. This population is called the core set. The individuals from the breed of interest are the selection candidates. The contributions of each breed to the core set are such that the genetic diversity of the core set is maximized. The parameter to be minimized is the mean kinship of individuals from the core set in the next generation. Thereby, it is assumed that the contributions of the selection candidates from the breed of interest are optimized, whereas individuals from all other breeds have equal contributions.</p>
<p>If the contributions of the selection candidates minimize the mean kinship in the core set, then they maximize the genetic diversity of the core set. This could be achieved by increasing the gentic diversity within the breed or by increasing the genetic distance between the breed of interest and the other breeds.</p>
<p>First, the contributions of the breeds to the core set are calculated with function <a href="https://rdrr.io/cran/optiSel/man/opticomp.html">opticomp</a>. A lower bound is specified for the breed of interest:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreSet &lt;-<span class="st"> </span><span class="kw">opticomp</span>(Kin$sKin, Cattle$Breed, <span class="dt">lb=</span><span class="kw">c</span>(<span class="dt">Angler=</span><span class="fl">0.1</span>))
CoreSet$bc</code></pre></div>
<pre><code>##     Angler  Fleckvieh   Holstein    Rotbunt 
## 0.47908531 0.42425265 0.04434691 0.05231513</code></pre>
<p>Second, the constraints are defined:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con         &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.sKin &lt;-<span class="st"> </span>meanKin  +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))</code></pre></div>
<p>Equal contributions are assumed for the females and only the contributions of males are to be optimized. The upper bound for the mean segment based kindship was derived from the effective population size as explained above. Now the optimum contributions of the selection candidates can be calculated with fumction <a href="https://rdrr.io/cran/optiSel/man/opticont4mb.html">opticont4mb</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin4mb &lt;-<span class="st"> </span><span class="kw">opticont4mb</span>(<span class="st">&quot;min.sKin.acrossBreeds&quot;</span>, Kin, <span class="dt">phen=</span>Cattle, CoreSet$bc, 
                       <span class="dt">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minKin4mb.s &lt;-<span class="st"> </span><span class="kw">summary</span>(minKin4mb)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin4mb.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;sKin&quot;</span>, <span class="st">&quot;sKinatN&quot;</span>, <span class="st">&quot;sKin.acrossBreeds&quot;</span>)]</code></pre></div>
<pre><code>##           valid     meanBV    meanMC       sKin    sKinatN
## minKin4mb  TRUE -0.2754909 0.5364332 0.04294231 0.05805911
##           sKin.acrossBreeds
## minKin4mb        0.03014493</code></pre>
<p>Note that parameter <code>phen</code> of function <a href="https://rdrr.io/cran/optiSel/man/opticont4mb.html">opticont4mb</a> is a data frame containing individuals from all genotyped breeds. In contrast, parameter <code>phen</code> of function <a href="https://rdrr.io/cran/optiSel/man/opticont.html">opticont</a> contains only the selection candidates.</p>
</div>
</div>
<div id="pedigree-based-ocs" class="section level2">
<h2>Pedigree-based OCS</h2>
<p>All evaluations using pedigree data are demonstrated at the example of the Hinterwald cattle. A pedigree is contained in the package. The pedigree and the functions dealing with pedigree data have already been described in the <a href="ped-vignette.html">companion vignette for basic pedigree-based evaluations</a>.</p>
<p>The pedigree completeness is an important factor to get reliable results. If an animal has many missing ancestors, then it would falsely considered to be unrelated to other animals, so it will falsely obtain high optimum contributions. There are several approaches to overcome this problem:</p>
<ul>
<li><p>Calculate the pedigree completeness for all selection candidates and exclude individuals with a small number of equivalent complete generations from the evaluations. The number of equivalent complete generations can be computed with function <a href="https://rdrr.io/cran/optiSel/man/summary.Pedig.html">summary</a>.</p></li>
<li><p>Classify the breed of founders born after some fixed date as <code>unknown</code>, and restrict the genetic contribution from these founders in the offspring. The breed names can be classified by using appropriate values for parameters <code>lastNative</code> and <code>thisBreed</code> in function <a href="https://rdrr.io/cran/optiSel/man/prePed.html">prePed</a>.</p></li>
<li><p>Classify the breed of founders born after some fixed date as <code>unknown</code>, so that these founders are considered non-native, and restrict or minimize the kinship at native alleles (in addition to the classical pedigree based kinship).</p></li>
</ul>
<p>Of course, all 3 approaches can be followed simultaneously. First, we prepare the pedigree and classify the breed of founders born after 1970 to be <code>unknown</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;PedigWithErrors&quot;</span>)
Pedig &lt;-<span class="st"> </span><span class="kw">prePed</span>(PedigWithErrors, <span class="dt">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="dt">lastNative=</span><span class="dv">1970</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(Pedig)</code></pre></div>
<pre><code>##                             Indiv Sire  Dam  Sex   Breed Born
## S276000890888469 S276000890888469 &lt;NA&gt; &lt;NA&gt; male unknown   NA
## S276000891160479 S276000891160479 &lt;NA&gt; &lt;NA&gt; male unknown   NA
## S276000811209491 S276000811209491 &lt;NA&gt; &lt;NA&gt; male unknown   NA
## S276000811063904 S276000811063904 &lt;NA&gt; &lt;NA&gt; male unknown   NA
## S276000892147965 S276000892147965 &lt;NA&gt; &lt;NA&gt; male unknown   NA
## S276000891895524 S276000891895524 &lt;NA&gt; &lt;NA&gt; male unknown   NA</code></pre>
<p>We define the individuals included in data.frame <code>Phen</code> to be selection candidates if their number of equivalent complete generations is at least 5.0. These are the individuals contained in vector <code>keep</code>. In a real application the individuals would be used as selection candidates that could become parents of the forthcomming birth cohorts.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;Phen&quot;</span>)
Summary &lt;-<span class="st"> </span><span class="kw">summary</span>(Pedig, <span class="dt">keep=</span>Pedig$Indiv %in%<span class="st"> </span>Phen$Indiv)
keep    &lt;-<span class="st"> </span>Summary[Summary$equiGen&gt;=<span class="fl">3.0</span>, <span class="st">&quot;Indiv&quot;</span>]
<span class="kw">table</span>(Pedig[keep, <span class="st">&quot;Sex&quot;</span>])</code></pre></div>
<pre><code>## 
## female   male 
##     47     98</code></pre>
<p>A matrix containing the <em>pedigree based kinship</em> between all pairs of individuals can be computed with function <a href="https://rdrr.io/cran/optiSel/man/pedIBD.html">pedIBD</a>. It is half the additive relationship matrix. The <em>pedigree based kinship at native alleles</em> can be calculated from the results of function <a href="https://rdrr.io/cran/optiSel/man/pedIBDatN.html">pedIBDatN</a>. Both kinships are computed below and combined into a single R-object with function <a href="https://rdrr.io/cran/optiSel/man/kinlist.html">kinlist</a>. Below, the <em>pedigree based kinship</em> is named <code>pKin</code>, and the <em>kinship at native alleles</em> is named <code>pKinatN</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Kin &lt;-<span class="st"> </span><span class="kw">kinlist</span>(
    <span class="dt">pKin    =</span> <span class="kw">pedIBD</span>(Pedig, <span class="dt">keep.only=</span>keep),
    <span class="dt">pKinatN =</span> <span class="kw">pedIBDatN</span>(Pedig, <span class="dt">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="dt">keep.only=</span>keep)
)</code></pre></div>
<pre><code>## Number of Migrant Founders: 159
## Number of Native  Founders: 150
## Individuals in Pedigree   : 1521
## Mean kinship at native alleles:  0.078</code></pre>
<p>The <em>breed composition</em> of individuals can be estimated with function <a href="https://rdrr.io/cran/optiSel/man/pedBreedComp.html">pedBreedComp</a>. The migrant contributions <code>MC</code> of the Angler cattle are added as an additional column to the pedigree.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cont     &lt;-<span class="st"> </span><span class="kw">pedBreedComp</span>(Pedig, <span class="dt">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>)
Pedig$MC &lt;-<span class="st"> </span><span class="dv">1</span>-cont$native
<span class="kw">head</span>(cont[keep, <span class="dv">2</span>:<span class="dv">6</span>])</code></pre></div>
<pre><code>##                    native  unknown      unbek0  Fleckvieh Vorderwaelder
## 276000812496744 0.7360840 0.062500 0.024658203 0.10644531    0.07031250
## 276000891862786 0.6164551 0.187500 0.022705078 0.09130859    0.08203125
## 276000812202159 0.4367676 0.265625 0.014160156 0.08032227    0.20312500
## 276000812749837 0.5534668 0.312500 0.013427734 0.05810547    0.06250000
## 276000891618444 0.2949219 0.515625 0.003173828 0.04565430    0.14062500
## 276000812922523 0.3649902 0.437500 0.022460938 0.06567383    0.10937500</code></pre>
<p>Below, the columns from the pedigree are added to data frame <code>Phen</code>. Hence, it contains the individual IDs in Colmumn 1 (<code>Indiv</code>), sexes in Column 2 (<code>Sex</code>), breed names (<code>Breed</code>), years of birth (<code>Born</code>), breeding values (<code>BV</code>), and the migrant contributions (<code>MC</code>) of the selection candidates. The breeding values are simulated such that breeding values and migrant contributions are positively correlated. This mimics historic introgression from a high-yielding commercial breed:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Phen &lt;-<span class="st"> </span><span class="kw">merge</span>(Pedig, Phen[,<span class="kw">c</span>(<span class="st">&quot;Indiv&quot;</span>, <span class="st">&quot;BV&quot;</span>)], <span class="dt">by=</span><span class="st">&quot;Indiv&quot;</span>)
Phen &lt;-<span class="st"> </span>Phen[Phen$Indiv %in%<span class="st"> </span>keep, <span class="kw">c</span>(<span class="st">&quot;Indiv&quot;</span>, <span class="st">&quot;Sex&quot;</span>,<span class="st">&quot;Breed&quot;</span>, <span class="st">&quot;Born&quot;</span>, <span class="st">&quot;BV&quot;</span>, <span class="st">&quot;MC&quot;</span>)]
Phen$BV &lt;-<span class="st"> </span>Phen$BV -<span class="st"> </span><span class="kw">mean</span>(Phen$BV)
<span class="kw">head</span>(Phen)</code></pre></div>
<pre><code>##             Indiv    Sex         Breed Born          BV        MC
## 1 276000810958027   male Hinterwaelder 2003  1.66612826 0.6257324
## 2 276000811082118   male Hinterwaelder 2003 -1.35282365 0.5294189
## 3 276000811287745 female Hinterwaelder 2001 -1.23463901 0.4195862
## 4 276000811597070   male Hinterwaelder 2001  1.06713987 0.6991882
## 5 276000811597110   male Hinterwaelder 2003  0.08216756 0.4387512
## 6 276000811597136 female Hinterwaelder 2002  1.63222142 0.7654114</code></pre>
<p>The parameters that can be constrained or optimized can be viewed with function <a href="https://rdrr.io/cran/optiSel/man/help.opticont.html">help.opticont</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help.opticont</span>(Kin, Phen)</code></pre></div>
<pre><code>## Available objective functions: 
##        min.pKin  min.pedIBDandN  min.pKinatN  
##        min.BV  min.MC  
##        max.BV  max.MC  
## 
## Available constraints: 
##        ub.pKin  ub.pedIBDandN  ub.pKinatN  
##        lb.BV  lb.MC  
##        ub.BV  ub.MC  
##        eq.BV  eq.MC  
##        ub  lb
## 
## Attention: Probably not all these objective functions and
##            constraints make sense in animal breeding</code></pre>
<p>Compared to the introductory example the possibility to restrict or to minimize migrant contributions becomes available because column <code>MC</code> is now included in data frame <code>Phen</code>. Additionally, there is the possibility to minimize or to restrict the <em>kinship at native alleles</em> <code>pKinatN</code> and the <em>pedigree based kinship</em> <code>pKin</code>.</p>
<p>For defining appropriate threshold values for the constraints, the mean kinships <code>meanKin</code> and <code>meanKinatN</code>, the mean migrant contribution <code>meanMC</code>, and the mean breeding value in the actual population need to be known. They are equal to the respective values in a hypothetical offspring population that is created with no selection at all, meaning that all selection candidates are used for breeding, and all candidates of a particular sex have equal contributions. Thus, the values for the actual population can be obtained as</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eqCont     &lt;-<span class="st"> </span><span class="kw">summary</span>(<span class="kw">opticont</span>(<span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span><span class="st">&quot;equal.cont&quot;</span>))
meanKin    &lt;-<span class="st"> </span>eqCont$pKin
meanKinatN &lt;-<span class="st"> </span>eqCont$pKinatN
meanMC     &lt;-<span class="st"> </span>eqCont$meanMC
Phen$BV    &lt;-<span class="st"> </span>Phen$BV -<span class="st"> </span>eqCont$meanBV
meanBV     &lt;-<span class="st"> </span><span class="dv">0</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eqCont[,<span class="kw">c</span>(<span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>, <span class="st">&quot;meanMC&quot;</span>)]</code></pre></div>
<pre><code>##        pKin    pKinatN    meanMC
##  0.03456986 0.07689685 0.4707445</code></pre>
<p>Depending on what the objective of your breeding program is, you may continue reading at the appropriate section:</p>
<ul>
<li><a href="#maximize-genetic-gain-pedigree-based">Maximize Genetic Gain</a></li>
<li><a href="#minimize-inbreeding-pedigree-based">Minimize Inbreeding</a></li>
<li><a href="#recover-the-native-genetic-background-pedigree-based">Recover the Native Genetic Background</a></li>
</ul>
<div id="maximize-genetic-gain-1" class="section level3">
<h3><a name="maximize-genetic-gain-pedigree-based"></a>Maximize Genetic Gain</a></h3>
<p>This is the traditional approach proposed by <span class="citation">T. H. E. Meuwissen (1997)</span>. First we create a list of constraints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con         &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.pKin &lt;-<span class="st"> </span>meanKin +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))</code></pre></div>
<p>Here, equal contributions are assumed for the females and only the contributions of males are to be optimized. The upper bound for the mean segment based kinship was derived from the effective population size as explained above. Now the optimum contributions of the selection candidates can be calculated:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;max.BV&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
maxBV.s &lt;-<span class="st"> </span><span class="kw">summary</span>(maxBV)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">maxBV.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##       valid    meanBV    meanMC       pKin   pKinatN
## maxBV  TRUE 0.8069863 0.5409889 0.03939717 0.0966208</code></pre>
<p>This approach may be apppropriate for a population without introgression and complete pedigrees, but for populations with historic introgression, the kinship at native alleles should be restricted as well in accordance with the desired effective size, and the migrant contributions should be restricted in order not to increase. Otherwise the genetic originality of the breed may get lost in the long term.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$ub.pKinatN &lt;-<span class="st"> </span>meanKinatN +(<span class="dv">1</span>-meanKinatN)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
con$ub.MC      &lt;-<span class="st"> </span>meanMC
maxBV2         &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;max.BV&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;slsqp&quot;</span>)
maxBV2.s       &lt;-<span class="st"> </span><span class="kw">summary</span>(maxBV2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(maxBV.s, maxBV2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid    meanBV    meanMC       pKin    pKinatN
## maxBV   TRUE 0.8069863 0.5409889 0.03939717 0.09662080
## maxBV2  TRUE 0.6308613 0.4707445 0.03798457 0.08151239</code></pre>
<p>Thus, genetic gain in Method 2 is only slightly below the genetic gain in Method 1, but the migrant contributions do not increase and the kinship at native alleles increases at a lower rate.</p>
</div>
<div id="minimize-inbreeding-1" class="section level3">
<h3><a name="minimize-inbreeding-pedigree-based"></a>Minimize Inbreeding</a></h3>
<p>Minimizing inbreeding means to minimize the average kinship of the population in order to enable breeders to avoid inbreeding. This is the appropriate approach e.g.Â for companion animals suffering from a historic bottleneck. It can be done with or without accounting for breeding values. In the example below no breeding values are considered since accurate breeding values are not available for most of these breeds.</p>
<p>First we create a list of constraints:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con  &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))</code></pre></div>
<p>Again, equal contributions are assumed for the females and only the contributions of males are to be optimized. The segment based kinship is not constrained in this example because it should be minimized.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.pKin&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minKin.s &lt;-<span class="st"> </span><span class="kw">summary</span>(minKin)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minKin.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid    meanBV    meanMC       pKin    pKinatN
## minKin  TRUE 0.2612448 0.5294628 0.02695391 0.07148733</code></pre>
<p>The approach shown above has the disadvantage that kinships between individuals are less reliable if ancestors are missing in the pedigree. The alternative approach, shown below, is to minimize the kinship at native alleles and to restrict pedigree based kinship.</p>
<p>While in livestock breeds the migrant contributions should be diminished in order to maintain the genetic originality of the breeds, in several companion breeds the opposite is true. Several companion breeds have high inbreeding coefficients and descend from only very few (e.g.Â 3) founders. Hence, a sufficient genetic diversity cannot be achieved in the population. For these breeds it may be appropriate to use unrelated individuals from a variety of other breeds in order to increase the genetic diversity. However, only a small contribution from other breeds is needed, so the migrant contributions should be restricted also for these breeds in order to preserve their genetic originality. The difference between a breed with high diversity and a breed with low diversity suffering from inbreeding depression is, that the optimum value for the migrant contribution is larger than 0 for the latter. For such a breed it is advisable to allow the use of individuals from other breeds but to restrict the admissible mean contribution from other breeds.</p>
<p>In summary, the alternative approach is to minimize the kinship at native alleles and to restrict pedigree based kinship and migrant contributions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con  &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.MC   &lt;-<span class="st"> </span><span class="fl">0.95</span>*meanMC
con$ub.pKin &lt;-<span class="st"> </span>meanKin +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))

minKin2     &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.pKinatN&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">solver=</span><span class="st">&quot;slsqp&quot;</span>)
minKin2.s   &lt;-<span class="st"> </span><span class="kw">summary</span>(minKin2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(minKin.s, minKin2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##         valid      meanBV    meanMC       pKin    pKinatN
## minKin   TRUE  0.26124483 0.5294628 0.02695391 0.07148733
## minKin2  TRUE -0.02294647 0.4472072 0.03195152 0.06735677</code></pre>
<p>The pedigree based kinship is slightly higher in the second approach, but the kinship at native alleles is lower. Since pedigree based kinships are less reliable due to missing ancestors in the pedigree, the second approach is recommended. However, the use of pedigree data has the disadvantage that only the expected kinships can be minimized. The expected kinships deviate from the realized kinships due to mendelian segregation. Hence, for breeds with serious inbreeding problems it is recommended to genotype the selection candidates and to perform marker-based optimum contribution selection.</p>
</div>
<div id="recover-the-native-genetic-background-1" class="section level3">
<h3><a name="recover-the-native-genetic-background-pedigree-based"></a>Recover the Native Genetic Background</a></h3>
<p>For endangered breeds the priority of a breeding program could be to recover the original genetic background by minimizing migrant contributions. However, since the individuals with smallest migrant contributions are related, this may considerably increase the inbreeding coefficients if the diversity at native alleles is not preserved. Hence, constraints are defined below not only for the pedigree based kinship, but also for the kinship at native alleles in accordance with the desired effective size:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con            &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">ub=</span><span class="kw">c</span>(<span class="dt">M=</span><span class="ot">NA</span>, <span class="dt">F=</span>-<span class="dv">1</span>))
con$ub.pKin    &lt;-<span class="st"> </span>meanKin    +<span class="st"> </span>(<span class="dv">1</span>-meanKin)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
con$ub.pKinatN &lt;-<span class="st"> </span>meanKinatN +<span class="st"> </span>(<span class="dv">1</span>-meanKinatN)*(<span class="dv">1</span>/(<span class="dv">2</span>*Ne))
minMC   &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.MC&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minMC.s &lt;-<span class="st"> </span><span class="kw">summary</span>(minMC)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">minMC.s[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##       valid     meanBV    meanMC       pKin    pKinatN
## minMC  TRUE -0.2395387 0.4045795 0.03939921 0.07657431</code></pre>
<p>For some breeds, migrant contributions and breeding values are positively correlated, so minimizing migrant contributions results in negative genetic. This can be avoided by adding an additional constraint for the breeding values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">con$lb.BV &lt;-<span class="st"> </span><span class="dv">0</span>
minMC2    &lt;-<span class="st"> </span><span class="kw">opticont</span>(<span class="dt">method=</span><span class="st">&quot;min.MC&quot;</span>, <span class="dt">K=</span>Kin, <span class="dt">phen=</span>Phen, <span class="dt">con=</span>con, <span class="dt">trace=</span><span class="ot">FALSE</span>)
minMC2.s  &lt;-<span class="st"> </span><span class="kw">summary</span>(minMC2)</code></pre></div>
<p>For comparison, the summaries of both scenarios are combined into a single data frame with <code>rbind</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Results &lt;-<span class="st"> </span><span class="kw">rbind</span>(minMC.s, minMC2.s)
Results[,<span class="kw">c</span>(<span class="st">&quot;valid&quot;</span>, <span class="st">&quot;meanBV&quot;</span>, <span class="st">&quot;meanMC&quot;</span>, <span class="st">&quot;pKin&quot;</span>, <span class="st">&quot;pKinatN&quot;</span>)]</code></pre></div>
<pre><code>##        valid        meanBV    meanMC       pKin    pKinatN
## minMC   TRUE -2.395387e-01 0.4045795 0.03939921 0.07657431
## minMC2  TRUE  4.956989e-06 0.4083974 0.03939950 0.07731175</code></pre>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-Meuwissen2009">
<p>Meuwissen, T H E. 2009. âGenetic Management of Small Populations: A Review.â <em>Acta Agriculturae Scand Section A</em> 59: 71â79.</p>
</div>
<div id="ref-Meuwissen1997">
<p>Meuwissen, T. H. E. 1997. âMaximising the Response of Selection with a Predefined Rate of Inbreeding.â <em>J. Animal Sci</em> 75: 934â40.</p>
</div>
<div id="ref-Wellmann2009">
<p>Wellmann, R., and I. Pfeiffer. 2009. âPedigree Analysis for Conservation of Genetic Diversity and Purging.â <em>Genetical Research (Camb)</em> 91 (34).</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
