<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Robin Wellmann" />

<meta name="date" content="2024-06-12" />

<title>1. Pedigree-based Evaluations</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">

div.csl-bib-body { }
div.csl-entry {
clear: both;
margin-bottom: 0em;
}
.hanging div.csl-entry {
margin-left:2em;
text-indent:-2em;
}
div.csl-left-margin {
min-width:2em;
float:left;
}
div.csl-right-inline {
margin-left:2em;
padding-left:1em;
}
div.csl-indent {
margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">1. Pedigree-based Evaluations</h1>
<h4 class="author">Robin Wellmann</h4>
<h4 class="date">2024-06-12</h4>



<p>Animal breeding relies on the prediction of breeding values of
selection candidates and subsequent selection of animals for breeding in
a way that restricts the rate of inbreeeding and maintains genetic
originality of the breed. The methods of choice are the utilization of
pedigree data, of genetic marker data, or a combination of both. This
section covers the following evaluations based on pedigree data:</p>
<ul>
<li><a href="#data-preparation">Data Preparation</a>
<ul>
<li>Pedigrees (includes plotting)</li>
</ul></li>
<li><a href="#quality-control">Quality Control</a>
<ul>
<li>Pedigree Completeness</li>
<li>Number of Equivalent Complete Generations</li>
<li>Number of Fully Traced Generations</li>
<li>Number of Maximum Generations Traced</li>
<li>Index of Pedigree Completeness</li>
</ul></li>
<li><a href="#individual-specific-parameters">Individual Specific
Parameters</a>
<ul>
<li><a href="#inbreeding-coefficients">Inbreeding Coefficients</a></li>
<li><a href="#breed-composition">Breed Composition</a></li>
<li><a href="#kinship">Kinship</a></li>
<li><a href="#kinship-at-native-alleles">Kinship at Native
Alleles</a></li>
</ul></li>
<li><a href="#population-specific-parameters">Population Specific
Parameters</a>
<ul>
<li><a href="#genetic-diversity">Genetic Diversity</a></li>
<li><a href="#kinship-and-diversity-at-native-alleles">Kinship and
Diversity at Native Alleles</a></li>
<li><a href="#native-effective-size">Native Effective Size</a></li>
<li><a href="#effective-size">Effective Size</a></li>
<li><a href="#change-of-native-genome-equivalent-and-ne">Change of
Native Genome Equivalent and Ne</a></li>
<li><a href="#change-of-breed-composition">Change of Breed
Compostion</a></li>
</ul></li>
</ul>
<p>Note that the calculation of breeding values is not included as there
already exist a couple of R packages for this purpose. Recommended are
the free package <a href="https://www.jstatsoft.org/htaccess.php?volume=033&amp;type=i&amp;issue=02&amp;filename=paper">MCMCglmm</a>
for small and medium sized data sets, and package <a href="https://asreml.kb.vsni.co.uk/wp-content/uploads/sites/3/ASReml-R-Reference-Manual-4.2.pdf">asreml</a>
for large data sets.</p>
<div id="data-preparation" class="section level2">
<h2>Data Preparation</h2>
<p>A pedigree is a data frame or data table with the first three columns
being the individual ID, the sire ID, and the dam ID. Depending on the
intended evaluation, the pedigree may also provide the names of the
breeds (column <code>Breed</code>), the years of birth or generation
numbers (column <code>Born</code> ), and the sexes (column
<code>Sex</code>). A toy example of a pedigree may look like this:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>Pedigree <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="at">Indiv=</span> <span class="fu">c</span>(<span class="st">&quot;Iffes&quot;</span>,     <span class="st">&quot;Peter&quot;</span>,    <span class="st">&quot;Anna-Lena&quot;</span>, <span class="st">&quot;Kevin&quot;</span>,    <span class="st">&quot;Horst&quot;</span>),</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>  <span class="at">Sire =</span> <span class="fu">c</span>(<span class="st">&quot;Kevin&quot;</span>,     <span class="st">&quot;Kevin&quot;</span>,     <span class="cn">NA</span>,          <span class="dv">0</span>,         <span class="st">&quot;Horst&quot;</span>),</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  <span class="at">Dam  =</span> <span class="fu">c</span>(<span class="st">&quot;Chantalle&quot;</span>, <span class="st">&quot;Angelika&quot;</span>, <span class="st">&quot;Chantalle&quot;</span>, <span class="st">&quot;&quot;</span>,           <span class="cn">NA</span>),</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="at">Breed=</span> <span class="fu">c</span>(<span class="st">&quot;Angler&quot;</span>,    <span class="st">&quot;Angler&quot;</span>,   <span class="st">&quot;Angler&quot;</span>,    <span class="st">&quot;Holstein&quot;</span>, <span class="st">&quot;Angler&quot;</span>),</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  <span class="at">Born =</span> <span class="fu">c</span>(<span class="dv">2015</span>,        <span class="dv">2016</span>,       <span class="dv">2011</span>,       <span class="dv">2010</span>,        <span class="dv">2015</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  )</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>Pedigree</span></code></pre></div>
<pre><code>##       Indiv  Sire       Dam    Breed Born
## 1     Iffes Kevin Chantalle   Angler 2015
## 2     Peter Kevin  Angelika   Angler 2016
## 3 Anna-Lena  &lt;NA&gt; Chantalle   Angler 2011
## 4     Kevin     0           Holstein 2010
## 5     Horst Horst      &lt;NA&gt;   Angler 2015</code></pre>
<p>Individual IDs must be unique and should not contain blank spaces.
The above pedigree contains a loop, so it is not a valid pedigree (Horst
is an ancestor of himself). Function <a href="https://rdrr.io/cran/optiSel/man/prePed.html">prePed</a> will
prepare the pedigree and turn it into a valid one. The function</p>
<ul>
<li>renames the first three columns as <code>Indiv</code>,
<code>Sire</code>, <code>Dam</code> (if needed),</li>
<li>replaces 0, ““, and” ” in columns <code>Sire</code> and
<code>Dam</code> by <code>NA</code>,</li>
<li>breaks loops and reports which animal was included in a loop,</li>
<li>adds or updates column <code>Sex</code>. Sexes will be denoted as
<code>male</code> and <code>female</code>.</li>
<li>adds one row for each founder if the row is not already
included,</li>
<li>sorts the pedigree such that individuals appear in the pedigree
before they appear as parent,</li>
<li>guesses the breed name of animals with missing breeds (including
added founders),</li>
<li>adds columns <code>numIndiv</code>, <code>numSire</code>, and
<code>numDam</code> with numeric IDs of the individuals (if
<code>addNum=TRUE</code> is specified),</li>
<li>sets the breed name of founders from <code>thisBreed</code> that are
born after <code>lastNative</code> to unknown (if these parameters are
specified).</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;optiSel&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>Pedig <span class="ot">&lt;-</span> <span class="fu">prePed</span>(Pedigree)</span></code></pre></div>
<pre><code>## Pedigree loops were detected. We recommend to correct them manually before
## using prePed(). The parents of the following individuals are set to unknown
## to remove the loops.
##        Sire  Dam
## Horst Horst &lt;NA&gt;</code></pre>
<p>The result is:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>Pedig[,<span class="sc">-</span><span class="dv">1</span>]</span></code></pre></div>
<pre><code>##            Sire       Dam    Sex          Breed Born   I Offspring
## Chantalle  &lt;NA&gt;      &lt;NA&gt; female         Angler   NA  NA      TRUE
## Angelika   &lt;NA&gt;      &lt;NA&gt; female         Angler   NA  NA      TRUE
## Kevin      &lt;NA&gt;      &lt;NA&gt;   male       Holstein 2010 NaN      TRUE
## Horst      &lt;NA&gt;      &lt;NA&gt;   &lt;NA&gt; Pedigree Error 2015 NaN     FALSE
## Anna-Lena  &lt;NA&gt; Chantalle   &lt;NA&gt;         Angler 2011 NaN     FALSE
## Iffes     Kevin Chantalle   &lt;NA&gt;         Angler 2015   5     FALSE
## Peter     Kevin  Angelika   &lt;NA&gt;         Angler 2016   6     FALSE</code></pre>
<p>Plots can be created to show specified individuals and their
relatives. In the first step, function <a href="https://rdrr.io/cran/optiSel/man/subPed.html">subPed</a> is used
below to create a small pedigree that includes only the individuals to
be plotted, which are <code>&quot;Chantalle&quot;</code> and
<code>&quot;Angelika&quot;</code>, up to <code>prevGen</code> previous (ancestral)
generations, and up to <code>succGen</code> succeeding (descendant)
generations. The pedigree was plotted with function <a href="https://rdrr.io/cran/optiSel/man/pedplot.html">pedplot</a>. It is
basically a wrapper function for function <code>plot.pedigree</code>
from package <code>kinship2</code>, but has additional arguments.
Parameter <code>label</code> contains the columns of the pedigree to be
used for labeling. By default, symbols of individuals included in vector
<code>keep</code> are plotted in black and for individuals from other
breeds the symbol is crossed out.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>sPed <span class="ot">&lt;-</span> <span class="fu">subPed</span>(Pedig, <span class="at">keep =</span> <span class="fu">c</span>(<span class="st">&quot;Chantalle&quot;</span>,<span class="st">&quot;Angelika&quot;</span>), <span class="at">prevGen =</span> <span class="dv">2</span>, <span class="at">succGen =</span> <span class="dv">1</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">pedplot</span>(sPed, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;Indiv&quot;</span>, <span class="st">&quot;Born&quot;</span>, <span class="st">&quot;Breed&quot;</span>), <span class="at">cex =</span> <span class="fl">0.55</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAk1BMVEUAAAAAADoAAGYAOpAAZrY6AAA6AGY6Ojo6OpA6ZmY6ZrY6kJA6kNtmAABmADpmAGZmOmZmOpBmZgBmZjpmZmZmZrZmkNtmtv+QOgCQOjqQOmaQ27aQ2/+2ZgC2Zjq2Zma2tma2tv+2/9u2///bkDrbkGbbtmbb2//b/7bb////tmb/trb/25D/29v//7b//9v///9kXeBPAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAIxElEQVR4nO2di3YaNxRF5Rg3bTJ2k75IWnDSMmljSjD//3XVCzwPzRxJozuG+uysYB66utK2RgithawOZBT13A04dygIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgL8PwQJ9qJw1apDUuDZpbERacUTq4uuPbGn7XLxfU4WeiaCkseCaj9IThNXvp9qOlmC0q8W1b4vlaabqgAUlFZdVO0ZE65q35VK00lVgtkFKck00XXHM7cgJZomuu54uoKkaKebUENqj6YCR1AgH2z5YC0KFUhKE+aSBSlUIC1NmAsWpFCBxDRhzk4QrvTQe28vkybMGQiCK7hgLQoVSE4T5twERVR6KCAolCbMOQhCHyNDtShUYDRN8NUwpQWhqXAg32jDA7VIpIlNLktOvuj+T0sjXZNcvl4MruSFC5JJI12TXD4KKh9DQQJppGuSy0dB5WPORdBaqYX5sQi9uK2G8+1ulFr2Snx78IHtF1ptNC/Vi84TQ2nCbTP1B6NCTBK01cnXy8P+3btN6NXhlu9/3Bz2d6tOif37OEERnTuFhNpm4ucRtPve9qiuaj0alHq1sTd6dFytfDsO+1t1/dB4Wp2aaH78cOOftyVqtXAPtCAbF2rjttq98dVqoY+/b3RpEzzQrae27W+vfl665tgR9PihOrZ0jGmXmM58/fD4caVN6dFUL+2NvlB0o7wF07yq8bTLZ1/XJa4f/PO2hO6we6CbXzcuwLagt7eVr/Zwv9KyfPXhbjXaVi8fPyxdc8y/t3erw7GlcoJMX83vwf7abdcOph16flmr679NB/UspRZPTzdH0L+bp+ftPTMi3INtZeOCgl799XHlqj14jzY43K1G2+5X+sY3x4w6PXZckwQFmcmyrvQsdGzlsSXr5giy945P+znozrSuUfwoyD+wPQ+3UZfUM5h72U4wo4IabdNF1kvfHHsR+2ygj5Pfxa6/6glXt/nzSZC+7n4xbrbmt6UvelU1nu6+i/nn/Vha+Ad2DlLhS8zMsNdf3cvmLWpM0L7RNl3j66VvjotZHFsqJyid51sH2RE0S/JJPJ+g5vuibPJJvLCVdI/dd/qi13PfwOI6mC81Rg+FVxv9BmRK6+vG35ueJrrBU9jdLGwrBhbX4ZYnxmgptZlk9YyydjN9f27JSROmtKA3urFmxVENrMBCLU+P2S6/rLSZ/Z86wt4rkSZMcUH6nfV+5RawkfnSY9wSZutW1PfRgkCaMMUF6U+w+hOAXcBG5kuOMaubLydB8SMIpAlTXtBhfbVyC9jIfKkxdrb1M0+dMAehNGEEBO3v/nAL2NBcONDylBjzearKeBdDacJwHTRbTXL5KKh8DAUJpJGuqbOEH3in6E64KTFmmrUfFhpFT/v809OEKSeos4SPEpQWExB03OcvkSZMOUGd7XG/77593d5PVxNivCD9vm423u1n1tpuydl9+NbOfU6aMMUEdbfHj/vFnTWKmhBjFsF6FVxX9sOqWQL9o0fQcR++tXOfkyZMMUHd7XG/7z46OSTG+BGki+7e2FFwtdKCTvvwg4Ii04QpJqi7Pe733UcFJcZ4QX4E2YJuBPU3pnPShCklqLc97vfdxwSlxrTnoFrZfbPFaR9+SFBsmjAS66Cx7fGhfDkxI+SkKZYcMrY9PpQvJ2aEnDTFkk/iJa+kxfJRUPkYChJII12TXD4KKh9DQQJppGuSy0dB5WMoSCCNdE1y+SiofAwFCaSRrkkuHwWVj6EggTTSNcnle0mCok+smxaTk2awqjlRGRkzYnLSjNU1GyojZUZMThpQ2Tyozs+EGOEQXNssqN4diZCs30NEC+RRwbulQxoFi/RtPkFq8EHJkFaxEp2bTZAafVgqpFOoQO84glLaIAvnoOhcfBcbz8V10HgyrqTHs/Gz2Hg6fpovn+8l7QddHhQEoCAABQEoCEBBAAoCSApKOByvj/km2O6msv8jMQcWHb/h9C3joJcggoJSDscLRL93B4wlnCPRON1u6It26QgKSjkcr4/u4u7mamX+uyPvbntH3XVxgmzp0xftor40N4acoKTD8fo0RpAtbL8/N57RXWK29OmLdklH4YSQE5R0OF6fhqC1PwwNHbnhLi9X+vRFu6ndkBOUdDhen/YIsqDeutdt6acv2k3thpigfdLheIH4J0G2cN0/brOLs2FLP33Rbmo/uA4CUBCAggAUBJhHUOdkv6j1cf80QHiyVj8k52jJNvMI6pzs527SYva/JqdprBDyme0Sa57sZ28SY/S6M2Y4NEM+/XQxI6hzsl/kR9BWjP6x+y0tZN0/NT+dmQS1T/aLE9SNiehtO+TT5mIEdU72ixLUiTGDCAVlhEDmEdQ52S9KUMxpgDDkcuagi4WCABQEoCDALIJyTunLiJlwGOAwcwjKOaUvI2bKYYDDzCEo55S+jJgphwEOM4OgnFP6MmImHQY4zAyCck7py4iZdBjgMDMIyjmlLyNm0mGAw8gLyjmlLyNm2mGAw8y7Dsr5+zoZMZl/xifIvIJy/r5ORkzmn/EJwpU0gIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCDAfzziH66Svla8AAAAAElFTkSuQmCC" /><!-- --></p>
<p>Hereby, <code>SAnna-Lena</code> denotes the unknown sire of
<code>Anna-Lena</code>.</p>
</div>
<div id="quality-control" class="section level2">
<h2>Quality Control</h2>
<p>The next step is to find out if the completeness of the pedigree is
sufficient for the intended evaluation, and to identify individuals with
sufficient pedigree information. This will be demonstrated at the
example of a pedigree of Hinterwald cattle. Data frame <code>Phen</code>
contains selection candidates with breeding values in column
<code>BV</code>, and data frame <code>PedigWithErrors</code> contains
their pedigree. The column with individual IDs is called
<code>Indiv</code> in both data sets.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PedigWithErrors&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;Phen&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>Pedig <span class="ot">&lt;-</span> <span class="fu">prePed</span>(PedigWithErrors)</span></code></pre></div>
<p>Function <a href="https://rdrr.io/cran/optiSel/man/completeness.html">completeness</a>
can be used to determine the proportion of known ancestors of each
specified individual in each ancestral generation:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>compl <span class="ot">&lt;-</span> <span class="fu">completeness</span>(Pedig, <span class="at">keep=</span>Phen<span class="sc">$</span>Indiv, <span class="at">by=</span><span class="st">&quot;Indiv&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">head</span>(compl)</span></code></pre></div>
<pre><code>##               Indiv Generation Completeness
## 269 276000891901739          0          1.0
## 270 276000891901739          1          0.5
## 279 276000891471209          0          1.0
## 280 276000891471209          1          0.5
## 283 276000891862755          0          1.0
## 284 276000891862755          1          0.5</code></pre>
<p>or the mean completeness of the pedigrees of specified individuals
within sexes:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>compl <span class="ot">&lt;-</span> <span class="fu">completeness</span>(Pedig, <span class="at">keep=</span>Phen<span class="sc">$</span>Indiv, <span class="at">by=</span><span class="st">&quot;Sex&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="fu">ggplot</span>(compl, <span class="fu">aes</span>(<span class="at">x=</span>Generation, <span class="at">y=</span>Completeness, <span class="at">col=</span>Sex)) <span class="sc">+</span> <span class="fu">geom_line</span>()</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAyVBMVEUAAAAAADoAAGYAOmYAOpAAZrYAv8QzMzM6AAA6ADo6AGY6kNtNTU1NTW5NTY5NbqtNjshmAABmADpmAGZmOgBmtv9uTU1uTW5uTY5ubqtuq+SOTU2OTW6OTY6OyP+QOgCQOjqQkDqQkGaQtpCQ27aQ2/+rbk2rbm6rbo6ryKur5P+2ZgC22/+2/7a2///Ijk3I///bkDrb/7bb///kq27k///r6+v4dm3/tmb/yI7/25D/27b/29v/5Kv//7b//8j//9v//+T///+mDbWcAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALgUlEQVR4nO2dDXvbthHHmcxynTppq3XqtnSLu3Xu1nqzq7k1ZW1WFH7/D1W8kQBIEAcQgEQQ93+elOKLWeGnAwgecLiqQVlVnfsLzF0ICBACAoSAACEgQAgI0GRAT0xiI+V/YNotYjKwCgEBQkCAEBAgBAQIAQFCQIAQECAroJc//cy2H799++Wv3YYLATXN89svGKBPP3xofvmq3QghoObh3Y/cgj7+9WdqTGIjziKgpqtiL3/+tfn4l5/Ehhz4jIhfcZn8C55bLoCev2RkxEac479lfYkWZLAgKgHo6RIuzOIB2dqgPqEiAX364T1/ir0fPsXqJ51QeYDoP0s/iADSCJUFyCYFkEoIAbVSASmEEFArDZAkhIBatV+1Z0IIqJUOqCOEgFr1ALWEEFCrPiBBCAG1QkCABoA4IQTUqvuqOiEE1MoAiBJCQK1MgAghBNTKCOjpEgG1MgNCC+o0AqjvgkVAPUKNt5O6NEDeTuriAPk6qcsD5OmkLhCQn5N6qYCk6uGhBQ24xpj+UgO/dpkWZAfk4cUvE5CHk7pQQO5O6lIBOTupiwXk6oMtAJBKCAG1ggE5OqkLBuTmpEZACGgUkJMXv2hALl78EgAphPqFgb34CAgB2QDBXnwEhICsgMBhjiIASUIIqJUzIGgcCAEhIAAQMFCGgBBQqYC62JVf3lJ9YNsvRkIya0th7EOt2QLSY3hpuOHDB+U0AtLi52ik4ad//qSc9gFkHYvOFpAWgUlNiVQ5WtEaNai3k2H8WVHGQ9HjgNQYXrZ9+aNqRV4WZIsoW4QFPXeRhl071P/udXGA1Dbo4X17FAF1UmJ4ecWiZvTpX2MrL9gBWWYzZAtI9INYUC+vaaQf9K57kCEgQAgI0OC71/bSjc6HyQnQYb05rKvX9y5/WCSgu1WzfX2/Xbn8YYmAiAEdb1bNzsmEfAGNzqjKC9BhfY2ANKmAjjfXu1e3tKI5qERAzf6qWjV3F48ufzj87jVQupE5eTkB8hECAuQPaGROXk6AgvpBJQAK6gcVACisH8QJWUtnnNWZF6CAflABgML6QQUACuwHMUL20pmmveYEyEcICJDxu9dnB7Stqoo0E/GkAyK332ynV7HzA9rSB8yu2oRikdL7QRf/5U96B5m/ew2UzjCzPCag482Gl8PpN3bS4DG/mf6YnwOga/mxou8EW1LfaN/l9IBGZB+CTj4GvasqDoPVA9pakE6LW79lRFoV29Iq5sh75MetDb/26Syo4a00AcJ+Zfpz7998/3nAD95rpMkP0P4EEwE1fUK9K4bBG/Ef84f1xSPlxMuyDWuyo6eNmAEgYjW38ll8V4XUsAR5NepBYU4HiJARm13bGdq9/s86xIT6rxpUAU+xMwNq7igX2kAfb4gJEUrBTx39ZdXDGMe/u33NrkH4T/yeNGtF6WOevnkTTl7lsgE6+NiipTDD5WBOByi6dAuKA8i62knOgLwqq7UwtUeEXUaADusqQiPNNT6zPDxVScbuDuXA6KxOBKQTWhagUH+QdmBksD5nQOH+IO3AyEBZcDagMzbSwf4g/YB5mCMhoP6F7fVzBWT24ucLKIY/qHfA5KTOGFAMf1Bv3+SDDU24dUZAHnItjMHFmC0g8bIasQ2iB4YeNAS0TEDCi8t93hEBGTxogTntzm1BbvIozMCDliugnmTGPhHuPDmVsXVJ7tMBOt6MP6H3Y2ND4+9iStQzj6Kbnsp4JoBGITgD0t7FZMShCFQNSGXcczGeB9BhXVEnPnV3Hb/7O+nv8WFYOlKxYYCME1jHXzVkzKoIdzalMnZVb0Q61Qg09WX2pVvQ3XWzXTE3/v5qRY8cfn9LN/JcT+OAZNSzCHc2pTJ2/vl1H2xY4tGAKkZxHL6+P35329B/dJcW/Ot7ec4CSH8X0zMXk3bIlMo4O0BrNsFKA3RXkZolz9kAae9iepNDAIWlU9ec1OcDxC1EAcQmOLAqNrSeISBVMupZhDsbUxlnBoi2M7uLRwUQPbh/cyvPOQNSop5FuPPkfpBGiB0Iyu0bAog9ylQLol2b3/1hI8+NAWrHfCIN+8wR0CQld3fIA+owBwJaKCD2Ru/2xjqhdMo4UKaAtnxiX0Sf9BigoPzi53Z3RHaYKZ+VgTIEtERADZvYl66KqSOJWQLy6gqVCMhL00onh1rdsyMhoHwARZkGbD9Q5wwo0jTg5QJKNexjBOSegGw+gKJNA7YdqOcEyDbOYQAUbxrwUgGdoJE+JyCPwR4zICWgMR2gznGWBtDlQAog98EeM6BTNNKKc9o1CWJEC3Ie7DEDOkkjLefDnBeQfbDHDMilzVoIIGiwxwzIK1Zjutph6ATjz86AoMEeqRO/i7EDwoTOWsWAwZ5ZAHLNMzqfnrQS0JgUkG5COQFK7LTPHlBynzQCcj1Q5wnoZFVMM6GYgBLoLI10voA8FARIrWMIaDGAWBvN1rxwUXmA9le88bmbutit1wFJKBtA3UJWk1fiXDYg6S07QT8IAS0QkHRIu60sEAiIE8oJEO9GN86u+yiATKvizRYQX96KLSHn8ocFAuLDYq4rxYYC6upYRoC8hIAAISBAwYAYIQQEADKsq5gPIBnc8/INy0Bry/VcICAZ5EzDDGlYpi3X8wRAoo5lC0gGGD5TTA8frLmeCwSkh6iST9Zcz1NEx6BnnwXaLdczi8+05nqeYkHchBZhQR+/bXMZj6YyLg+QEuT88k3XPCOgTjLIWfCx53ouD5CMeqb9H9o8W3M9TwFECWUMyK5YgAYrcyIgBISAqKIAYo0QArID6q/MiYAQkM+BGgGBgPqtEAJCQAioiQaIxo8hIADQlNVvYzKwCgEBQkCAzg5o+D6GgBAQAmoiAmr6hBAQAvID1HcKLQRQRNVzHmCdiQV5L6wYk4FVcwDUq2MICAH5AtIJISAE5A2IEEJAACDPVd9iMrAKAQGaCSCVEAJCQBMAKYQQEAKaAkgSQkAIaBKgjhACGtuvERAEyD0MOiYDq+YESJgQAhrfr51DEGMysGpegBghBGQF5Dj1PiYDq2YGiBFymbQYk4FVcwP0VLvNqIrJwCqnoF7xKSyFnyMgishhPkxyMK1cgnrFp+mpjP0ANQTRJTTUegI0XC4BdeJTWBpRH0BPDNGlzUF0EjhULiGZ4lNIKuMp4pAGSv8/1uQS1Cs+BaUy9rQg8MBJ4FBNsiAqBHTWNigHQDKoV3wKS2W8PEBKKuPT9oOyAWQXAgKEgAAhIEAICBACchL8xgFeEeEWKYWAACEgQAgI0Bxm2s9aCAgQAgKEgAAhIEAhgDT3h1Ha6q8GMf+b9TbsCug2KRUASB8GMkpb/XWoZ1ps623YFdBtkioAkO6CNUlf/XWgh3c/Um+c5Tb8CuA2aRUASHfimyRXfx27BSm+/TbCoWm/TUoFANKHgUzSV381XUCKb78NQwjdJqWSWhCTrQFxsyDwNimVtA1iggDZb5MzIH0YyCR99VeDaPHtt2krofU2KZW8H/TO2kg594Ost0kp7EkDQkCAEBAgBAQIAQFCQIASAWI5kTfwdVz/+3ezf+OY1+zUSgLoeEPTAe4cs0bPFw5VEkAiz6ZbNskCAalpJI83VUVw7d98f8XqXLv/+d/IhuUMvKb/vaaQ6LlVI6+dhVIAkilbSZlXzJL2V8Satq/v5f5KXEcOUjjkHz1H/7XXJvhmE5QCEKsy1C5e3bLKRkDsrzbsuLb//0d+sQDEzu2oXW1mVO1SWhApJHuaiRo02KftOKXYAqJNluC1bEBtG0SBiHa6A6TuH9avblULKgZQ24DQQou0rW2htX1GZCctiJ7b8SZp4YB4zvpddfF4vCEQBAPeECv7lMj+6tUtrZJqI10AINa4sCc1fXTzisQLre43d+TzPwidu2qlPeYLALQcISBACAgQAgKEgAAhIEAICBACAoSAACEgQL8BIBX/nhGqTtwAAAAASUVORK5CYII=" /><!-- --></p>
<p>The completeness of the pedigree of each individual can be summarized
with function <a href="https://rdrr.io/cran/optiSel/man/summary.Pedig.html">summary</a>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>Summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(Pedig, <span class="at">keep.only=</span>Phen<span class="sc">$</span>Indiv)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="fu">head</span>(Summary[Summary<span class="sc">$</span>equiGen<span class="sc">&gt;</span><span class="fl">3.0</span>, <span class="sc">-</span><span class="dv">1</span>])</span></code></pre></div>
<pre><code>##                  equiGen fullGen maxGen       PCI  Inbreeding
## 276000812496744 6.032959       3     12 0.9677419 0.008903503
## 276000812202159 5.789307       1     12 0.7692308 0.027274609
## 276000812749837 4.419678       2     12 0.8333333 0.005828857
## 276000891618444 4.234375       1     12 0.5454545 0.000000000
## 276000812922523 5.575928       2     12 0.8333333 0.003438234
## 276000891862786 6.565186       3     12 0.9677419 0.013700247</code></pre>
<p>The following parameters were computed for each individual:</p>
<dl>
<dt><code>equiGen</code></dt>
<dd>
Number of equivalent complete generations, defined as the sum of the
proportion of known ancestors over all generations traced,
</dd>
<dt><code>fullGen</code></dt>
<dd>
Number of fully traced generations,
</dd>
<dt><code>maxGen</code></dt>
<dd>
Number of maximum generations traced,
</dd>
<dt><code>PCI</code></dt>
<dd>
Index of pedigree completeness, which is the harmonic mean of the
pedigree completenesses of the parents (MacCluer et al, 1983),
</dd>
<dt><code>Inbreeding</code></dt>
<dd>
Inbreeding coefficient.
</dd>
</dl>
<p>The best possibility to characterize completeness of pedigree
information by a single value is the <em>number of equivalent complete
generations</em>, averaged over all individuals of the actual breeding
population which were included in the evaluations.</p>
<p>The relevant parameter to identify individuals with insufficient
pedigree information to estimate inbreeding, however, is the
<code>PCI</code>. This is because inbreeding can be detected only if
both maternal and paternal ancestries are known. The harmonic mean
ensures that the less complete paternal pedigree is weighted more
heavily, so the <code>PCI</code> equals zero when either parent is
unkown. Inbreeding coefficients can be valid despite small
<code>PCI</code>s if the most recent founders were indeed unrelated,
e.g. because they were from other breeds.</p>
<p>Note that inbreeding coefficients can be computed faster with
function <a href="https://rdrr.io/cran/optiSel/man/pedInbreeding.html">pedInbreeding</a>.</p>
<div id="individual-specific-parameters" class="section level3">
<h3>Individual Specific Parameters</h3>
<div id="inbreeding-coefficients" class="section level4">
<h4>Inbreeding Coefficients</h4>
<p>The inbreeding coefficient of an individual is the probability that
two alleles chosen at random from the maternal and paternal haplotypes
are identically by descent (IBD). This parameter estimates the extent to
which the individual may suffer from inbreeding depression and predicts
the homogeneity of its offspring. It can be calculated with</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>Animal <span class="ot">&lt;-</span> <span class="fu">pedInbreeding</span>(Pedig)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="fu">mean</span>(Animal<span class="sc">$</span>Inbr[Animal<span class="sc">$</span>Indiv <span class="sc">%in%</span> Phen<span class="sc">$</span>Indiv])</span></code></pre></div>
<pre><code>## [1] 0.01943394</code></pre>
<p>Whether mating to a sound inbred individual should be favored or
avoided depends on whether the breeder wishes offspring with uniform or
heterogeneous breeding values. More important than the inbreeding
coefficient of an animal itself, however, is the expected inbreeding
coefficient of the offspring, which should be low. The expected
inbreeding coefficient of the offspring is equal to the kinship of the
parents.</p>
</div>
<div id="kinship" class="section level4">
<h4>Kinship</h4>
<p>The kinship between two individuals is the probability that two
alleles randomly chosen from both individuals are IBD. A matrix
containing the kinship between all pairs of individuals can be computed
with function <a href="https://rdrr.io/cran/optiSel/man/pedIBD.html">pedIBD</a>. It is
half the numerator relationship matrix. The R code below computes the
kinship between the female with ID <code>276000812750188</code> and all
male selection candidates that have a breeding value larger than 1.0 and
a pedigree with at least 5 equivalent complete generations.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>pKin   <span class="ot">&lt;-</span> <span class="fu">pedIBD</span>(Pedig, <span class="at">keep.only=</span>Phen<span class="sc">$</span>Indiv)</span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>isMale <span class="ot">&lt;-</span> Pedig<span class="sc">$</span>Sex<span class="sc">==</span><span class="st">&quot;male&quot;</span> <span class="sc">&amp;</span> (Pedig<span class="sc">$</span>Indiv <span class="sc">%in%</span> Phen<span class="sc">$</span>Indiv[Phen<span class="sc">$</span>BV<span class="sc">&gt;</span><span class="fl">1.0</span>])</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>males  <span class="ot">&lt;-</span> Pedig<span class="sc">$</span>Indiv[isMale <span class="sc">&amp;</span> <span class="fu">summary</span>(Pedig)<span class="sc">$</span>equiGen<span class="sc">&gt;</span><span class="dv">5</span>]</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>pKin[males, <span class="st">&quot;276000812750188&quot;</span>, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span></code></pre></div>
<pre><code>##                 276000812750188
## 276000811902819      0.28955731
## 276000812749676      0.06626139
## 276000812689003      0.03807358
## 276000812750190      0.18099209
## 276000813155662      0.17488755
## 276000812688988      0.16645344
## 276000812771544      0.01719556</code></pre>
<p>The males with lowest kinship should be favoured for mating.</p>
</div>
<div id="breed-composition" class="section level4">
<h4>Breed Composition</h4>
<p>Mating decisions should not only depend on the breeding value of the
male and the kinship between male and female, but also on the native
contribution of the male. Many endangered breeds have been graded up
with commercial high-yielding breeds. These increasing contributions
from other breeds displace the original genetic background of the
endangered breed, decrease the genetic contribution from native
ancestors, and reduce the conservation value of the breed.</p>
<p>For computing the breed composition of individuals it should be taken
into account that the breed name of founders born recently is in fact
unkown even if they have been classified as purebred. Hence, for these
individuals, the breed name should be changed to <code>&quot;unknown&quot;</code>.
Below, the breed name of founders born after 1970 is changed to
<code>&quot;unknown&quot;</code> if they had been classified as Hinterwald cattle.
Thereafter, function <a href="https://rdrr.io/cran/optiSel/man/pedBreedComp.html">pedBreedComp</a>
was used to estimate the contribution of each individual from each
foreign breed and from native founders. The contribution an individual
has from native founders is called the native contribution NC of the
individual. Finally, column <code>NC</code> containing the native
contributions was added to the pedigree.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>Pedig <span class="ot">&lt;-</span> <span class="fu">prePed</span>(PedigWithErrors, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="at">lastNative=</span><span class="dv">1970</span>)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>cont  <span class="ot">&lt;-</span> <span class="fu">pedBreedComp</span>(Pedig, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>Pedig<span class="sc">$</span>NC <span class="ot">&lt;-</span> cont<span class="sc">$</span>native</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="fu">head</span>(cont[<span class="fu">rev</span>(Phen<span class="sc">$</span>Indiv), <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>])</span></code></pre></div>
<pre><code>##                    native    unknown     unbek0  Fleckvieh Vorderwaelder
## 276000812749835 0.6096039 0.09570312 0.03677368 0.11021423    0.14770508
## 276000812749569 0.5945129 0.14160156 0.04071045 0.11599731    0.10717773
## 276000891724277 0.0000000 0.50000000 0.50000000 0.00000000    0.00000000
## 276000891920412 0.5501404 0.12109375 0.03527832 0.10354614    0.18994141
## 276000812496874 0.3734741 0.41601562 0.03430176 0.08294678    0.09326172
## 276000811287745 0.5804138 0.09765625 0.03259277 0.08963013    0.19970703</code></pre>
<p>The columns are ordered so that the most influential foreign breeds
come first. It can be seen that the contribution from native founders
varies considerably between individuals. Individuals with high genetic
contribution from native founders should be favored for mating provided
that the kinship between male and female is sufficiently low.</p>
</div>
<div id="kinship-at-native-alleles" class="section level4">
<h4>Kinship at Native Alleles</h4>
<p>Since animals with high native contributions tend to be related, the
inbreeding level could increase considerably when introgressed genetic
material is removed from the population. This could be avoided by
restricting the incease in kinship at native alleles in the population.
The kinship at native alleles is also called the native kinship. An
R-Object containing the information needed to estimate native kinship
from pedigree can be obtained as:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>pKinatN <span class="ot">&lt;-</span> <span class="fu">pedIBDatN</span>(Pedig, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="at">keep.only=</span>Phen<span class="sc">$</span>Indiv)</span></code></pre></div>
<p>For pairs of individuals the native kinship can be estimated as:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>pKinatN<span class="sc">$</span>of <span class="ot">&lt;-</span> pKinatN<span class="sc">$</span>Q1<span class="sc">/</span>pKinatN<span class="sc">$</span>Q2</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>pKinatN<span class="sc">$</span>of[<span class="st">&quot;276000891862786&quot;</span>,<span class="st">&quot;276000812497659&quot;</span>]</span></code></pre></div>
<pre><code>## [1] 0.182625</code></pre>
<p>However, the kinship at native alleles between two individuals says
nothing about the amount of genetic material the individuals have from
native ancesors. It only quantifies how different these native alleles
are. Hence, in any case, the native contributions of the individuals
should also be considered:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>Pedig[<span class="fu">c</span>(<span class="st">&quot;276000891862786&quot;</span>, <span class="st">&quot;276000812497659&quot;</span>),<span class="fu">c</span>(<span class="st">&quot;Born&quot;</span>,<span class="st">&quot;NC&quot;</span>)]</span></code></pre></div>
<pre><code>##                 Born        NC
## 276000891862786 2004 0.6164551
## 276000812497659 2004 0.6274414</code></pre>
<p>The native contributions of both indivividuals are larger than the
mean NC of the phenotyped individuals, which is 0.44. However, their
native kinship is larger than the average, which is</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>pKinatN<span class="sc">$</span>mean</span></code></pre></div>
<pre><code>## [1] 0.0776925</code></pre>
<p>The correlation between the kinship and the native kinship is high
(as expected):</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>subKin    <span class="ot">&lt;-</span> pKin[Phen<span class="sc">$</span>Indiv, Phen<span class="sc">$</span>Indiv]</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>subNatKin <span class="ot">&lt;-</span> pKinatN<span class="sc">$</span>of[Phen<span class="sc">$</span>Indiv, Phen<span class="sc">$</span>Indiv]</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a><span class="fu">diag</span>(subKin)    <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="fu">diag</span>(subNatKin) <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">c</span>(subKin), <span class="fu">c</span>(subNatKin), <span class="at">use=</span><span class="st">&quot;complete.obs&quot;</span>)</span></code></pre></div>
<pre><code>## [1] 0.8916129</code></pre>
<p>The correlation is even higher if only individuals with high native
contributions are considered.</p>
</div>
</div>
<div id="population-specific-parameters" class="section level3">
<h3>Population Specific Parameters</h3>
<div id="genetic-diversity" class="section level4">
<h4>Genetic Diversity</h4>
<p>The genetic diversity of a population is the probability that two
alleles chosen at random from the population are not IBD. It is one
minus the average kinship of the individuals. A simple estimate can be
obtained as</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">mean</span>(pKin[Phen<span class="sc">$</span>Indiv, Phen<span class="sc">$</span>Indiv])</span></code></pre></div>
<pre><code>## [1] 0.9755853</code></pre>
<p>The diversity of this population is high. A high genetic diversity
enables to avoid inbreeding and ensures that polygenic traits have a
high additive variance, which is required to achieve genetic gain.
However, the diversity of this population seems to be extremely high.
This could have two reasons:</p>
<ul>
<li><p>The diversity is indeed very high because the individuals are in
fact crossbred individuals with genetic contributions from many
unrelated breeds, or</p></li>
<li><p>The completeness of the pedigrees is insufficient for such an
evaluation.</p></li>
</ul>
<p>For this breed it is unclear whether pedigree completeness is low
because pedigrees of indivduals from other breeds were cut, or because
previously unregistered animals have been registered. Hence, pedigrees
of introduced individuals from other breeds should not be cut to reduce
this uncertainty.</p>
<p>A parameter which depends not so much on the completeness of the
pedigrees is the diversity at native alleles.</p>
</div>
<div id="kinship-and-diversity-at-native-alleles" class="section level4">
<h4>Kinship and Diversity at Native Alleles</h4>
<p>The kinship at native alleles in the population is the probability
that two alleles chosen at random from the population are not IBD, given
that both alleles originate from native founders. The mean kinship at
native alleles is</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>pKinatN<span class="sc">$</span>mean</span></code></pre></div>
<pre><code>## [1] 0.0776925</code></pre>
<p>The genetic diversity at native alleles is one minus the kinship at
native alleles <span class="citation">(named <em>conditional gene
diversity</em> in Wellmann, Hartwig, and Bennewitz 2012)</span>. Thus,
it can be calculated as</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> pKinatN<span class="sc">$</span>mean</span></code></pre></div>
<pre><code>## [1] 0.9223075</code></pre>
<p>Note that incomplete pedigrees result in an overestimation of the
genetic diversity. This is not the case for the genetic diversity at
native alleles because alleles originating from founders born after 1970
were classified to be non-native. Hence, the diversity at alleles
originating from these individuals does not affect the estimate.</p>
<p>A high diversity at native alleles is important if a goal of the
breeding program is to remove introgressed genetic material from the
population. Without maintenance of a high diversity at native alleles,
inbreeding coefficients will soon rise to an unreasonable level.</p>
</div>
<div id="native-effective-size" class="section level4">
<h4>Native Effective Size</h4>
<p>The <em>native effective size</em> of a population is defined as the
size of an idealized random mating population for which the <em>genetic
diversity</em> decreases as fast as the <em>diversity at native
alleles</em> decreases in the population under study <span class="citation">(Wellmann, Hartwig, and Bennewitz 2012)</span>. Thus,
the native effective size quantifies how fast the diversity at native
alleles decreases. In contrast, the effective size quantifies how fast
the genetic diversity decreases.</p>
<p>In a population without historic introgression the native effective
size (native Ne) is equal to the effective size (Ne) as estimated by
<span class="citation">Wellmann and Bennewitz (2011)</span>. But in a
population with historic introgression it can be argued that the
effective size is not useful to describe the history of a population
because even a small amount of introgression with unrelated individuals
prevents a drop of genetic diversity, so the effecive size would be
infinite.</p>
<p>For estimating the native effective size, the diversity at native
alleles needs to be estimated at various times, and then the native
effective size is estimated from the slope of a regression function.</p>
<p>An estimate of the native effective size is automatically provided
when the kinship at native alleles is computed and parameter is
specified:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>pKinatN  <span class="ot">&lt;-</span> <span class="fu">pedIBDatN</span>(Pedig, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="at">keep.only=</span>Phen<span class="sc">$</span>Indiv, <span class="at">nGen=</span><span class="dv">6</span>)</span></code></pre></div>
<pre><code>## Number of Migrant Founders: 237
## Number of Native  Founders: 150
## Individuals in Pedigree   : 1658
## Native effective size     : 49.5</code></pre>
<p>The native Ne is stored as an attribute of the result:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">attributes</span>(pKinatN)<span class="sc">$</span>nativeNe</span></code></pre></div>
<pre><code>## [1] 49.5</code></pre>
<p>Thus, the diversity at native alleles decreases as fast as in an
idealized population consisting of 49.5 individuals.</p>
</div>
<div id="effective-size" class="section level4">
<h4>Effective Size</h4>
<p>The effective size Ne of a population is the size of an idealized
random mating population for which the <em>genetic diversity</em>
decreases as fast as in the population under study. It is commonly
estimated from the mean rate of increase in coancestry <span class="citation">(Cervantes et al. 2011)</span>, whereby the increase in
coancestry between any pair of individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> is computed as <span class="math inline">\(\Delta
c_{ij}=1-\sqrt[\frac{g_i+g_j}{2}]{1-c_{ij}}\)</span>, where <span class="math inline">\(c_{ij}\)</span> is the kinship between <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and <span class="math inline">\(g_i,g_j\)</span> are the numbers of equivalent
complete generations of individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>. The effective size is then estimated
as <span class="math inline">\(N_e=\frac{1}{2\overline{\Delta
c}}\)</span>. Thus, the effective size can be estimated as:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>id     <span class="ot">&lt;-</span> Summary<span class="sc">$</span>Indiv[Summary<span class="sc">$</span>equiGen<span class="sc">&gt;=</span><span class="dv">4</span> <span class="sc">&amp;</span> Summary<span class="sc">$</span>Indiv <span class="sc">%in%</span> Phen<span class="sc">$</span>Indiv]</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>g      <span class="ot">&lt;-</span> Summary[id, <span class="st">&quot;equiGen&quot;</span>]</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>N      <span class="ot">&lt;-</span> <span class="fu">length</span>(g)</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>n      <span class="ot">&lt;-</span> (<span class="fu">matrix</span>(g, N, N, <span class="at">byrow=</span><span class="cn">TRUE</span>) <span class="sc">+</span> <span class="fu">matrix</span>(g, N, N, <span class="at">byrow=</span><span class="cn">FALSE</span>))<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>deltaC <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> (<span class="dv">1</span><span class="sc">-</span>pKin[id,id])<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span>n)</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>Ne     <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>(<span class="dv">2</span><span class="sc">*</span><span class="fu">mean</span>(deltaC))</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>Ne</span></code></pre></div>
<pre><code>## [1] 97.95922</code></pre>
<p>However, the above formula assumes that ancestors are missing at
random. In populations with historic introgression, pedigrees of
individuals from foreign breeds are often cut, so their parents are not
missing at random. In fact, even a small amount of introgression
suffices to prevent a decrease of genetic diversity, so that the
effective size of such a population is <span class="math inline">\(\infty\)</span>.</p>
</div>
<div id="change-of-native-genome-equivalent-and-ne" class="section level4">
<h4>Change of Native Genome Equivalent and Ne</h4>
<p>The native genome equivalent NGE of a population is the minimum
number of founders that would be needed to create a population
consisting of unrelated individuals that has the same diversity at
native alleles as the population under study <span class="citation">(Wellmann, Hartwig, and Bennewitz 2012)</span>. It is
assumed that the individuals were unrelated in base year
<code>base</code>, which could be well before pedigree recording had
started. Between the base year and the year <code>lastNative</code> in
which the last native founder was born, the population had a historical
effective size of <code>histNe</code>. This assumption implies that the
founders of the pedigree were related due to ancestors whose pedigrees
had not been recorded.</p>
<p>The decrease of native genome equivalents is estimated below for the
time since <code>lastNative=1970</code> by assuming that individuals
were unrelated in year <code>base=1800</code> and that the historical
effective size was <code>histNe=150</code> between 1800 and 1970. Since
the computation for the full pedigree could take some time, the
parameters are estimated below from a subset of the pedigree, which are
the individuals included in vector <code>keep</code>. Vector
<code>keep</code> contains from each birth cohort 50 randomly sampled
Hinterwald cattle.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;PedigWithErrors&quot;</span>)</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>Pedig <span class="ot">&lt;-</span> <span class="fu">prePed</span>(PedigWithErrors, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="at">lastNative=</span><span class="dv">1970</span>)</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">sampleIndiv</span>(Pedig[Pedig<span class="sc">$</span>Breed<span class="sc">==</span><span class="st">&quot;Hinterwaelder&quot;</span>,], <span class="at">from=</span><span class="st">&quot;Born&quot;</span>, <span class="at">each=</span><span class="dv">50</span>)</span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>cand <span class="ot">&lt;-</span> <span class="fu">candes</span>(<span class="at">phen   =</span> Pedig[keep,],</span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a>               <span class="at">pKin   =</span> <span class="fu">pedIBD</span>(Pedig, <span class="at">keep.only=</span>keep), </span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a>               <span class="at">pKinatN=</span> <span class="fu">pedIBDatN</span>(Pedig, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>, <span class="at">keep.only=</span>keep), </span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a>               <span class="at">quiet=</span><span class="cn">TRUE</span>, <span class="at">reduce.data=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>sy <span class="ot">&lt;-</span> <span class="fu">summary</span>(cand, <span class="at">tlim=</span><span class="fu">c</span>(<span class="dv">1970</span>, <span class="dv">2000</span>), <span class="at">histNe=</span><span class="dv">150</span>, <span class="at">base=</span><span class="dv">1800</span>, <span class="at">df=</span><span class="dv">4</span>)</span></code></pre></div>
<pre><code>## 
##         t    I  pKin pKinatN    Ne  NGE
## 1970 1970 5.46 0.004   0.005  55.4 5.06
## 1971 1971 5.44 0.004   0.006  55.4 5.01
## 1972 1972 5.42 0.005   0.007  55.4 4.95
## 1973 1973 5.39 0.006   0.009  55.3 4.88
## 1974 1974 5.37 0.006   0.011  55.1 4.79
## 1975 1975 5.35 0.007   0.014  55.0 4.66
## 1976 1976 5.33 0.008   0.015  54.7 4.61
## 1977 1977 5.31 0.009   0.017  54.0 4.54
## 1978 1978 5.29 0.010   0.018  52.7 4.52
## 1979 1979 5.27 0.010   0.018  50.7 4.49
## 1980 1980 5.25 0.011   0.020  48.2 4.44
## 1981 1981 5.23 0.011   0.021  45.5 4.40
## 1982 1982 5.22 0.012   0.023  42.9 4.33
## 1983 1983 5.20 0.012   0.024  40.7 4.29
## 1984 1984 5.18 0.013   0.028  38.9 4.18
## 1985 1985 5.17 0.014   0.030  37.8 4.09
## 1986 1986 5.16 0.015   0.033  37.5 4.01
## 1987 1987 5.15 0.017   0.037  38.0 3.90
## 1988 1988 5.14 0.018   0.042  39.5 3.78
## 1989 1989 5.14 0.017   0.044  42.1 3.71
## 1990 1990 5.14 0.018   0.048  46.2 3.62
## 1991 1991 5.14 0.018   0.049  51.8 3.59
## 1992 1992 5.14 0.017   0.049  59.2 3.58
## 1993 1993 5.15 0.018   0.052  68.7 3.54
## 1994 1994 5.16 0.019   0.052  80.3 3.53
## 1995 1995 5.17 0.017   0.051  93.4 3.56
## 1996 1996 5.18 0.017   0.051 106.6 3.56
## 1997 1997 5.20 0.016   0.050 117.7 3.57
## 1998 1998 5.22 0.014   0.051 124.8 3.55
## 1999 1999 5.24 0.013   0.052 127.8 3.53
## 2000 2000 5.26 0.012   0.054 127.3 3.49</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">ggplot</span>(sy, <span class="fu">aes</span>(<span class="at">x=</span>t, <span class="at">y=</span>Ne)) <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>))</span></code></pre></div>
<pre><code>## Warning: Removed 5 rows containing missing values or values outside the scale range
## (`geom_line()`).</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a><span class="fu">ggplot</span>(sy, <span class="fu">aes</span>(<span class="at">x=</span>t, <span class="at">y=</span>NGE)) <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">7</span>))</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAolBMVEUAAAAAADoAAGYAOpAAZrYzMzM6AAA6ADo6kNtNTU1NTW5NTY5NbqtNjshmAABmtv9uTU1uTW5uTY5ubqtuq8huq+SOTU2OTW6OTY6OyP+QOgCQ2/+rbk2rbm6rbo6ryKur5P+2ZgC22/+2///Ijk3I///bkDrb/7bb///kq27k///r6+v/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+2WVyMAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAHq0lEQVR4nO2dC3ebNgCFtbXrmrZLukfSuo816Zxs6eouScP//2uzsOIYA7pIIDDw3dNT5cTXV+SzAAGWZDLklRl6Aw5dABICkBCAhAAkBCChOED/brXzY520paeQoGoAJCwAEhYACQuAhAVAwgIgYQGQsDQCdPPbZZbdnR29+rItALSj1dHLy+z+wyK7/uWhANCOli8+rVvQ3dtL25JcAaDSLnbz+5fs7s25K9a//Gkt35smJglo9Son4wr3QuSn0slH21M1jQGVWhCAioA4BglA9x9ON2exU85ilYDoB8UostJOtrynagAkLAASFgAJC4CEBUDCAiBhAZCwzBGQCakGQCIEQCJkhoAMxyC/ACQEICEA+WXoB/kFIPE6gMTrABKv9wpohIr9Q2fTggxX8/6XAQSgdg4A+R0mtBoAiRAAiRAAiZCZATLB1QBIhABIhABIhABIhMwLkAmvBkAiBEAiBEAiZFaATEQ1ABIhABIhABIhcwJkpKMiBEAipDGg6yOrRV6+HOdwqPQtyA43XC7G2oKMdFSFhACyIw3vP57v/CZkw2cAyA6kuzvLd7RshIN62z1db/DufKjqza+7rSjkk51+C1ptRxpuj0MhGz40ICMdlSEBgJan258AVKHNjmWb0f2fYzzNJwfkRsuv+0EvtieykA2fPKAqhWz4wICMdFSHAEiEAEiEzAWQkY6aEACJEACJEACJkJkAMtJRFwIgEQIgETIPQEY6akMAJEIAJEJmAchIR30IgEQIgETIHADtT0UBoD0ByO8ozWUCoKIAJBxDAhqDOvrLptuCytMpsYsVBCC/o2I+LgDtCkDCASC/o2rGOwDtCEDCASC/o3JOSQA9CkB+R/WkpADaCkB+R82stgB60CEBMiW1+9viLY+OummRD6QFlZFVUYvfcu04cECVlfo5AWi76dWQugVU21w7AHRlzMnVk3+SAcq3v+WdLOlICOjiyd/HJ9/fPU0KqAJRp4Dqj3etAd0en6z/ZV9//CstoBIiAJVVQNQlIM8Js/0udmV3sdvj54pPJx3FHUSjAZR9taeZPT5uuHOCJfy2iDoE5OtxJTrNb0bRpVnKOGZKBL+jf0BuoGqiZUQj5q3xOrxd9naAbo+31wWFg7Qb7pxqKeOOH313/yS9nHhRPAi54c7JljI2odP0+hz+a75OdrHb4x/elzEuFwmXMg6cbN7j6PKauAbQV1N5obFcpFzKOGzJFI8jPaALU7rMcMOdky5lHLSuVb1DxbQG9P2dOSm3HjfcOelSxprQIQD69kxfYyQCpNtQF5RbAroq7179Aaq6CRIakhhQXT+oJ0DtT9Fd7KaD31H0Otr1ghucCscOqNV1VJPO1OgB+Q5EANoo9n5yo+uVKQCKfGTTbM2DSQCq2828IQ3nG58GoJjv9swMUGUj6uKu22QABX5LvvF92wkBChipE3Dnf0qASrtZXUjIs6NJAWo45j1ofqmJAWrw8DXw8ezUAMmHr6EP+KcHyHsIbnqUCnGMD9CWw76l3FeaKyD3tbS9znbUYKepArLa+e5e3XcdDwrQIHI3hPuuNepdkZ9KJx9tT9UASFgAJCwAEhYACQuAhAVAwgIgYQGQsABIWAAkLAASFgAJC4CEBUDCAiBhAZCwAEhYACQsABIWAAkLgIQFQMLSHNDN63wF2hGv9RxVTWNAdpihHZY53rWe46ppDGhlR9EtFyNe6zmumsaAXCsa8VrP7dQEkB2OOeK1nuOqCQB0d3bqfhrlUsaR1TQHdPN6e3gGUD2f8a71HFlNY0C2/2MPz6Nd6zmymsaAKhVZaSdb3lM1ABIWAAkLgIQFQMICIGEBkLAASFgAJCwAEhYACQuAhAVAwgIgYQGQsABIWAAkLAASFgAJC4CEpR2gGYkWVGMBkLAASFgAJCwAEhYACQuAhAVAwgIgYQGQsABIWAAkLAASFgAJC4CEBUDCAiBhAZCwAEhYACQsABIWAAkLgIQlGFDSJfz6CUkKKM1Sxv2GJAWUchnRSQBKtZTxwaspoGRLGfcY0lMLAlClOAYJJV3KeAqA6AcFKbLSTra8p2oAJCwAEhYACQuAhKUdoEd1ctVxyCEAEiEAEiEAEiF8014IQEIAEgKQEICEogDld85uXtu5XR+mOCvcDQkNeSgGCXFz1bq3FYssDtDKbsjd2SK73oSsXn0pPhUKDbH3cq9bh2yK0BA3V617W7GwigC0fPFp/bk93qS2/xfvyIaG5MXbyw5C3pyHhri5at3bioV9PXoXewRkURfv6YeGuBbULqRYBP09pXfvhEQDylu0nVkxDyo+FQoO2ezy7UJcERFi77e7txUL+2Krg/QfHx2b2BbkQuxMuquXl+1CHrKCQ/K5artvQXn0elfNlqfZ/lOh0BD3gbULcUVwyGYu1u6PQflxY3302cyhXHwqFBriWlC7EFeEhri5at3bioVVfAtaHW16DJuWGNmFcSGrzUSxrUOKXZhG2uvIddIPmpUAJAQgIQAJAUgIQEIjAfTf56FqHgegbz+/H6pqAAmNAtC3Z8Y8H6juUQCiBSkBSAhAQgASApDQ7fHJUFWPA1B2YZ4OVPNIAA0nAAkBSAhAQgASApAQgIQAJAQgIQAJ/Q9ZstiAmI1PuwAAAABJRU5ErkJggg==" /><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAq1BMVEUAAAAAADoAAGYAOpAAZrYzMzM6AAA6ADo6kNtNTU1NTW5NTY5NbqtNjshmAABmtv9uTU1uTW5uTY5ubo5ubqtuq8huq+SOTU2OTW6OTY6Obk2ObquOyP+QOgCQ2/+rbk2rbm6rjk2ryKur5OSr5P+2ZgC22/+2///Ijk3I///bkDrb///kq27k///r6+v/tmb/yI7/25D/27b/5Kv//7b//8j//9v//+T///+8XYRaAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAGrUlEQVR4nO3dfV/bVBjG8aiIFNiGijxoNzalSrF0xdKR9//KzMMpzdPJddIk5JT+rj/Iplfv5fPlJA2BZUFIahMMvQO+ByARgEQAEgFIBCARV6D5JtlfW+NS8noQQKIDkOgAJDoAiQ5AogOQ6AAkOgCJDkCiA5DoNAXa27CCLB2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARMcB6PlmdHwLkB3obhwuTh8AsmV1PeUQqwNann8xh9iPUewL7Y2nBuhsHCFxiNmBIpzV1fos3fPu+DdIA63+AKgWKH4X4xCrA1pdjk5e3sh63h3/BjkA5dLz7vg3CCDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdFoABVG63h3/BrVcQYFS2jugqsRILV6+G2l7DrKtpL1bQXV/UhURQLmUiQAqpHikAVRO7nQEUHVeiACyxRABZE9CBFBdIiKA6uPy5dpeA0Ul/TXtngPFqTcCKE7NQgJoHcvNEYByCYKiE0BVCTZQANWlvKC2HNTZHs39AjIdb+/d+gKUpA4JIBPf7t16BxSnCgmgQny5Nekt0Lxwhxugqnhw79ZvoMyRBpA1KRFANQkGvDXpBPR8Mx4UaD7grUknoNlocKDBbk26AC1//d0HoDjtv15rDfT0/kMYfv3+H/MhPsA+/5UeYl48/WWon7SpAZpdeHAOypTafUHb/Qpanj/4BWQnGgZoNopz4ROQjWigc5AXb/Ol0mvd/d9ZoKpLo8GAMul5dxqWCkavBLT+fkPgP9A8f3Ntb78WU6WgwQ8j7SXQvMEPI+0rkPMPI3UAFJ+mJ9GqPdgtIIfvrTn/abVAj4dHYTg5SN/Odgoo6QTZdLBHZaDJgflw/8O/OwiUSxEpqE3loBJQunBioB24DnLpFAnsgyp/nMICFOeNADUuxX93KVMvAn37eGR+9QYOsQ4Glc9B92bh7OhJuutBZaBw8t2nMH4z27m3+Ve7UHw8jA7DRAmgN3Ql3dMggESnBLRrtzv6HmRfQZPgqPTfAHrJ0/vqs3TPu+PfIAvQ16DqKhGgzeFVeREEUJpvH4Oqi2iATB4Pq96+AFrn3np4ARRHXAftbbiStnQAEh2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARMcBaHk28uLZHcMM0kCrq9tw+fMtQLYs3kUf7tZLqOfd8W+QwyFmVpEnDzcZKuIBS+snU7CCqrK6fPEBqCLLs/HmNz3vjn+DNFDOB6By0sfj8C5mP8Ry6Xl3/BsEkOgAJDoAiQ5AogOQ6AAkOgCJDkCiA5DoACQ6AIkOQKIDkOgAJDoAiQ5AogOQ6AAkOgCJDkCiA5DoACQ6AIkOQKLTFGhvwwqydAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdAASHYBEByDRAUh0ABIdgEQHINEBSHQAEh2ARAcg0QFIdAASHQeg1eXo9AEgq8/zzTicvQPIvoCup+HylylAtizPH3i4Se3jcU7XQHF6/nz5N0gDbVYQQJXhHCSA4ocH8S7GdZCt4wCUS8+7498ggEQHINEBSHQAEp2mQJl09mXHTgwCSAwCSAwCSAziL7OIACQCkAhAIgCJuAIlt86WZ6OT6ctDXnO3Q5yTGbTeDDrIPLnfvDS/ieMItIh3YHU5DmfpCxenD/lvC7kmOyi+oTvrZFC62WaQeXK/eWl+k8QN6O74z+jztblLHX/M35J1TG5QsrmedjTo6nabQebJ/eal+U1SaHKIbYBi3vxNfedkBpkV1H5QftN0UHlCdlAToGQlH9+afwsg/20h52QHpYd6+0Fms+Wg+Oa7eWl+k/zfpifp3z4bmzYryAyK/9WOxcm0/aD1vK0GJU/u72YFJeOiwzO8uwiL3xZyTmaQ+US1H2Q2Ww1Kn0zfzTkoOWdEZ5/neBUVvi3kvkObQWYFtR9kNtsMMk/uNy/Nb5I0WkGLUXqVkK6+FpcvZlC0SU5pXQwqXL64pnBRt/V10P4GIBGARAASAUgEIBHfgf77e+Ad8Bzo8adPA+8BQCJ+Az0eBsHRsLvgNxArSAUgEYBEABIBSOTp/YeB98BzoHASHAy7A74DDR6ARAASAUgEIBGARAASAUgEIBGARP4HsrcnEf0+rjUAAAAASUVORK5CYII=" /></p>
<p>Not only the native genome equivalents (column <code>NGE</code>), but
also the native effective size (column <code>Ne</code>), the mean
kinship (column <code>pKin</code>), the mean kinship at native alleles
<code>pKinatN</code>, and the generation interval (column
<code>I</code>) have been estimated for all birth cohorts
<code>t</code>.</p>
<p>It is possible that the diversity at native alleles increases for a
short period of time, in which case the estimate of the native effective
size would be <code>NA</code>. Choose <code>df&lt;4</code> to get a
smooth estimate for the native effective size.</p>
</div>
<div id="change-of-breed-composition" class="section level4">
<h4>Change of Breed Composition</h4>
<p>For monitoring the introgression from other breeds, the contributions
of foreign breeds to all birth cohorts can be estimated with function <a href="https://rdrr.io/cran/optiSel/man/conttac.html">conttac</a> and
then plotted, e.g. with function ggplot.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>use  <span class="ot">&lt;-</span> Pedig<span class="sc">$</span>Breed<span class="sc">==</span><span class="st">&quot;Hinterwaelder&quot;</span> <span class="sc">&amp;</span> Pedig<span class="sc">$</span>Born <span class="sc">%in%</span> (<span class="dv">1950</span><span class="sc">:</span><span class="dv">1995</span>)</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a>cont <span class="ot">&lt;-</span> <span class="fu">pedBreedComp</span>(Pedig, <span class="at">thisBreed=</span><span class="st">&quot;Hinterwaelder&quot;</span>)</span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>contByYear <span class="ot">&lt;-</span> <span class="fu">conttac</span>(cont, <span class="at">cohort=</span>Pedig<span class="sc">$</span>Born, <span class="at">use=</span>use, <span class="at">mincont =</span> <span class="fl">0.01</span>)</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a><span class="fu">ggplot</span>(contByYear, <span class="fu">aes</span>(<span class="at">x=</span>Year, <span class="at">y=</span>Contribution, <span class="at">fill=</span>Breed)) <span class="sc">+</span> <span class="fu">geom_area</span>(<span class="at">color=</span><span class="st">&quot;black&quot;</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAEgCAMAAABb4lATAAABCFBMVEUAAAAAADoAAGYAOmYAOpAAZrYAujgAv8QzMzM6AAA6ADo6AGY6OgA6OmY6OpA6ZrY6kJA6kNtNTU1NTW5NTY5NbqtNjshhnP9mAABmADpmAGZmOgBmOpBmZgBmZmZmkJBmtv9uTU1uTW5uTY5ubqtuq8huq+SOTU2OTW6OTY6ObquOyP+QOgCQOjqQOmaQZgCQkDqQkGaQtpCQ27aQ2/+rbk2rbm6rbo6rjk2ryKur5P+2ZgC2Zjq225C2/7a2/9u2//+3nwDIjk3I///bkDrb25Db/7bb/9vb///kq27k///r6+v1ZOP4dm3/tmb/yI7/25D/29v/5Kv//7b//8j//9v//+T///90qKJvAAAACXBIWXMAAA7DAAAOwwHHb6hkAAATzElEQVR4nO2di3bcthGGIUeXyE5VS3YS1UlTOU2dplo1lypJ440TSd40sq1KW8lrvv+bFMCABMDrgASXBDj/ORG5BGFu+O3gOhiwhBS12NBfgNSvCHDkIsCRiwBHLgIcuQhw5GoN+I0p+9ObvtPW/Dh8mk8ynkSAfab5JONJBNhnmk8ynkSAfab5JONJBNhnmk8ynkSAfab5JONJBNhnmk8ynkSAfab5JONJtYBv//pSHt9+efDh6+wAGvKtEmC86gDfHDyWgN99f5y8+ig9KA35VgkwXjWALx79BBb89quXwpjVQaUO+VYJMF6YIvr2i9fJ278/Vwd+4X0u+1+ZrvpD40cYwDcfSrLqoNKsn/F/p6soLViIAINiANxcBw/9lgdUDIDfff8MWtHPqlrRQ7/lARU8YPFfYz946Lc8oIIGXCcCDCLAkYsARy4CHLkIcOQiwJGLAEcuAhy5CHDkIsCRiwBHLgIcuQhw5CLAkYsARy4CHLkIcOQiwJGLAEcuAhy5ogVsaei3PKC8vL8eRRbcUdFaMAEGEeDIRYAjFwLwaibXIR7Vvc7re2dtSdSLAHcUCvA+/3tdS5gAj1VowPC3SgR4rHICvHzwDbt3JopswTM93h2yjW8J8EiFL6I3L5Pl7pb4yP8sNi/T493hPmdMgEcqfCOLI1zuHqnS+O7wyDouCPBIhbbgu8PNy+X9U84SgnvsZ0du2rzwJsDjFBqwsFwALIAmuSMBHqvcAV9vnMrL2VGUztSKHqvcWtEC8GrGTZbTTY93h1vUyBqv8I0sTlkClp83TvWRukljFo1FRy4CHLkIcOQiwJGLAEcuAhy5CHDkIsCRCwH4Tbnavnk31QDOYo++OhA6lsfHFE7YFgLwoLHiqx9j78Ehwr1fHBvJBBiEAVyacXDAVvxvEen93Q/PjWQCDAoXsBXBX5gyL7JFQZ2UbMqx5pc6JmHecWnGwQGbe3DI4+3nphWTBYPisOCbLNJ7Vg8TYFC4gM06+OJZepUA5xQuYGMPDiiYhRm/+5G6SbY8A14+OPPrntXYD5abckBJzfvBj7KGNAEG+QfcgWbZw9tmJMCgzoCXe1/LhWnLXX64O2T3Xjx4sXeZrE5OxYfOuAlwR3UHvLsvVj3c/elUlc/LB7+e8PO9y/l+sthqyyd7eNuMBBjUHTAvlFXBfPcEAJ8tONp9gZxfaQsofXjbjAQY5A/wXKxuUUa897ssocHrspMIcEf5Anx3eJQV0Werk+/2Lrtbr3x424wEGOQLsDzcP1VnC+FFPZeVc1tA6cPbZiTAIG9F9IKx9z45Ws14K/oMPOSFQ3zXEpoAd1W4I1kNIsAgAhy5ggIsBlNgKXqzCDAoJMAyaARWBBgUEmDRFUOLAINCcrpbzQiws4Jym3WJI0CAQSFZsBz8pEaWmzCAz8tE3aQwRIAjV1iAFypYCEIEGBQU4AUE10MRJsCgkACrfjCuLU2AQdECtjT0Wx5QmHfsAnhujkWsTk4BDGs9L0xFdEf5tuASwGIAqrX3HTWyOqoz4NSjQ3nPzj+TB3CZ5YBXs83/PDlr7y9N3aSO8gYYvGeT+ealuAIusxwwP1nuXSbCxbKVCHBHeQOsPHdEET0/Ui6zq5MP9iFWfHfAvIVFQ5Ut5Bvwz6cSMLjMrmYPP70kCx5UPVgwb1Qpl1leRC/2pfe7lzqY+sEt1BmwsM3FPQ14K1F1MC+ZRSNLtLM6rGEhwB3VvZvEuy4Pn2jAn2UxpE9lN0k6xfvoBy+yiUrUj4UAgwIcycKJAINCAuwkAgwKCTB1k1ooJMAgZI+LAIPCA4xc0UaAQSE53SlRN8lFQbnNguZkwQ4KyYJVIwu3JpUAgzCAt8tE3aQwRIAjV1iAr0URjRvOIsCgoACTT5a7WECAaTbJXSwkCybAzmLnIQEGtNmcUrbrSrrdir6QTBAwY8VTdt4dMMTN2M+7bCiXaHlHltTCr8Pwycp64GDBxq4rEAXc3oYlesDp20g/nZ8bWFl21h0wQKsBrNUFcEE64rvaqMHahiV6wNkMgUDMWHaqzuH03C9gFT9YenNIVx3hyLE6+S5LWj74M7aX0wxY79mgtlvRF+LfdYWZM0DGB2aAh1PMO8YCVs7Qs6PkevN34RItXXb2fsuSUufpNoALbrN61xW13Yq5DUvkFlw+wSex5k/91MHS0045QwNxcImWcYXtJMdiGmPBUhfH9oWYAVfzLcpjEa2coYUbtFiQBC7R3JKNpG6A7Sg7dpXLAU+mDnbh6xXwE6O5BS7RMq6wndQBsO10p3ddUdutGNuwRA3Yia/XRpZyhp6JyNG/SpdoGVdYJ3UtohdW/a13XVHbrUykH2xWs4w14fbcis58opVLtIgrrJM6WzA53Zl8mSSTdotYoX0lz4IayXJRbICzcSpm4QXEJmoJW98TEuBJj0Ubw1TAzmShUSvwyqz5GQEOQ8wYfbTMN4+FZSfwGwgH8LTXJqXjy6xgvpWCm8Jyupvs2iQocbMTDF+lAN1mcYoKcNp6cucblAVPN6S/bhX3AninTAMAnkhI/2KEfXsww4lvUICnUQezcsBsAoAnEdLf8MwwLhmA3fgGBXgSIf2ltbL8FY6VteIbFOApjEVn7eXsQtbzlWeufIMC7KSwAZ8bvnTabhkB1goTsOFKx8wJIsW1coSyP8DQ8AEvDkOo9pCaOay91wYcabRZw/v1vFTuXP1Z8GI//WOqH8AxrE0qdIJYzs91bICF8YrZfeUY+4dP7gkPrPc+OdJXNj44kjsKC5+OXSPasPIEsO7NN6Cim03Kd3PN+fseDLg7YHCNvUydcnaPYOtvjjF1lT2SzpVMHIR3pXawTX15rHvzD58OYMvndTyAwTVWRhzl+KRR8lMRkNRwld27/PnbrTTA/xPDi7Z4bw3gGIro/DiGxVSV1Xm8XfiiXHbqAUvXWGmNqxPtGJuFFAbkvzx58eAX4Vo5F91YnVS8tw5wBI2sHOCizbLtbVa4UkUvddapvmOnO+DVP1+IgNBFCza8aOd/+ePq5NstWchaXrQl99YBbqvBeOYlxyrsjyXFsb0wpZougGAZa5Zhz/juIN5OUz948ZmwKe0Ya9Sr6ZVrBrWwNHDe3NJJxXs9AR6nBWcT98bHMnM1+sOVtqlRpKzlmTxl2T0eBjoEMbNhzE0ybRkrV1loQ4sKlJexdlLx3krAso21miHXNo0KsO0xl32qMODUDOuK53IoJnR1TzgjWWLlmtAcN+MQAmCjHVU2lVBTwTbxlYh3wgI838qf1GpcgG2XSPNDWgw7eNPhAO8EBljP9ofXD7ZdIlPCzMCrCPvkq0SAm/H0A7jgwY6fK3LgiwI8Cqc7sbUHaLHmYKQlPjTOfCVUY/G9BmwwxY5ouPDFAH5Tru7wMDJ6Y8pwNelaeQRc8KFpA9iMrmDSLl2Csl7A47Bg3rjagO4Yrp/kDXDRh6YdYDuURokBN1BtxxcF+OMyDTDhL92iccGEPQMuOsK587V6u6UGXG+1xuhUrICd5BlwB8Jlc0Sl/d6GYlmPThHgZLyAmeUE69IvIsCWfAHOecF15mvNFbkYMBB25UuA0XzaEi6sO2HugNO5BHe+BLg/wEx3dy3vyOaZoirA6fgyAU5GARjCSBbbVK4VcAu71cIEBMcATr0zfGtowGZ52iJr6jJpLUtwXoRCgEs0DsAGStYacA3fRvQon6x6wHKmX/x5oWLJVrrAttEYAeNIF9pSzJxW8FRAN1bKrLsFZ942aSzZShfYNhoRYD1R3w6wgcwbX86hljX7uDNg7W2nPCqrXWDbaGDAtleruuYKuMjSYRVZA18OAvrILD3LGXDnIlr7y6aAK11g22iMgHG1cT3gOl9YJ74ChcSbwdY+eMKAOwMuWnClC2wbjQlwNo+LAVw/nFHTBdbTCg2AmSQqZWCBDyzle9W9kaXrYOX0XukC20Y1gHVw2dunIqR/uvlKv4Bzns2InGUs6/imrs4IvhyglI1IQGb+AEOjeTXjreizXGTZXgHrTVZEmHcR1h82X/EJOL/GYE2A04mFxgaW4ssJMwOvJM6ArwfA/ar6MTrA+43AfHGsNl9R6gNwNo+LIMycZ4wspLDAodaANV7DiFlm06JmvgoZsL1FAz9Tm68k/nZdKa4DQwNmLnO+9mIT4FdLV1mnJVUda95wC+YdjxOwtcmKCOevNl9RyR4suGRhCXpymGUzg018C4tNUCoCzvNWFXQcFvz2y2fqalYP9wLYqo19ATYKZTTgRr6ykMYCHonTXU7GJiu3T7PmlUfANXwxgNOZQRfAeL7NgK/gjnDcZvPSm6wovmrzFZXcGnBDPBQnwLUjGjZfSRgJGMUXD3ikFqx3XRH9X9G8UpuvgNoDro+HgiOcjUU2Tupby0B9GrASBnD578M3yoqHt83YFnAa8sYT4EaZUL0aMAGuAWzFQ2Gl63jV3V0Bt5nLR7SwCHADX4tlthy7ALhy3qG57nUFbNTVLgZMgFGAtwuRUc7TXXtLJw/9LxRkH2chOZz4EmAk3xLANR7T/ktomDqypxgIsE/ARcL6rBie0DdgbsAw0uzKlwA38a0EbBfXhX/GM2AGswnMla9vwCqanT8NDthYtlm03XLCPfAVVJX5uvElwFgDNvzVBwAMDBzbV34AKz+O5d7XwmeWA17NNn+DD+ALIPcTlrFm08shATYjLGSX6gmXTDEgopkh+AojduXrDzD4zHLA8/3EcKDlhyzWbHo5UMA60lEtYFbIVzlgafRt6yd/3bn6BwyH1ckH+9m11B8vjTXbYn/o8QFm2Vk54dJZftuRrgiYN6F64usd8Ozhp5fmNc41izU7dsANfI3Zv0bAeb7ay8owZsOA6wA7jUz2DvjklJfItgVnsWZjAFwx7JGNXVYY8E7qZaVRW3yvqgmzTgbcHbAguLhnADbKYqiD7VizIQPW07ulgFlhlxQLpRETNmWt+VYC7obXRzdpwdjDJwZgfjAcaO8ZsWZHDthexVAa7DW3fswmXB2boQDPXqoPgCsCCnbkSyNZ5YCV12oB8HYN4ErLL0FX4Ftlwl35ThVwxRRQZowFoAXVe/TUG3Apw1KH2c58Jwq4bI6vfimCG2EXwHoQI/uo/aI6850q4JI5vrT9iw6NwnSOWsDMaF/VFcLM+OiJLhLwSJ3uHIQG7LaFDWPnZjTvagNmQAwB2A9VE/DI1YsFly0gY854AbE6NgG+qkKsicr5fP+Amy14UK0XsCtfg6KBN4WeATYWARZ7TJoF68GApws4T7gjYGMyMbuS8rUg5hAzO40Ao+UIuCvfbbU1VTGAO8ubJasw4Ks++E4YcHEPwW6Ai8TT9WRYwG2mewkwDrBb5CokXwW4gI1V8L3qge8kAZf4u/ZiwGo9WQFb5UgVAXbQKACLWYWSclcT7gMpAS4C7oGvniMsgcjWxpcA92rAOxVDF4wxAgzqE7DtZ7NOwFew1KiPVjMBVoOSecD+S+i6CSEILbkGvhMFzNREQa9VMKsdm2L5QFcE2E1NgGUHtc8SujkMzjr4ThNwNgXEbOTrBbweEWC4QoAHUp+Atxnry4C311P+IjQ9wEb4DJZGgPbOlwBj1SvgbdYX4NGU0FMEbE3Z5q/4AjwWvlMHDFs1E+Dh1C9gacI9GDABRqsHwDaIyEvokAHrTTnUmb6QVAPO4WQ54gR4zcJsyqHO9AUhLGD8SgYC3IcwAcHVmREhPKkBXGDhDK/8N8HMG4bGqhUuYB3SX53pC4VNOSpW3/Snfl9KTMJsyqHOrF06bAuuilrfU9qaH4dPWwsyN7WyYKEh3yoBxstPHUyA1YfxCbMphzrTF4SGfKsEGC/MphzN/WACrD6MT35Gsgiw+jA+EWCfaT7JeBIB9pnmk4wnEWCfaT7JeBIB9pnmk4wn+Rn0e7/5Fp/51vy41vnGIALcY74xiAD3mG8MIsA95huDaOItchHgyEWAIxcBjlwEOHJ1ACwn/2+fHjzmh1cHB+JoTShi8r37/uDR8wSVz8gmnnZwcOz+OHXoNd/I1B7wDRA9Tl7x/+mLY3HJdqxF5rv58DUmn5UtSbDZrHzC4ehVm3xwQOUbm1oDvnj0E/+JKz+tdz9IVy3bqQeRT2TA5bOyiSz8r/Pj5OGrl+3ywbdtyjc6dSyi0//1L2WRabvlIfLdfvFvUUSj8t1agIUlOT9OWXCLr1nidRiIOgKWhdej57efP0+4FduOtZh8T+WvApXPyAYGnDg/TlWizvnUAZdvZPLQyPoblM+8PsWbhsrnYhrW48S7drDg9HH8d3jz+KVzPvvbNuQbmboCTqAGFbo4xlVSRr63/8BXbtbjLp4luCrfyqdM0Dlf+m0nWAfLWu0jaVDvfnxpO9Yi8olWNLcLVD4zGzTqnB+nLNg5nzrg8o1MXS345kB2DXnPFNuftfLxDNgOppUNCkrnx920/Jr6206oH0wKQgQ4chHgyEWAIxcBjlwEOHIFCHg1E/vaJ8vdraG/SQgKEDBHu8//zjcvh/4iIShEwMmCm/By92jorxGEggR8d7iVzEUBvZoxJsrr5S5jbD9ZPvhGfiRpBQk4ud747v6p4MspLzYv7w6PpFlTtVxUmIC55Ypq+FqYK6f7P1EbL++fUrFdVJiAAW2ygKhonPU1P2ycLoVZkyyFDVg1pO8ON06lBRPggoIGfL0BQK8F6Guy4DIFDXg142Q5WAF6uUuAyxQ0YNlNElY854d/HR4R4KICBUzCigBHLgIcuQhw5CLAkYsARy4CHLkIcOQiwJGLAEeu/wMdSDsqf7BL3QAAAABJRU5ErkJggg==" /><!-- --></p>
</div>
</div>
<div id="references" class="section level3 unnumbered">
<h3 class="unnumbered">References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Cervantes2011" class="csl-entry">
Cervantes, I., F. Goyache, A. Molina, M. Valera, and J. P. Guitérrez.
2011. <span>“Estimation of Effective Population Size from the Rate of
Coancestry in Pedigreed Populations.”</span> <em>Animal Breeding and
Genetics</em> 128.
</div>
<div id="ref-Wellmann2011" class="csl-entry">
Wellmann, R., and J. Bennewitz. 2011. <span>“Identification and
Characterization of Hierarchical Structures in Dog Breeding Schemes, a
Novel Method Applied to the Norfolk Terrier.”</span> <em>Journal of
Animal Science</em> 89: 3846–58.
</div>
<div id="ref-Wellmann2012" class="csl-entry">
Wellmann, R., S. Hartwig, and J. Bennewitz. 2012. <span>“Optimum
Contribution Selection for Conserved Populations with Historic
Migration; with Application to Rare Cattle Breeds.”</span> <em>Genetics
Selection Evolution</em> 44 (34).
</div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
