<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Robin Wellmann" />

<meta name="date" content="2026-02-18" />

<title>2. Marker-based Evaluations</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">

div.csl-bib-body { }
div.csl-entry {
clear: both;
margin-bottom: 0em;
}
.hanging div.csl-entry {
margin-left:2em;
text-indent:-2em;
}
div.csl-left-margin {
min-width:2em;
float:left;
}
div.csl-right-inline {
margin-left:2em;
padding-left:1em;
}
div.csl-indent {
margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">2. Marker-based Evaluations</h1>
<h4 class="author">Robin Wellmann</h4>
<h4 class="date">2026-02-18</h4>



<p>Marker-based evaluations are an alternative or addition to pedigree
based methods. Pedigree based methods enable to estimate the
expectations of population specific and individual specific parameters.
The realized values, however, deviate from these expectations due to
mendelian segregation. In an increasing number of breeds pedigree based
methods are being replaced or combined with marker based methods to
overcome this limitation. In any case, the goal is to select animals for
breeding in a way that accelerates genetic gain, restricts the rate of
inbreeding, and maintains genetic originality of the breed.</p>
<p>All genomic evaluations included in this package are based on
haplotype segments. Every individual has two haplotypes: a paternal and
a maternal one. A haplotype segment is usually present in many
individuals because the individuals have a common ancestor from which
the segment originates. It is, however, also possible that two segments
are identical by chance. Of course, the likelihood for being identical
by chance is the smaller the more markers are included in the segments.
Therefor, a part of a chromosome is considered a segment only if it
contains at least <code>minSNP&gt;=20</code> consecutive markers.</p>
<p>If a segment is in a region with low recombination rate or high
marker density, however, then two segments could be identical even
though they originate from an ancestor who lived long time before breed
separation. Therefor, segments must have a minimum length which may be
measured in centiMorgan (<code>unitL=&quot;cM&quot;</code>) or Mega base pairs
(<code>unitL=&quot;Mb&quot;</code>). The minimum length of a segment is often
defined as <code>minL=2.0</code> Mb.</p>
<p>The proportion of the genome captured by a segment is usually
obtained from it’s length in Mega base pairs (<code>unitP=&quot;Mb&quot;</code>)
as the length of the segment divided by the length of the genome.</p>
<p>This section covers the following evaluations based on marker
data:</p>
<ul>
<li><a href="#data-preparation">Data Preparation</a>
<ul>
<li><a href="#genotype-file-format">Genotype File Format</a></li>
<li><a href="#marker-map-format">Marker Map Format</a></li>
</ul></li>
<li><a href="#example-data-set">Example Data Set</a></li>
<li><a href="#individual-specific-parameters">Individual Specific
Parameters</a>
<ul>
<li><a href="#inbreeding-coefficients">Inbreeding Coefficients</a></li>
<li><a href="#kinship">Kinship</a></li>
<li><a href="#genetic-distances">Genetic Distances</a></li>
<li><a href="#haplotype-frequencies">Haplotype Frequencies</a></li>
<li><a href="#breed-composition">Breed Composition</a></li>
<li><a href="#kinship-at-native-segments">Kinship at Native
Segments</a></li>
</ul></li>
<li><a href="#population-specific-parameters">Population Specific
Parameters</a>
<ul>
<li><a href="#genetic-diversity">Genetic Diversity</a></li>
<li><a href="#kinship-and-diversity-at-native-segments">Kinship and
Diversity at Native Segments</a></li>
</ul></li>
<li><a href="#multi-breed-specific-parameters">Multi-Breed Specific
Parameters</a>
<ul>
<li><a href="#kinships-within-and-between-breeds">Kinships Within and
Between Breeds</a></li>
<li><a href="#genetic-distances-between-breeds">Genetic Distances
Between Breeds</a></li>
<li><a href="#prioritizing-breeds-for-conservation">Prioritizing Breeds
for Conservation</a></li>
</ul></li>
</ul>
<p>Note that the calculation of breeding values is not included as there
already exist R packages for this purpose.</p>
<div id="data-preparation" class="section level3">
<h3>Data Preparation</h3>
<div id="genotype-file-format" class="section level4">
<h4>Genotype File Format</h4>
<p>In all genotype files, markers must be in rows and individuals in
columns. This enables to process one marker at a time without the need
to have the whole file in memory. For all functions included in this
package</p>
<ul>
<li>there is one file for each chromosome,</li>
<li>genotypes must be phased and imputed,</li>
<li>each file has a header and no row names,</li>
<li>cells are separated by blank spaces,</li>
<li>the number of rows is equal to the number of markers from the
respective chromosome,</li>
<li>the markers are in the same order as in the map,</li>
<li>there can be some extra columns on the left hand side containing no
genotype data,</li>
<li>the text within columns must have no white spaces,</li>
<li>the first rows may contain some comments,</li>
<li>the alleles of an individual are separated by a character, e.g. A/B,
0/1, A|B, A B, or 0 1,</li>
<li>the same two symbols must be used for all markers,</li>
<li>the IDs of the individuals are used as column names and must have no
white spaces,</li>
<li>if the blank space is used as separator (i.e. A B, or 0 1), then the
ID of each individual must be repeated in the header, so that the number
of column names is equal to the number of columns.</li>
</ul>
<p>Example 1:</p>
<pre><code>Note that the marker names are ignored when the file is processed.
I id 6415 6415 2636 2636
M ARS-BFGL-NGS-16466 0 0 1 0 
M ARS-BFGL-NGS-98142 0 1 0 1
M ARS-BFGL-NGS-114208 0 0 1 0</code></pre>
<p>Example 2:</p>
<pre><code>There can be some comments in the first lines
and some extra columns on the left hand side.
Column1 Column2 Column3 6415 2636
M NA dfdf 0/0 1/0
X NA sdfg 0/1 0/1
N NA fgjh 0/0 1/0</code></pre>
</div>
<div id="marker-map-format" class="section level4">
<h4>Marker Map Format</h4>
<p>For all functions reading from genotype files, a marker map must be
provided in argument <code>map</code>. This is a data frame or data
table with columns including</p>
<p><code>Name</code>: marker name</p>
<p><code>Chr</code>: chromosome number, and possibly</p>
<p><code>Mb</code>: position on the chromosome in Mega base pairs,
and</p>
<p><code>cM</code>: position in centiMorgan.</p>
<p>The order of the markers must be the same as in the genotype
files.</p>
</div>
</div>
<div id="example-data-set" class="section level3">
<h3>Example Data Set</h3>
<p>All evaluations are demonstrated at the example of cattle data
contained in the package. Breed names, years of birth, simulated
breeding values, simulated sexes, and herds are provided in data frame
<code>Cattle</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;optiSel&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">data</span>(Cattle)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>phen <span class="ot">&lt;-</span> Cattle</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="fu">head</span>(phen)</span></code></pre></div>
<pre><code>##           Indiv Born  Breed         BV    Sex herd
## Angler1 Angler1 1991 Angler -1.0706066   male &lt;NA&gt;
## Angler2 Angler2 1994 Angler -0.3362574 female    2
## Angler3 Angler3 1986 Angler -2.0735649 female    1
## Angler4 Angler4 1987 Angler  1.5968307   male &lt;NA&gt;
## Angler5 Angler5 1994 Angler  1.0023969   male &lt;NA&gt;
## Angler6 Angler6 1998 Angler -0.2426676   male &lt;NA&gt;</code></pre>
<p>The data frame contains information on 4 breeds:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">table</span>(phen<span class="sc">$</span>Breed)</span></code></pre></div>
<pre><code>## 
##    Angler Fleckvieh  Holstein   Rotbunt 
##       268       100       100       100</code></pre>
<p>The “Angler” is an endangered German cattle breed, which had been
upgraded with Red Holstein (also called “Rotbunt”). The Rotbunt cattle
are a subpopulation of the “Holstein” breed. The “Fleckvieh” or
Simmental breed is unrelated to the Angler.</p>
<p>The marker map is:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">data</span>(map)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">head</span>(map)</span></code></pre></div>
<pre><code>##                                    Name Chr Position cM       Mb
## ARS-BFGL-NGS-16466   ARS-BFGL-NGS-16466   1   267940  0 0.267940
## ARS-BFGL-NGS-98142   ARS-BFGL-NGS-98142   1   471078  0 0.471078
## ARS-BFGL-NGS-114208 ARS-BFGL-NGS-114208   1   533815  0 0.533815
## ARS-BFGL-NGS-65067   ARS-BFGL-NGS-65067   1   883895  0 0.883895
## ARS-BFGL-BAC-32722   ARS-BFGL-BAC-32722   1   929617  0 0.929617
## ARS-BFGL-BAC-34682   ARS-BFGL-BAC-34682   1   950841  0 0.950841</code></pre>
<p>This small example data set contains only genotypes from the first
parts of the first two chromosomes:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">tapply</span>(map<span class="sc">$</span>Mb, map<span class="sc">$</span>Chr, max)</span></code></pre></div>
<pre><code>##        1        2 
## 40.42745 33.79876</code></pre>
<p>Consequently the results obtained for specific individuals will be
rather inaccurate. The genotypes are included in the following
files:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>dir     <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package=</span><span class="st">&quot;optiSel&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>GTfiles <span class="ot">&lt;-</span> <span class="fu">file.path</span>(dir, <span class="fu">paste</span>(<span class="st">&quot;Chr&quot;</span>, <span class="fu">unique</span>(map<span class="sc">$</span>Chr), <span class="st">&quot;.phased&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>))</span></code></pre></div>
</div>
<div id="individual-specific-parameters" class="section level3">
<h3>Individual Specific Parameters</h3>
<div id="inbreeding-coefficients" class="section level4">
<h4>Inbreeding Coefficients</h4>
<p>The inbreeding coefficient of an individual is the probability that
two alleles chosen at random from the maternal and paternal haplotypes
belong to identical segments. It is calculated as the proportion of the
genome included in runs of homozygosity (ROH). This parameter estimates
the extent to which the individual may suffer from inbreeding depression
and predicts the homogeneity of its offspring. It can be calculated
with</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>Animal <span class="ot">&lt;-</span> <span class="fu">segInbreeding</span>(GTfiles, map, <span class="at">minSNP=</span><span class="dv">20</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">head</span>(Animal)</span></code></pre></div>
<pre><code>##           Indiv       Inbr
## Angler1 Angler1 0.04986142
## Angler2 Angler2 0.00000000
## Angler3 Angler3 0.00000000
## Angler4 Angler4 0.02484842
## Angler5 Angler5 0.00000000
## Angler6 Angler6 0.00000000</code></pre>
</div>
<div id="kinship" class="section level4">
<h4>Kinship</h4>
<p>The segment based kinship between two individuals is the probability
that two alleles randomly chosen from both individuals belong to
segments which are identical in both individuals. A matrix containing
the kinship between all pairs of individuals can be computed with
function <a href="https://rdrr.io/cran/optiSel/man/segIBD.html">segIBD</a>:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>sKin <span class="ot">&lt;-</span> <span class="fu">segIBD</span>(GTfiles, map, <span class="at">minSNP=</span><span class="dv">20</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>sKin[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##             Angler1     Angler2     Angler3
## Angler1 0.524930710 0.004341751 0.000000000
## Angler2 0.004341751 0.500000000 0.003575048
## Angler3 0.000000000 0.003575048 0.500000000</code></pre>
<p>The R code below displays the kinship between the female with ID
<code>Angler2</code> and all genotyped Angler males with breeding value
larger than 2.0.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>Males  <span class="ot">&lt;-</span> phen<span class="sc">$</span>Indiv[phen<span class="sc">$</span>Sex<span class="sc">==</span><span class="st">&quot;male&quot;</span> <span class="sc">&amp;</span> phen<span class="sc">$</span>Breed<span class="sc">==</span><span class="st">&quot;Angler&quot;</span> <span class="sc">&amp;</span> phen<span class="sc">$</span>BV<span class="sc">&gt;</span><span class="fl">2.0</span>]</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>sKin[Males, <span class="st">&quot;Angler2&quot;</span>, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span></code></pre></div>
<pre><code>##              Angler2
## Angler147 0.01469276
## Angler166 0.02370936
## Angler171 0.01384046
## Angler206 0.02430921
## Angler216 0.02241010</code></pre>
<p>In general, the males that have lowest kinship with the female should
be favoured for mating. In this case, however, all kinships are low, so
this criterion can be neglected.</p>
</div>
<div id="genetic-distances" class="section level4">
<h4>Genetic Distances</h4>
<p>There are several possibilities to compute a dissimilarity matrix
from a similarity matrix. One possibility which seems especially
suitable for multidimensional scaling is to define the dissimilarity of
individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> as <span class="math display">\[D_{ij}=(-log(b + (1-b)f_{ij}))^a,\]</span>
whereby the term <span class="math inline">\(b + (1-b)f_{ij}\)</span>
adjusts the kinship between individuals <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> for non-detectable ancestral
inbreeding, the function <span class="math inline">\(g(x)=(-log(x))^a\)</span> maps the adjusted
kinships from the interval [0,1] to positive real numbers, and parameter
<span class="math inline">\(a\)</span> may be chosen such that the
stress value is minmized. This can be done with function <a href="https://rdrr.io/cran/optiSel/man/sim2dis.html">sim2dis</a>,
whereby <code>b=baseF</code>:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>D     <span class="ot">&lt;-</span> <span class="fu">sim2dis</span>(sKin, <span class="at">a=</span><span class="fl">6.0</span>, <span class="at">baseF=</span><span class="fl">0.03</span>, <span class="at">method=</span><span class="dv">1</span>)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>color <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">Angler=</span><span class="st">&quot;red&quot;</span>, <span class="at">Rotbunt=</span><span class="st">&quot;green&quot;</span>, <span class="at">Fleckvieh=</span><span class="st">&quot;blue&quot;</span>, <span class="at">Holstein=</span><span class="st">&quot;black&quot;</span>)</span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>col   <span class="ot">&lt;-</span> color[phen[<span class="fu">rownames</span>(D), <span class="st">&quot;Breed&quot;</span>]]</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>Res   <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(D)</span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a><span class="fu">plot</span>(Res, <span class="at">pch=</span><span class="dv">18</span>, <span class="at">col=</span>col, <span class="at">main=</span><span class="st">&quot;Multidimensional Scaling&quot;</span>, <span class="at">cex=</span><span class="fl">0.5</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;&quot;</span>, <span class="at">asp=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAvVBMVEUAAAAAADoAAGYAAP8AOjoAOmYAOpAAZpAAZrYA/wA6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kLY6kNtmAABmOgBmOjpmkLZmkNtmtrZmtttmtv+QOgCQZgCQZjqQkDqQkGaQkLaQttuQtv+Q29uQ2/+2ZgC2Zjq2ZpC2kDq2kGa227a229u22/+2/9u2///bkDrbkGbbtmbbtpDb27bb////AAD/tmb/25D/27b//7b//9v///9jB0b3AAAACXBIWXMAAA7DAAAOwwHHb6hkAAALRUlEQVR4nO2di3bbuBVFKWucSm7dtEnteTRJp9O0djpt0miSWLJj/v9nFU8SAEkdAQQk0jl7dTkcEgTJ7YuLS5pSq5rspTr1CUwdCgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCBAt6Laqqmd3dmnlbXt8ozY9/vujXbT7nH30VkRw6H5uu8f/XFbV4sWH/Y0P6zlJ0OJGLHy9HhD08P3ZaQWJRcXi9d7GBQVVV2Lh/mJAkNQR7hOuyY9zuZvKMHjUiN9WiqA/qt7Vaazs1Yt4utLHVQbP/mdG27uL6vxtE0Gi1erzZbV4Vf8m1t+oc30nftUv79R1Lf4l/kOv/s0OEnstn3+oqu9+7LT7/L042vMb75pv5QHqBxFHKoQ+ia6+e9VtrPfwentUS97vM0XQny/EGHt8c/Z3KMgGuyNIc2l/wWbN+cf2Fy8H8KYJAXPhZoWzqNrtmkVPkLpAEeJXtQl5FfRBYyuo7U2f7uJyrKBXb8Th7i+e/dorSK/Qi+KUVnfy1+IIuqo/ibN4Jc/sSu39wVyKWHH+Qa+WkXYn29k+RQt9MX47sfF3d3p3f4gtfnxvzlecwl/uHq71GXiNG0G2N3O6t9VYQa9vRc+76moDBTXbWkFirf6pzlTtJi9JtNuoISFWr+SPc3OFXl67DdpJvsgpa9WTpM/f3tld6s2L953GjaCmN+d0xwnaiR5uq9dQkPixqmtnFpORodquzAhQmd5E+EaFudqoL1GlHG/C2QTt6sef9e4rP+9+urQD15v9/cZNDnKO+qx1mi5ocSMu66/Xi5udI8gJ81aQvoh9gnZVryBRKui1//Q72slfdyDy/B9frkNBQsaXXy6MCjvTho07gtzTHSdI/YLlKAOC3F/JYAQ1xYorSCj65VIdZG8E6eT11RdkknOn/AobF4wgU0RrQRs7nRyWg3xBVkdXkODxb80V+jmoabezCcRN0mLL4iex+PlCttZ77p6/7TTuCMqVg0SHOzVv6symJgBZdNiT3LSzrjMt9AvSNcvDdXCqauqR48xehj+LuRH07E4e2xtiTaEoDyO6eik7qV6HjbuCMs1iYm9xLNGtFtTk2avWSlMo3nbqoECQqYN07dOe6ruwvvHqoCCZh0naqb7q5hS6jbuCMtVBegBpTXI03IuE+uK/7SymytH3ppL+9UJUscM5SOSan4VgdV/pDTE1Dz0PK+mf6qCdnJjO326Ce71Htbeuu00lLfcMGncFqVP/04dxOegb4NadDynIQYzGl04JqqAgB1uVqcc5BgpyefjhInzSRkEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQECCzoGo2nEpQ3u7KQUEACgLMX9BWkK2zLvMXVBf18xQElYWCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgwKwFlb1N1cxZkH3QwccdQ1g/JQ0dWVD8s/ADYAQNcoQkNFtB0k3Zh62auQrSiafw82jJTAVZL+UNzVPQESLHMk9BfnYuqmtGgroWtoNb8jEfQTpMXBc2cBhBmiNNWwEzEqRo/RxJVSFB9juKBr8DNIPvOQvamG9Pq3d2YVR3B1DMVhFBj28aLZuB74qNE9STevwV5XJTEUFfr5uv9NsNDLLICAovfxsa+cYjKGTbPiub563Gxn4rZNYcFMSMXrWtyxoqNIvZrzse/LbqpDqop0rclp7u51QHbb1bi7Yg0qvmlKQzd+cWh/pnj43mtmPkiYXMoFBsZ6xt85jM3Ra0jTpPzBwKRXfKUll5G8ZUuzQPQQUKxUbPdjvkp0wWOnKhmP5nnwE/W6dBbJcHMY8IkjTjq7M+uqsYpl8o2sRsc3Tw2Kx0KT2DQtFx05Ol68K19PTroK191uqkoKBk/MafSYd+vE3FU1KhWUxmnl2WQtErEpsIGiycs8/25QSp+cuZ8NO6axOzP9G3466zA+pyuTzw2FEnGi3IqBk3zTtJ2f7jJuyk9LOciKD7CyVo5BPFTgnt5aO0/DwRQVkiSNIZWe6EnzDDL+NCqJAgWQOtapuux3UXGmom+pRsLOSo/x0sqdQ0L/9vYW6GC+kRERRO9VFoMzFBNIM6qO5IsqVR9FFbP09FkMnRVe8gS3u5YxnlZ+KCjAyR0LrTWOJbZstlnKFJC7KRIiKo9oOndu9YD/dj5TyZCHK1rNfhCEtI1cvlk4qgtUDmH+lGLW+3fr5OOHCkn0kLWiuqtXbjR1CqIFMEnb4OytKdNGM0WUYaWi5jB9mkBel7LRU+VaVc+XJG1YsHGpq4IGGlnedt/jGeUg9t5rEDDU1bkB5UInjcSsima0mCqcgxdmRBsX8XExFUyyxdmRiq9UP6toaOFRSdhKYfQWsdQevwramk+7HlMtbQtAV5U1i1VtmoHvXG9BOLoMaQiCI50mo9OP0XqaKO+9Sm+aoJH8G6zcy1/YPhQCA5Vx+YWEberk5cUO0I2qq5y0bQ1klJnb2cy29VtP8+mXsxJ4LWlbkzaybAplrsi6EwgtzHrE/nXqx2k7QcYurWo2XfGPPxn3FMIYIyvYLnTWJGkd/isLnMPmlt/uvUgrK9gucZUiNu2yiKfufOjLW4Px1O/gWq7iAzhg4uhLQTu6wGW8QjxSO/gpfQnV8rGj0xcePM6vFPO2YRQZWVU8lnH1KQEz2BqbDqWboR1NzGTyAH5XoFr3LCZ9vefg39RSO4dLcI6l/GJ3Bgu1O9gtcMrLX3wMxs7eahngjS/zrDLCpNz6YOsjdl+55wDF62neYjh1fEiZ7ycYeZwswwE7sOPW3dc+1NER379svEC8W6GV8qG6nbDHM/n/6i3RQiKONnNfxSWimLORMHE2CRY2zq07zvp+9m43CccXY4ky8Uq8ZPrf0kJ8PY9GyOn71hnfmzGpX1o54nrsd9/Vm8oqkXiirpVGsz1lSSjjkRh9hXywxHLhTjPw5lko57m5rEcpkwx9dTr4Mk5lH9uh6ZnlMSUD0HQbWNO7WYJCktPZtjZ29Y5/2sRl27s1d0FKWOrPbg2RvWWT+roRu3+Sc6gka40cfO3rDO/Ka9YlR+nmYE5fmsRh5GpKC5RNBIpico52c1xjJmDis3zef7rIa7U5rWUXl6FnWQ3SfpPmzkNFZSkHPPmqO7tAgaNb7qOQnK+iXmEYfN3rAlq6CDxtfYqrDvuNkbthw9gsbNVwOHzd6wJXsOgswsgo7Y3X4mdzd//O4AU6ukj99dOSgIQEGAkwmaDScSdPgBBtbHrc7Uyz4oCEBBAAoCUBCAggAUBKAgAAUBZnNvcCooCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQoIig+z+od2B3lXxfr3ehH7B5VN+plBD09Vq9JLyTbzPKc+4u9AM2j+o7mQKCdvqjCvotmdtVz0I/YPOovtPJL2hXXanXzPUr55uzj92F/h3B5lF9p1MkB+mL+P2NXuwu9O8FNo/qO51ygnRCED+7CwN77d88qu90KAjAIQbIKWhTVfozCsWSdHLf6ZSLoCLTfGrf6ZQTVKZQTO07mYKC5JDTZ9xd6AdsHtV3KrxZBVAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCEBBAAoCUBCAggAUBKAgAAUBKAhAQQAKAlAQgIIAFASgIAAFASgIQEEACgJQEICCABQEoCAABQEoCPB/xCYzbv8ZzRgAAAAASUVORK5CYII=" /><!-- --></p>
</div>
<div id="haplotype-frequencies" class="section level4">
<h4>Haplotype Frequencies</h4>
<p>Artificial selection and the substantial genetic drift in populations
with small effective sizes have increased the frequencies of various
haplotype segments in commercial breeds. Although these segments may
contribute to the economic value of a breed, their presences in an
endangered breed decreases the conservation value of the breed because
they are so common in the species that their conservation does not need
to be subsidized.</p>
<p>For individuals with genetic contributions from several breeds, each
marker belongs to a haplotype segment that originates from a specific
breed. Gene flow is usually from commercial breeds to the endangered
breeds but not vice-versa. Thus, if a sufficiently long segment from an
endangered breed can also be found in a commercial breed, then it can be
concluded that the segment is not native in the endangered breed.
Instead, the segment is assigned to the breed in which it has maximum
frequency. This can be done with function <a href="https://rdrr.io/cran/optiSel/man/haplofreq.html">haplofreq</a>.</p>
<p>Below, the frequency of each segment from haplotype 2 of Angler bull
“Angler1” in Rotbunt cattle is plotted with function <a href="https://rdrr.io/cran/optiSel/man/plot.HaploFreq.html">plot</a>
(red area).</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>Haplo <span class="ot">&lt;-</span> <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">refBreeds=</span><span class="st">&quot;Rotbunt&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">plot</span>(Haplo, <span class="at">ID=</span><span class="st">&quot;Angler1&quot;</span>, <span class="at">hap=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAEgCAMAAABb4lATAAAA8FBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZmYAZpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kJA6kLY6kNtmAABmADpmAGZmOgBmOjpmZgBmZmZmkJBmkLZmkNtmtrZmtttmtv+QOgCQOjqQOmaQZgCQZjqQkDqQkGaQkLaQtpCQttuQtv+Q29uQ2/+2ZgC2Zjq2ZpC2kDq2kGa2tma225C227a229u22/+2/7a2/9u2///bkDrbkGbbtmbbtpDbtrbb25Db27bb29vb/7bb/9vb////AAD/tmb/trb/25D/27b//7b//9v///9P3Bo+AAAACXBIWXMAAA7DAAAOwwHHb6hkAAASFklEQVR4nO2d/YPTthnHfcddm3C8DIoDbGMdCW1hsMasGy1wbum6LfNCkv//v5kevVmyZcWJ5Vh+7vn+cJfYsvRIHz+SLEtKsiOhVjK0AaR+RYCRiwAjFwFGLgKMXAQYuQgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXAQYuQgwchFg5OoKeLtILlfi/2RfGPPQ365bJ5EnUuftrzlGDjPFUa6ze+8rJ95Ok+RhkJRVDs8evN8fWChrWxrDAP789ABYcQBmxb+0ThRwLA2Sss5h+yxGDri1eaBTAXarBFzJYF4lfrxKwK3vmKEAf3rKjLy3FBb8+PskuXhfhvnEvt/60wrOgUPcESXGPGFuhobKj519qG8JsyTZ59dPk/P3ZhioK8+e8xyLbG9mvJzKIOyq79mXCxHNL3eYGc/1LcnTL7Nihy2NX0/5fx1pJp3aSqVqmzNhVw4FLZbGpCmPVnYv3oic/nxnb70eFnBRVmZZWbHJMPI+ZR/5ufOXIluSTHkl4wO6UHeoDVhEYYTRLlYBbATJjdhVUqkqVu0LCrBdIevqJ4P/ZaQScC0VyzZ3wo4cChM+TeGsI56G7OYtarYAgMsajH35cgX3Ycqzw27CX+Thy5V0gTwRJ5lR7MCc2z6xQsPJ9zISA2rCHY19vny/+5cZht1UkxXc4RXARpCc1w0iaRb80erzjAU007cA67AmYGG+Fam4j81Uqra5E27OoXKEajzu7ILpKyg0X70eFDDot7/fkZh4zQdFLsoosxyW/dFOP7dCywo21zdmBTCEM8OIeFlkNmAziLhKVIAiSP7gJyt9C7AOW8lgNVIGuJ5Ki4Sbc3j2fOfKYz27G3mHXvy0D1BYwNvv9EfJE4qBl5zyBF4uMrPC4MuVFZqZbVeSNmB+1Aij4q22wWY0uar6J1aHz0hfZkW0wcuKX6v8reqR1lNpkXBzDpOyajBD2dnlhknHSWSvpllB22BI8eIvv80MwOAgPIwqMtWnuuZZmMub0whdJLXsW52s5U639bIR1Dku403NIJVy1p1hI/0yK02AL97Y6YqA9VRaJOzIofBlUYHX42nK7uen4uhfPYCCAhatxGbW0oPhmlx1QEwPnltJOAAbYRwezM0wo2nwYCP9MiYHYGb8D6J5rEVaT6VFwo4cyso6t2sCFaopu+yWeMUaRNfju1ZQwIVqctI2bTDPzx1lr9kGNz9wys9mGKNRkqcL2VxO7KvMG6+498ZKv8yKEzCvRSe7eqT1VFok7Mih9OCFAdgI1ZRdYeC33m50aA++XDErK/3iei/abGZSYbbZi2Z9DVZbqZJ3ADbDiG5lpp4bUigmcYfpIGY5s+APoTqRt6LRBfUClrdgLdLagTYJ13Oo5Y7HnV3eL4d6+mQerDskk/I5WHSX7edg3r6IE/LuM0Krhz5N1QXYDGNcq3sjqRXEgpZpE430y6w0AWZRs6CuSO0DbRKu59BslR3xNGT3rd2UOxV2oAN60RdvVHf+I/v2oDqS9TW/6q0YtSrUw5URmtVU303VR5H9OmAzjDm0s2bdjgc/io5TGcSGxgeUuBll+mVWmgBDwU7qkdYPtEm4lkOpew3xWNn9YZrcEgN3PMak+hbEVm+vC1sNluqH0IMGp49PrzF9vBoUMGs+ZKBhAJfp49WAgHnDkrYN3T09X/p4NSzgs0etQ3dPz5c+XtGUHeQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXBEALqeer68cb/wKx5QjZ0DrlBGCfRRztcp3v63iLA/kqX1y83hpHlhPU//16rPbbDBuvluHWWpc0/CAWWnBP8i8qwC2iz/W16V6ANejAsBXfDJfOcW9RZybmXrBvHlyXQNsXTnl08Ear995AW9mc37DZf1MHhoesOFyrgJYX72rHz0K8K5QJd4mzjz5nTqQpzs/4KuXqe/6nRdwweqv7WLO//egIQGzukmsuxGL/Njnq3+oL/ruzye7jBffS17HsuBfLuZQUiLQ+t5TsUiCVXJTKKV/Xn0wo9oZgDWXNnH+ulI4ti+WIo6XapXgF1BFwz0CyUI8H78RtxTPkUijvF7aIP8WfObs9sUrc0IneHHlrgmlAQHzuun82vTg6QSOQLHl8n6GbBd82nwKp+CaPAEYMhBQBTFmxd1UtJZGVDJmkYbA0TpOBUhWCGZouIE+wj11fyUCFBP4K3Mk82c303Ius6yQtwv4qDsFufTiHjQgYO4yurvCi/H2UuFQ93PBV13P1Sn5tQwEx/nl91dvXz/cLtQpGV7GbANuGacCBMjgpBGaW83uqTyVAZmXVwz3dLJYEG5Krlw4F4t6epkAOCBg2fhU2mDV5xVVnVwrMVGncn2NDKRKcfP4+28+PPnAu0NGVDJmq4puG6cGDM6uY8014PX9/7xYatMfSvilI7oBQxpnArBqdXO5VAUbYIcHS8BWBwn+3l7KU7a37YxSzJ4xH3412XkAl1V2qzirVbQRWlj99jWvoUXA7FkrD4Z6oOLBuZobjg1wpQ2WdapsCuXRTK5sSeUpu700bwbeAt9Njcq0Cli1eW3jdAPWbTD0l1IdgD8redpgBRhOZ+DBsGJ6rjIOwtcGyz6nLAdWcc4VFd6lhoO6JT5/d6X7qV8I/xGBdCnyPrQ4YEa1MwY6yi5WqzhV1AZ2GforGfpJeQcIRzR60cqxjXtA2pY8WszZk7gaeBGLyTD2oo9VHyXhjTPvUHeuv3YedrgrwufgIwT1WRF4PdHeOJWnHqOfl87DDsBoR7IOU75nOWwvcXZxYbfqgMUjdXiNDTDpQBFg5CLAyEWAkSsM4HKXGOdBUhANCNgZ3Ugrh2jNJsBhFK3ZBDiMojWbAIdRtGYT4DCK1mwCHEbRmj1mwOarUzWByR7+7+ltqUMEuG10R0ctcGf69dlaz190BgwsAtw2uo6AS3oEmGvMgNfltONdZYYa38Ue5rjwKRB6+gwE1hM/gupGA4YtdfmkxKYGsQNgMe1YftvxqVPGrC05iUnPu0yN6VxhdZMB53zz8Ym5zMcT3YGAjYnMaqmYMe+SV9HZRAEuJ1AH1w0GvF2kamFe3jCBqEsVbc6S43PozJnTcp6pPcORAIe9is/zFNVo0xymYIDVDEnbg1MCfPRVtZ8CqeuUHswfk1QbfHvJ22Dm1XxW8vm1UVMHF1bA4pcD/LPLdBvMUe8zQn1u8XrTBXgzm8h5xjDPWU0kZkZ+9UQFhhMH53if8ALe7WfcVy86JkVrdpg2OK8NEh4dXbQl5Ve0ZgcAXHDfhOYuhBHRlpRf0ZrdFTD8RoUg22LpQB62Fx2TojW7cy+6w7Cfs+cUbUn5Fa3ZYx6LjknRmt0ZcMZ/+rDbMhwC3J+6AhbvX+Hps1nq1ycbG2kC3J86t8FyhxJPDytX/t3o6AS4P3UErAanmvrHO2v8KvBQZUyK1uyuVXTOh6fEvotuKSffBX/ZEJOiNbtzJ8vYvcgt8uBBdYoX/moMmtrgAXSK52D5U+PNPxpPgPtTV8B7H4EONSLakvIrWrM7PweH2LyGAPenzs/BIbaSIcD9KdBARzgjoi0pv6I1u/NAx74pWYcaEW1J+RWt2V3b4CLE/CYC3J+6z6qkXjQoWrPpfXAYRWs2AQ6jaM3uDJhV0perjtuOE+D+1LmTdbbMYTFBJ8IEuD8FeB8M74g874M9sdCku/4VYKADAHfbbpsA96dAHpx12lmeAPenMG1w3m24gwD3pxC96M775RPg/kTPwWEUrdkEOIyiNZvGosMoWrPDeHDHHyUiwP0pUBWddXotTID7UyDALVxY/SjznuiiLSm/ojU7EGDPUKVupptbagLcn8IA9q4ulPPdyYMHUaBetHekcjOD0wR4EJ3mOTiDDegI8BA60UBHnqQEeBCFGujYN9axnt4iwEMozLTZFm+TfBsIEuD+1Pl9sKDWtPL3cCOiLSm/ojW7axUtf8W+3VglbYR2egXy4KNmdNCcrBMowIwO9jfv9safAPenMDM6OrXABLhPneI5mDZCG1AnAEwboQ2pEFW0f+kKbaM0qMJMm/UtXaGN0AZVoInvtJXh0AY0qfNY9P6lK7QR2pAK5MHegQ7aCG1AhWmDaelKtGaHGeigpSvRmn2iF/7to4u2pPyK1uwAbXBYI+oGRVt2pqI1MkAvOqwRlkG1d03RKlojO3eyet2MNPkvU7RlZypaI0PNyepn8RkB7qy4O1kccHjCZa0frP4nwG2jOylgFj98DpAASsBheli7/YA7lZ7jYhkha1lE9ATYfRUHHOJBaQ/go12Yd8FdgEWMMvYgVQQBdsWyd9LdsYBltE3uSYDbXRWvB2t05tWJ0fTqG4AAe64aC2BZR5QVhoqWAPuvGglgTbl2jAD7rzoZ4IMJ1AEbOB2nOucAK+CWSwsPMSIEYCdLAtz7Va2iixXwnjAEuG10vQH2qEX0+wZDCHDb6NyAD0uwL8D1p3a32TFpHIAPdOHjATcBFEMjnqHxGw34+LVJiYNB7WKHjgXseD2pY9TxEuCKOqxNagH4UJjNgEuA0ov5P8cj9b4sRKX+AXdZ2eABfKy3OgHzlxLGNzn8JT/XQtsGxD2vqH/AXdYmOQCr3k4ArhVsh4St3Vk3GHBgDw7ltYF1gwF3WZtUBxwhW66bDLjD2iQDcMAmtw/daMAHRdcAOHYR4LbREeCwOinggzdCwwQ40OPUwW9OT5JKQyyuh0jXwBSpgwYETIpWBBi5Ar9sIMWmwC8bSLEp8FAlKTYFftlAik3kwcgV+GUDKTYFftlAik30HIxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXAQYuQgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLlGApjPuIeJQs696ddXkczx9JjJTw0wk20cgMUkv+xs6QS8mUUyiddj5nbBjubJ5OQ2jQLwesqLa7uAH7CuAS5iWUrjM1Oc8/w4c18aBWA1C/t/18w1nk2TZL7bPH4lsTK3iWQavtdMrqLjb38eoTEA3i7Kmm0zg6rubLmZGVN34wC810xWe5MHu7SZpfZnVt+ZxyIBvNfMQdYLjA/wnP+xGrkYAbvMHKCPNQrAdt0XLeB9Zg6z3mcMgHXvpUjmEQPeY2Y+zHquUQCuPn9ECthrpl64d2KNArCo3fhYQcyAfWaupwOtxxwHYLGsUY8BxgrYY2YulmXSczApsAgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXAQYuQgwchFg5Bov4O1Crscsaj/EZq+rt5SJNUKuldh8HVExd+/UpAMYkqsJ49aoAQuw2UGAedjK1hnmlQcwI8C9arv44vGS/d88+cPBgBu3uyHA8Wi7mGTAsbj8FqBB3QtLNmHfog8AmK+oz8X2cuVuRhIwZ1PIvefWU34tgwufLv/NIcuTm9mfZ4lcuw30ze/rKeyVBLut3H49HWQbuxYaNeCCk00BWsZa1ZyvvGZfwIP5ino4AkuvyyrZ8OCCQ50o2HPBT7bB6qTaDwkuEifL7+upjJ7dF3N+RYQaNWDYo3LzeMmgbaC2BlK8dmZ/eHGLqro4vy7rbNUGT+TPXTPSavm4CVifVPshlQHK72LVfn5+LT9E+aO7owa8XTDHuVwpr5T1LJC4m0hAO7ldlfIu2YvWlPhJQdgEbJyc65ZZu7f8LgKxRPSHkxZAO40a8C6fsBqaeyVrbM/fTTWAs9e3l6IdBZmAtZsJKsJdVRtcAaz2YGgCDGnAX/GBAIcVAF5ffXyxBGgcyLoELKjrIm8GrHrCGdTGR3swAe5BAHi7eMaaYQaNt6NFYgDWtEEuwLqZVSEa2mAfYNn0ig8ZtcFBxTeWy2BMSnrwZpakJWC+8Sfv7krnFDIoqI4yZ1woD06rvWgfYNkX1x8i1MgB81KVbfDZMtMel4qeMn8OPr92enD5HFyI/Y14oKz2HOwDLPYMLj9EqPECjkgxD2kR4AAiwMhFgEmDiQAjFwFGLgKMXAQYuQgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5/g91EZ1GxiuYeAAAAABJRU5ErkJggg==" /><!-- --></p>
<p>It can be concluded that the first chromosome originates from Rotbunt
cattle or from a closely related breed. In contrast, the second
chromosome does not originate from Rotbunt cattle.</p>
<p>This evaluation can be done simulateneously for several reference
breeds. Below, the frequencies of each Angler haplotype segment within
Rotbunt, Holstein, and Fleckvieh are computed, the results are combined
into the single R object <code>Haplo</code> with function <a href="https://rdrr.io/cran/optiSel/man/freqlist.html">freqlist</a>, and
plotted with function <a href="https://rdrr.io/cran/optiSel/man/plot.HaploFreq.html">plot</a>.
The red area shows the frequency of each haplotype segment in Rotbunt
cattle, whereas the black line shows the <em>maximum</em> frequency the
segment has in one of the evaluated reference breeds.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>Haplo <span class="ot">&lt;-</span> <span class="fu">freqlist</span>(</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">refBreeds=</span><span class="st">&quot;Rotbunt&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>),</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">refBreeds=</span><span class="st">&quot;Holstein&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>),</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>  <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">refBreeds=</span><span class="st">&quot;Fleckvieh&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>  )</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a><span class="fu">plot</span>(Haplo, <span class="at">ID=</span><span class="dv">1</span>, <span class="at">hap=</span><span class="dv">2</span>, <span class="at">refBreed=</span><span class="st">&quot;Rotbunt&quot;</span>)</span></code></pre></div>
<p><img role="img" src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAEgCAMAAABb4lATAAAA8FBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZmYAZpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZpA6ZrY6kJA6kLY6kNtmAABmADpmAGZmOgBmOjpmZgBmZmZmkJBmkLZmkNtmtrZmtttmtv+QOgCQOjqQOmaQZgCQZjqQkDqQkGaQkLaQtpCQttuQtv+Q29uQ2/+2ZgC2Zjq2ZpC2kDq2kGa2tma225C227a229u22/+2/7a2/9u2///bkDrbkGbbtmbbtpDbtrbb25Db27bb29vb/7bb/9vb////AAD/tmb/trb/25D/27b//7b//9v///9P3Bo+AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAT0ElEQVR4nO2dC5vbxBWGtcsu2NmQNAE7SduUYgdImhQrpQWSrIBCi1GNrf//bzpn7iONZFkaWaOT8z1PsrY0lzPzai6SzoyTgoRaydgGkIYVAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXAQYuQgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRqy/gwzq53oq/s2Nh7EP/uG2dRZZIXbaP00UeM8VRrot7b0onXs+T5GGQnFUJLx68OR5YKG1bG+MA/v3JCbDiAMyqf+OcyOHYIkjOuoTtixg54Nbmgc4F2C8DuFTArEy8uwzg1lfMWIB/esKMvLcRFnz3xyS5emPC/MS+f/CXLZyDBnFX1BhrCSs7NHR+7OxDfUnYNck+v3ySXL6xw0BfefE5L7Eo9n7J68kEYbG+Zl+uRDI/3mVmfK4vSZ6/KYob1hi/m/O/OtFUNmonl7Jt3ox9JRS0WB6zujI6xb16JUr6w92j/XpYwLnpzFLTsckw8jplH/m5y+eiWJKMicn4gK7UFeoCFklYYXQTKwG2gmRW6iqrhapW3RYUYLdD1t1PCn9NohJwJRfHNn/GnhIKE36aw1lPOjXFzVr0bAEAmx6MffloC9fhgheHXYQ/ysPXW9kEskScZEaxAytu+8wJDSffyEQsqAlvaOzz9ZviP3YYdlHNtnCFlwBbQTLeN4isWfBPt78vWUA7fwewDmsDFuY7iYrr2M6lbJs/4/oSqoZQTsdfXDB9C5XW1K8HBQz65Z93JSbe80GVizpKnQbL/tONfuWElh1spi/MEmAIZ4cR6bLEXMB2EBFLdIAiSPbgeyd/B7AOWypgOVEGuJpLi4zrS3jxeeErY7W4e3mFXn1/DFBYwIev9EfJE6qB15xqCbxeZGGFwddbJzQz2+0kXcD8qBVGpVseg+1kMtX1z5wJn5W/LIoYgzeldq3Kt60mWs2lRcb1JUxM12CHcovLDZMNJ5GzmnoFHYMhx6u//bK0AEMD4WFUlak51S0vwkpenFboPKkU35lkbQo91stBUJfYpLuwg5TqWU+GrfxNUeoAX71y8xUBq7m0yNhTQtGWRQdeTaeuuL8/EUf/3gAoKGAxSuyXLVswxMnUBMRuwSsnCw9gK4ynBXMz7GRqWrCVv0nJA5gZ/40YHiuJVnNpkbGnhLKzztyeQIWqKy67JF6wAdF3+64VFHCuhpxFmzGYl+eustceg+tvOOVnO4w1KMnTuRwuZ24s+8LL771y8jdF8QLmveisqCZazaVFxp4Syha8tgBboeqKKwz8snEaHboFX2+ZlaV5cXUWbQ8zC2G2PYtmcw3WW6ma9wC2w4hpZaruGxZQTeIK00HsembBH0J3Ii9FawraCFhegpVEKwfaZFwtoZY/HX9x+bwc+umztWA9IZmZ+2AxXXbvg/n4Ik7Iq88KrW76NFUfYDuMFVfPRhZOEAdaqk208jdFqQPMkmZBfYm6B9pkXC2hPSp70qkp7mt3KPcq7IMOmEVfvVLT+Xfs24Pyk6zPeKzX4qlVrm6urNCsp/pqrj6K4lcB22HsRzs7Nu148J2YOJkgLjT+QImbYfI3RakDDBU7qyZaPdAm40oJpe7VpOMU95t58oF4cMdTTMpvQVwN9rqw1cNSfRN60sPp7vnV5o9XowJmw4cMNA5gkz9ejQiYDyyLtqH759eUP16NC/ji09ah++fXlD9ekcsOchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUauCAAb1/PdjeeNX+5xOfIGdE5ZIdhH4atl3v22StMcyBbuyf2jjX1gN180x1ef/WaDcatiF2apcUXjA2a1BX+g8L4KOKz/XF2X2gC4mhQAvuHOfMbFvUWa+6V6wbx/fFsB7MScc3ew2vhFI+D9csUvuHQY56HxAVtNzlcBu5tvq0c7AS5yVeNt0sySP6gD2aJoBnzzfNEUv2gEnLP+67Be8b8DaEzArG8S627EIj/2+eZf6ou++rNZkfLqe877WBb8o/UKakoE2t17IhZJsE5uDrX0881bO6nCAqy5tEnz31uF4/BsI9J4rlYJfghdNFwjkC2k8+4LcUnxEok8THxpg/w/556zh2cvbIdOaMWlqyaURgTM+6bLW7sFz2dwBKotk9czFDvnbvMLOAVxsgRgyEBAFcSY5R8vxGhpJSVTFnkIHK3TVIBkh2CHhgvoHVxT97ciQD6D/2WJZPncYVr6MssO+bCGj3pSkMlWPIBGBMybjJ6u8Gq8s1E41PWc81XXK3VKfjWB4DiPfn/7+uXDw1qdkuFlyi7glmkqQIAMTlqhudXsmsoWMiBr5SXDGyZZLAg3JVNNOBOLegZxABwRsBx8SmOwmvOKrk6ulZipU5mOIwOpWtw/+vqLt4/f8umQlZRM2emi26apAUNj16lmGvDu/q/PNtr0hxK+aYh+wJDHhQCsRt1MLlXBBtjTgiVgZ4IE/9/ZyFNuayusWkyfsjb8YlY0ADZddqs0y120FVpY/fol76FFwPRpqxYM/UCpBWfKNxwb4NIYLPtUORTKo6lc2bKQp9zx0r4Y+Aj88cLqTMuA1ZjXNk0/YD0Gw3xpoQPwe6WGMVgBhtMptGBYMb1SBQfhG4PlnFPWA+s4V4oKn1LDQT0SX357o+epH4r2IwLpWuRzaHHATqqwHnSYKVarNFXSFnYZ+hMZ+rG5AkRDtGbRqmFb14C0Lfl0vWJ34urBi1hMhnEW3VVD1ERjmlmPvnP3mfewp7kivA/uIOjP8sDriY6mqVpqF/2w8R72AEb7JOs0ZUeWww6SZp8m7FcVsLilDq+pASadKAKMXAQYuQgwcoUBbHaJ8R4kBdGIgL3JTbRziNZsAhxG0ZpNgMMoWrMJcBhFazYBDqNozSbAYRSt2WgB2+9VlXeT+27g6KtUx2OmkqR70DJbelKtKsHtDI+kGk5oAbsStZjqd2s77dzoDWh/9qGoQrEB39z6A/mvKAIcQj5WBDh0rFbJnZD0Yf10nszEj1cIx+GMb6eunBaVT3JRcl/jW9yDAwz3j9C+NRBYe4V4IhXC4UZFV+7QTj4mWqG8TORu8slKOKpzz3ZmuT5p+WIPoXMAhi11uVNiXTE6AwYXpevt7s5Gr+SYWe4u0idZfiu4X5Xl0iU9nBRgEdhpwY5bMj8nnaNVB5DxjxDVGt9FtJnyE8uEH5X4lgvXyN39ny1/6llwbwRLZwCc8c3HZ/Yyn4bkTgK80v8K4UDDWqUcZy03ukJV+sJxyuTR0pkCbLyrlYxTo46kXN6UG5jwhGVRk7JPpx1R+tWwdguuctwz2krVWDmEhgd8WC/UwrysxoEoAGDhOAxdtewmLIe3QvWzK8etWjqhuu6PfqdGEwk2m12pblu6Q8O3GsAiTKYAX38zAy+MX4XDu/GnnjRgftGLPqiuJ+oPWDoOs38v3fZj+zpK90m3BS9aALYiiVKUvC/rAcsYCvCKu+mBZ/TO8bONF3Dlp0CqOksL1o7DC5WJBzDcJqnh9M6Gj8GsVXOXZc1M+RubSLsbawyWrnYiuvKLrQcsw8gVL2o1YC6mdiXH2oHUuwVnybH9sfUYzFEfM0J9bvF60wIsHYdTvtyAX3I+wHBKTIjBCVp5GbMSfPJYBYYTFbfk8iyaRzfu0AqwE03PouHSNrPow3rG/S3NydgBF8cZDzWLHkw1bslNSrpFG15hxuCs8hywc3IxAK5xS25S0i3a8AoAOE9E/9Pdxz42wB0Urdl9AcNtgyDb4mY9CzuLjknRmt17Ft1jxYB35hRtTTUrWrOn/CzanXt2eiMYTHgBp/LGLpQRnWtK3+pa34++MAomtIBFhcp7zxqpX5+sHaQDAjb0CDBX7zFY7lDSMMPKVPuubejdu+i+bwSDCStg9XCqbn5cOM+vAj+qDPBGMJiwAoZtKopCvWT3SjXyIvjLhgBvBIMJLeDC2r3Ir2FbcM83gsGEF/BxZeoZ9CBjcL83gsH0PgNWPzVe/6Px4QCf+kYwmNACPnoLdKoR6vNvNbKC9nkjGFpoAach3MXoUeVw6n0fHGIrGQI8nAI96AhnRLQ11axoze79oOOYS9apRkRbU82K1uy+Y3AeYsJCgIdTf6/KgWbRE1O0Zk/5fXBMitZsAhxG0ZrdGzDrpK+3PbcdJ8DDqfcki6+86Xk7TICHU4D3wfCOqOF9cEMq5HQ3vAI86ADA/Ra4EuDhFKgFp712lifAwynMGJz1e9xBgIdTiFl07/3yCfBwovvgMIrWbAIcRtGaTc+iwyhas8O04J7bABHg4RSoi057vRYmwMMpEOAWTVj9KPOR5KKtqWZFa3YgwA2PKvUwXT9SE+DhFAZw4+pC6e9OLXgUBZpFNz6phP0kCfBIOs99cMp3gSXAI+hMDzqyZEGAR1GoBx3HnnXs5h8Q4DEUxm22xdukphVBBHg49X4fLKjVrfw93Yhoa6pZ0Zrdt4uWv2Lf7lklbYR2fgVqwZ08Osgn6wwK4NHB/s/6vfEnwMMpjEdHrxGYAA+pc9wHn2cjtHEVrdlnADzYRmgxKVqzQ3TRzUtXhttGKSZFa3YYt9mmpSvDbYQWk6I1O5Dj+yhbGcakaM3u/Sz6+NKVwTZCi0nRmh2oBTc+6BhqI7SYFK3ZYcZgWroSrdlhHnTQ0pVozT7TC//2yUVbU82K1uwAY3BYI6oGdTDR/im80ssMcfr0JI/lGDzFQAowiw5rhGNQ5V3TyQkm/HMpGQLcPlaQ366uBwz7y/YCbGBKxF2vmfY5xqVQPlnDLD4LANgcs949E+BBYrVKrgq4A2E/YP3RJBmsLRPgtskNB9h81Emy9AFxgOKgBBxmhlUcB3yykXaE36rHOGDeX4vkCbA/Fgcc4kbpCODTm7CMwH8CwA8YqOqfCSDA/lh9AR93uusM2PMrDw7g3+QfAtwUK9oW7PDVhJ1PjgiwP9ZUAEvIdb/jQoDrYk0JcKMIsD/W2QCfSoAAG/UD3HJp4SlGEOCwmsKDjigAHwlDgNsmFyngYw9DCHDb5PyAT8twKMDVu3a/2TFpGoBPbMLdAdcBTHiYhkfj7zXg7muTEg+DSmSPugL2vJ7UKep0YwLseRRYDtEt3VMC91ib1ALwqTDrARuAxjfAl35UgI/mPjzgPisbGgB3ba1ewNx3y/omDiXycyW0a8BAPiKtFAPgPmuTPIDVbCcA1xK2U8JWrqz3GHDgFhyq1QbWewy4z9qkKuAI2XK9z4B7rE2yAAcccofQOICT0t+GIN0SDqQWgGMXAW6bHAE+QbEBbr0RmrrxmDTg0v1Tr9spE9VtAJ4KbKjbE/LrFKuSitcnixRWIwImRSsCjFyBXzaQYlPglw2k2BT4USUpNgV+2UCKTdSCkSvwywZSbAr8soEUm+g+GLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXAQYuQgwck0EMPe4B0ch7970u5tIfDwbzOSnRvBkmwZg4eSXXmy8gPfLSJx4G8w8rNnRLJmd3aZJAN7NeXUd1vAD1hXAeSxLaZrMFOcafpx5KE0CsPLC/t8taxpP50myKvaPXkisrNlE4obfaCZX3vO3PztoCoAPa9Oz7ZfQ1V1s9kvLdTcOwEfNZL03tWCf9suF+5n1d/axSAAfNXOU9QLTA7zi/zmDXIyAfWaOMMeaBGC374sW8DEzx1nvMwXAevaSJ6uIAR8xMxtnPdckAJfvPyIF3GimXrh3Zk0CsOjd+LOCmAE3mbmbj7QecxqAxbJG/QwwVsANZmZiWSbdB5MCiwAjFwFGLgKMXAQYuQgwchFg5CLAyEWAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGrukCPqzlesy88kNs7rp6R6lYI+Rbic3XEeUr/05NOoAluZowbk0asACbngSYhy1tnWHHPIEZAR5Uh/WHjzbs7/7xn04GXLvdDQGOR4f1LAWO+fWXAA36XliyCfsWvQXAfEV9JraXM7sZScCcTS73ntvNeVwGFz5d/5dDlif3y78uE7l2G+jb33dz2CsJdlu583I+yjZ2LTRpwDknuwBoKRtVM77ymn2BFsxX1MMRWHptumSrBecc6kzBXgl+cgxWJ9V+SBBJnDTfd3OZPLsuVjxGhJo0YNijcv9ow6DtobcGUrx3Zv/x6hZddX55a/psNQbP5M9dM9Jq+bgNWJ9U+yGZAOa7WLWfXd7KD1H+6O6kAR/WrOFcb1WrlP0skPg4kYAKuV2Val1yFq0p8ZOCsA3YOrnSI7Nu3vK7CMQy0R/OWgHtNGnARTZjPTRvlWywvfx2rgFcvLyzEeMoyAasm5mgIpqrGoNLgNUeDHWAIQ/4X3wgwGEFgHc3755tABoHsjOABXVd5fWA1Uw4hd64cwsmwAMIAB/WT9kwzKDxcTRPLMCaNsgHWA+zKkTNGNwEWA694kNKY3BQ8Y3lUngmJVvwfpksDGC+8Sef7srGKWRRUBNlzjhXLXhRnkU3AZZzcf0hQk0cMK9VOQZfbFLd4hZipszvgy9vvS3Y3AfnYn8jHiit3Ac3ARZ7BpsPEWq6gCNSzI+0CHAAEWDkIsCk0USAkYsAIxcBRi4CjFwEGLkIMHIRYOQiwMhFgJGLACMXAUYuAoxcBBi5CDByEWDkIsDIRYCRiwAjFwFGLgKMXP8HuSLojbdO2fIAAAAASUVORK5CYII=" /><!-- --></p>
<p>Hence, most segments from Chromosome 2 have very low frequency in
other breeds, so they can be classified to be native for Angler.</p>
<p>The classification of haplotype segments can be done with a single
call to function <a href="https://rdrr.io/cran/optiSel/man/haplofreq.html">haplofreq</a>. In
this case, argument <code>refBreeds</code> is either the vector with
breeds to be used as reference breeds, or the default
<code>refBreeds=&quot;others&quot;</code> is used, in which case all breeds with
genotypes are used as reference breeds, except
<code>thisBreed</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>Haplo <span class="ot">&lt;-</span> <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">refBreeds=</span><span class="st">&quot;others&quot;</span>, <span class="at">minL=</span><span class="fl">2.5</span>)</span></code></pre></div>
<p>The result is a list. Component <code>freq</code> is a matrix
containing the maximum frequency each haplotype segment has in one of
the reference breeds.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>Haplo<span class="sc">$</span>freq[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##                     Angler1 Angler1 Angler2
## ARS-BFGL-NGS-16466     0.25       0       0
## ARS-BFGL-NGS-98142     0.25       0       0
## ARS-BFGL-NGS-114208    0.25       0       0
## ARS-BFGL-NGS-65067     0.25       0       0
## ARS-BFGL-BAC-32722     0.25       0       0
## ARS-BFGL-BAC-34682     0.25       0       0
## ARS-BFGL-NGS-3964      0.25       0       0
## ARS-BFGL-NGS-98203     0.25       0       0
## ARS-BFGL-BAC-2376      0.25       0       0
## ARS-BFGL-BAC-31722     0.25       0       0</code></pre>
<p>Component <code>match</code> is a matrix containing for each segment
the first letter of the name of the breed in which the segment has
maximum frequency. If the frequency of the segment is smaller than
<code>ubFreq=0.01</code> in all reference breeds, then the segment is
classified to be native and coded as <code>1</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>Haplo<span class="sc">$</span>match[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##                     Angler1 Angler1 Angler2
## ARS-BFGL-NGS-16466  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-NGS-98142  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-NGS-114208 &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-NGS-65067  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-BAC-32722  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-BAC-34682  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-NGS-3964   &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-NGS-98203  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-BAC-2376   &quot;R&quot;     &quot;1&quot;     &quot;1&quot;    
## ARS-BFGL-BAC-31722  &quot;R&quot;     &quot;1&quot;     &quot;1&quot;</code></pre>
<p>If individuals are genotyped for many markers, then the working
memory could become a limitation. This can be avoided by writing the
results to files. Results will be written to files if argument
<code>w.dir</code> is defined as the name of a directory. In this case
function <a href="https://rdrr.io/cran/optiSel/man/haplofreq.html">haplofreq</a>
returns a data frame with file names:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>wdir  <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">&quot;HaplotypeEval&quot;</span>)</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>wfile <span class="ot">&lt;-</span> <span class="fu">haplofreq</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>, <span class="at">w.dir=</span>wdir)</span></code></pre></div>
</div>
<div id="breed-composition" class="section level4">
<h4>Breed Composition</h4>
<p>Mating decisions should not only depend on the breeding value of the
male and the kinship between male and female, but also on the genetic
contribution of the male from foreign breeds. Many endangered breeds
have been graded up with commercial high-yielding breeds. These
increasing contributions from other breeds displace the original genetic
background of the endangered breed, decrease the genetic contribution
from native ancestors, and reduce the conservation value of the breed.
The breed composition of individuals can be estimated with function <a href="https://rdrr.io/cran/optiSel/man/segBreedComp.html">segBreedComp</a>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>Comp  <span class="ot">&lt;-</span> <span class="fu">segBreedComp</span>(Haplo<span class="sc">$</span>match, map)</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a><span class="fu">head</span>(Comp[,<span class="sc">-</span><span class="dv">1</span>])</span></code></pre></div>
<pre><code>##            native            F          H          R
## Angler1 0.6820668 0.0000000000 0.06420559 0.25372765
## Angler2 0.5952110 0.0392385937 0.23581096 0.12973943
## Angler3 0.8953642 0.0647239060 0.02188978 0.01802207
## Angler4 0.5127613 0.0004341256 0.18018789 0.30661670
## Angler5 0.3338797 0.0031128187 0.23518540 0.42782203
## Angler6 0.3158735 0.0000000000 0.24816876 0.43595772</code></pre>
<p>The average breed composition of Angler cattle is</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>Average <span class="ot">&lt;-</span> <span class="fu">apply</span>(Comp[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">2</span>,mean)</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a><span class="fu">round</span>(Average, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## native      F      H      R 
##  0.477  0.020  0.237  0.266</code></pre>
<p>Since Red Holstein is a subpopulation of Holstein cattle, their
contributions should be added. Thus, the average contribution of Angler
cattle from Holstein is 0.503, the contribution from Fleckvieh is only
0.02, and the native contribution is 0.477. This is in good accordance
with pedigree-based results</p>
</div>
<div id="kinship-at-native-segments" class="section level4">
<h4>Kinship at Native Segments</h4>
<p>Since animals with high native contributions tend to be related, the
inbreeding level could increase considerably when introgressed genetic
material is removed from the population. This could be avoided by
restricting the increase in kinship at native haplotype segments in the
population. Matrix <code>sKinatN$of</code> containing the kinships of
individuals at native haplotype segments can be calculated from the
results of function <a href="https://rdrr.io/cran/optiSel/man/segIBDatN.html">segIBDatN</a>:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>sKinatN <span class="ot">&lt;-</span> <span class="fu">segIBDatN</span>(GTfiles, phen, map, <span class="at">thisBreed=</span><span class="st">&quot;Angler&quot;</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>sKinatN<span class="sc">$</span>of <span class="ot">&lt;-</span> sKinatN<span class="sc">$</span>Q1<span class="sc">/</span>sKinatN<span class="sc">$</span>Q2</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>sKinatN<span class="sc">$</span>of[<span class="st">&quot;Angler2&quot;</span>, <span class="st">&quot;Angler4&quot;</span>]</span></code></pre></div>
<pre><code>## [1] 0.09438978</code></pre>
</div>
</div>
<div id="population-specific-parameters" class="section level3">
<h3>Population Specific Parameters</h3>
<div id="genetic-diversity" class="section level4">
<h4>Genetic Diversity</h4>
<p>The genetic diversity of a population is the probability that two
alleles chosen at random from the population do not belong to identical
segments. It is one minus the average segment based kinship of the
individuals. Thus, it can be computed as</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> phen<span class="sc">$</span>Indiv[phen<span class="sc">$</span>Breed<span class="sc">==</span><span class="st">&quot;Angler&quot;</span>]</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">mean</span>(sKin[keep, keep])</span></code></pre></div>
<pre><code>## [1] 0.9431438</code></pre>
<p>The diversity of this population is high due to historic
introgression with other breeds.</p>
</div>
<div id="kinship-and-diversity-at-native-segments" class="section level4">
<h4>Kinship and Diversity at Native Segments</h4>
<p>The kinship at native segments in the population is the conditional
probability that two alleles chosen at random from the population belong
to identical segments, given that the segments originate from native
founders. The kinship at native segments is</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>sKinatN<span class="sc">$</span>mean</span></code></pre></div>
<pre><code>## [1] 0.06695171</code></pre>
<p>The genetic diversity at native segments is one minus the kinship at
native segments. Thus, it can be calculated as</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> sKinatN<span class="sc">$</span>mean</span></code></pre></div>
<pre><code>## [1] 0.9330483</code></pre>
<p>The diversity at native segments is high. This could have several
reasons:</p>
<ul>
<li><p>breeds that have been used for introgression are missing in the
data set, so contributions from these breeds were wrongly classified as
native and contribute to the diversity at native segments,</p></li>
<li><p>the minimum lentgh of haplotype segments is too high or the
marker density is too low, so that short introgressed segments cannot be
classified to be non-native.</p></li>
<li><p>the diversity at native segments is indeed high.</p></li>
</ul>
<p>In this case, an introgressed breed, the Norwegian Red, is missing in
the data set. A high diversity at native segments is important if a goal
of the breeding program is to remove introgressed genetic material from
the population. Without maintenance of a high diversity at native
segments, inbreeding coefficients will soon rise to an unreasonable
level.</p>
</div>
</div>
<div id="multi-breed-specific-parameters" class="section level3">
<h3>Multi-Breed Specific Parameters</h3>
<p>Most evaluations for multiple breeds with segment based methods
require high density marker genotypes that enable the detection of short
haplotype segments that originate from common ancestors who lived before
breed separation. Hence, the examples shown below are only
illustrative.</p>
<div id="kinships-within-and-between-breeds" class="section level4">
<h4>Kinships Within and Between Breeds</h4>
<p>Average segment based kinships between and within breeds can be
computed with function <a href="https://rdrr.io/cran/optiSel/man/opticomp.html">opticomp</a>:</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>sKin  <span class="ot">&lt;-</span> <span class="fu">segIBD</span>(GTfiles, map, <span class="at">minSNP=</span><span class="dv">20</span>, <span class="at">minL=</span><span class="fl">1.0</span>)</span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>CoreSet <span class="ot">&lt;-</span> <span class="fu">opticomp</span>(sKin, phen)</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="fu">round</span>(CoreSet<span class="sc">$</span>f, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##           Angler Fleckvieh Holstein Rotbunt
## Angler     0.057     0.006    0.046   0.046
## Fleckvieh  0.006     0.073    0.005   0.005
## Holstein   0.046     0.005    0.112   0.094
## Rotbunt    0.046     0.005    0.094   0.111</code></pre>
<p>It can be seen that inbreeding is lowest in Angler cattle and highest
in Rotbunt cattle. The kinship between Holstein and Rotbunt is almost as
high as the kinships within the breeds, so both breeds are closely
related. In contrast, Fleckvieh is only distantly related to all other
breeds included in the data set.</p>
</div>
<div id="genetic-distances-between-breeds" class="section level4">
<h4>Genetic Distances Between Breeds</h4>
<p>Genetic distances between breeds can be computed from the kinships
between breeds. There are various possibilities to define genetic
distances and the method of choice depends on the intended use. The
distance between two breed <span class="math inline">\(i\)</span>, <span class="math inline">\(j\)</span>, defined as <span class="math display">\[\Delta_{bl} =
\sqrt{\frac{f_{bb}+f_{ll}}{2}-f_{bl}}\]</span> can be considered an
estimate of the expected differences in population means for a neutral
polygenic trait <span class="citation">(Wellmann, Bennewitz, and
Meuwissen 2014)</span>. It can be obtained as</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="fu">round</span>(CoreSet<span class="sc">$</span>Dist, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##           Angler Fleckvieh Holstein Rotbunt
## Angler     0.000     0.242    0.196   0.193
## Fleckvieh  0.242     0.000    0.295   0.295
## Holstein   0.196     0.295    0.000   0.130
## Rotbunt    0.193     0.295    0.130   0.000</code></pre>
</div>
<div id="prioritizing-breeds-for-conservation" class="section level4">
<h4>Prioritizing Breeds for Conservation</h4>
<p>Since resources available for conservation are limited, prioritizing
breeds for conservation is of high importance to halt the erosion of
genetic diversity observed in livestock species. This requires to
estimate conservation values of breeds.</p>
<p>In the core set approach, a hypothetical subdivided population is
considered, consisting of individuals from various breeds. This
population is called the core set. The contributions of each breed to
the core set are determined such that the diversity of the core set is
maximized. The conservation value of a particular breed measures how
much the diversity decreases if the breed is removed from the core
set.</p>
<p>Function <a href="https://rdrr.io/cran/optiSel/man/opticomp.html">opticomp</a> can
be used to compute the contributions of the breeds to a core set with
maximum diversity.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>CoreSet <span class="ot">&lt;-</span> <span class="fu">opticomp</span>(sKin, phen)</span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a>CoreSet<span class="sc">$</span>bc</span></code></pre></div>
<pre><code>##     Angler  Fleckvieh   Holstein    Rotbunt 
## 0.47908531 0.42425265 0.04434691 0.05231513</code></pre>
<p>The Rotbunt cattle have only a small contribution to the core set, as
their genes are already present in Angler and Holstein cattle. For this
core set the diversity is</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>CoreSet<span class="sc">$</span>value</span></code></pre></div>
<pre><code>## [1] 0.9656585</code></pre>
<p>Removing the Angler cattle from the core set</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a>CoreSet <span class="ot">&lt;-</span> <span class="fu">opticomp</span>(sKin, phen, <span class="at">ub=</span><span class="fu">c</span>(<span class="at">Angler=</span><span class="dv">0</span>))</span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>CoreSet<span class="sc">$</span>bc</span></code></pre></div>
<pre><code>##    Angler Fleckvieh  Holstein   Rotbunt 
## 0.0000000 0.5910616 0.1876452 0.2212932</code></pre>
<p>increases the Rotbunt and Holstein contributions, and decreases the
diversity of the core set:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a>CoreSet<span class="sc">$</span>value</span></code></pre></div>
<pre><code>## [1] 0.9548693</code></pre>
</div>
</div>
<div id="references" class="section level3 unnumbered">
<h3 class="unnumbered">References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Wellmann2014" class="csl-entry">
Wellmann, R., J. Bennewitz, and T. H. Meuwissen. 2014. <span>“A Unified
Approach to Characterize and Conserve Adaptive and Neutral Genetic
Diversity in Subdivided Populations.”</span> <em>Genetical Research
(Camb)</em> 96.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
